<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jim Zhang" />

<meta name="date" content="2016-06-01" />

<title>Developer gives a name to the analysis to be implemented</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="index_files/highlight/default.css"
      type="text/css" />
<script src="index_files/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="index_files/navigation-1.0/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Developer gives a name to the analysis to be implemented</h1>
<h4 class="author"><em>Jim Zhang</em></h4>
<h4 class="date"><em>2016-06-01</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p><strong>Introduction</strong></p>
<p>Developer of this R Markdown template should describe here the goal, scope, input data and major steps of this analysis.</p>
<p>This skeleton helps developers to implement their own reproducible analysis using the RoCA framework. Visit the <a href="http://zhezhangsh.github.io/RoCA">RoCA site</a> for more information about this project.</p>
<p>Definitation of terms:</p>
<ul>
<li><strong><em>RoCA framework</em></strong> : RoCA is a paradigm of reproducible data analysis for bioinformaticians to share their custom analysis procedure. It creates dynamic data analysis reports based on R Markdown templates. Visit <a href="http://zhezhangsh.github.io/RoCA">RoCA home</a> for more information.</li>
<li><strong><em>R Markdown template</em></strong> : An .Rmd document that includes both R code and documentation of a data analysis procedure. It generates dynamic report according to user-specific input data and parameters for each analysis.</li>
<li><strong><em>YAML key-value pairs</em></strong> : The pairs are specifications of each analysis performed using the R Markdown template. They are formatted and saved together as a <a href="http://www.yaml.org">YAML file</a>. It is highly recommended to pair each R Markdown template with a YAML file having the same file name.</li>
<li><strong><em>Knitr report</em></strong> : Data analysis report in HTML, PDF, or Word file, generated by the <a href="http://yihui.name/knitr">knitr tool</a> for each analysis, based on the R Markdown template and the YAML specifications.</li>
<li><strong><em>Developer</em></strong> : A developer implements a data analysis procedure into a R Markdown template, so the analysis can be repeated by any data analysts using their own data and parameters. The developer should also provide introduction and examples about how to set up an analysis using the template.</li>
<li><strong><em>Analyst</em></strong> : Given an available R Markdown template generated by a developer, an analyst only needs to work with the pairing YAML file to set up a specific analysis. The analyst should follow instructions to prepare input data and choose parameters for each analysis.</li>
</ul>
</div>
<p> </p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<p><!-- plug this line anywhere you want to add a link to the project home page in the report --></p>
<!--
This whole section is dedicated to the <description> block in the YAML file.
Each key-value pair in that block will be written out as a subsection. 
Keep it unchanged for all RoCA templates
-->
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>Reporting of Custom Analysis</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.2</span> Analysis</h2>
<p>Just a skeleton for developer to write their own Rmd templates</p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="the-first-result-section" class="section level1">
<h1><span class="header-section-number">2</span> The first result section <!-- Change section title --></h1>
<!-- replace this paragraph in your template -->
<p><a href="http://rMarkdown.rstudio.com">R Markdown</a> generates dynamic documents using R code for data analysis and <a href="https://daringfireball.net/projects/markdown">Markdown</a> markup language for text. For each analysis step in an R Markdown file, the deveoper should describe the analysis in terms of its goal, rational, method, etc.</p>
<!-- this is a silent code chunk that does some analysis but will not be written into the report -->
<div id="the-first-subsection" class="section level2">
<h2><span class="header-section-number">2.1</span> The first subsection <!-- Change subsection title --></h2>
<!-- replace this paragraph in your template -->
<p>There are two ways to show a table: by using the kable function to include a small table in the final report, or by adding a <a href="table/chromosome.html">link</a> that points to an HTML table in Jave DataTables format. By default, all HTML tables will be saved in the <em>table/</em> subfolder.</p>
<!-- this is a code chunk that prepares for a summary table to be included in the report -->
<div style="color:darkblue; padding:0 3cm">
<strong>Table 1</strong> Write the table title and caption here.
</div>
<div align="center" ,="" style="padding:0 3cm">
<p><!-- Horizontally align the table at the center of the page --> <!-- Use the kable function to directly put a small table in the report --></p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="center">Min.</th>
<th align="center">1st Qu.</th>
<th align="center">Median</th>
<th align="center">Mean</th>
<th align="center">3rd Qu.</th>
<th align="center">Max.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Draw_1</td>
<td align="center">-2.257</td>
<td align="center">-0.6439</td>
<td align="center">0.132400</td>
<td align="center">0.032890</td>
<td align="center">0.7073</td>
<td align="center">2.501</td>
</tr>
<tr class="even">
<td align="left">Draw_2</td>
<td align="center">-2.489</td>
<td align="center">-0.8184</td>
<td align="center">-0.012930</td>
<td align="center">-0.002154</td>
<td align="center">0.7198</td>
<td align="center">2.854</td>
</tr>
<tr class="odd">
<td align="left">Draw_3</td>
<td align="center">-2.622</td>
<td align="center">-0.6294</td>
<td align="center">0.031190</td>
<td align="center">0.035480</td>
<td align="center">0.7299</td>
<td align="center">3.083</td>
</tr>
<tr class="even">
<td align="left">Draw_4</td>
<td align="center">-2.738</td>
<td align="center">-0.7021</td>
<td align="center">-0.004288</td>
<td align="center">-0.007244</td>
<td align="center">0.8318</td>
<td align="center">2.224</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="the-second-subsection" class="section level2">
<h2><span class="header-section-number">2.2</span> The second subsection <!-- Change subsection title --></h2>
<!-- replace this paragraph in your template -->
<p>By default, all figures will be saved in the <em>figure/</em> subfolder as both .pdf and .png files.</p>
<!-- this is a code chunk that will generate a plot to be written into the report -->
<div align="center">
<!-- Horizontally align the figure at the center of the page --> <img src="figure/figure_1-1.png" title="plot of chunk figure_1" alt="plot of chunk figure_1" width="600px" />
</div>
<div style="color:darkblue; padding:0 3cm">
<strong>Figure 1.</strong> Write the figure caption here.
</div>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="the-second-result-section" class="section level1">
<h1><span class="header-section-number">3</span> The second result section <!-- Change section title --></h1>
<!-- replace this paragraph in your template -->
<p>The develop will decide how to split a full data analysis procedure into multiple steps and present the results in different sections and subsection. Be careful with the dependency between steps.</p>
<!-- this is a silent code chunk that does some analysis but will not be written into the report -->
</div>
<div id="more-sections" class="section level1">
<h1><span class="header-section-number">4</span> More sections …</h1>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<!-- Keep this section unchanged for all RoCA templates -->
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">5</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">5.1</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p>
<ul>
<li><em>output</em> : where you want to put the output files</li>
<li><em>home</em> : the URL if you have a home page for your project</li>
<li><em>analyst</em> : your name</li>
<li><em>description</em> : background information about your project, analysis, etc.</li>
<li><em>input</em> : where are your input data, read instruction for preparing them</li>
<li><em>parameter</em> : parameters for this analysis; read instruction about how to prepare input data</li>
</ul></li>
<li><p>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</p></li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">5.2</span> Session information</h2>
<pre><code>## R version 3.2.2 (2015-08-14)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.10.5 (Yosemite)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] GenomicRanges_1.22.4 GenomeInfoDb_1.6.3   IRanges_2.4.8       
##  [4] S4Vectors_0.8.11     BiocGenerics_0.16.1  htmlwidgets_0.5     
##  [7] DT_0.1               awsomics_0.0.0.9000  yaml_2.1.13         
## [10] rmarkdown_0.9.6      knitr_1.12.3         RoCA_0.0.0.9000     
## [13] RCurl_1.95-4.8       bitops_1.0-6         devtools_1.11.1     
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.4     XVector_0.10.0  magrittr_1.5    zlibbioc_1.14.0
##  [5] highr_0.5.1     stringr_1.0.0   tools_3.2.2     withr_1.0.1    
##  [9] htmltools_0.3.5 digest_0.6.9    formatR_1.3     memoise_1.0.0  
## [13] evaluate_0.9    stringi_1.0-1   jsonlite_0.9.20</code></pre>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<hr />
<p><em>END OF DOCUMENT</em></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
