<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jim Zhang" />

<meta name="date" content="2019-08-31" />

<title>De novo motif discovery with rGADEM,</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="index_files/navigation-1.1/tabsets.js"></script>
<link href="index_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="index_files/highlightjs-9.12.0/highlight.js"></script>
<script src="index_files/kePrint-0.0.1/kePrint.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">De novo motif discovery with <strong><em>rGADEM</em></strong>,</h1>
<h4 class="author"><em>Jim Zhang</em></h4>
<h4 class="date"><em>2019-08-31</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p><strong>Introduction</strong></p>
<p>This analysis performs the main functionality of Bioconductor package <strong>rGADEM</strong> to discover de novo motif(s) over-represented in a set of DNA sequences. Please read the package <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/rGADEM/inst/doc/rGADEM.pdf">manual</a> first for background about <strong>GADEM</strong> algorithm and how to choose its parameters.</p>
<p>The set of DNA sequences, within which de novo motifs and their position weighted matrix (PWM) will be discovered, can be provided as a named character vector, an <em>DNAStringSet</em> object, or a set of genomic loci within a reference genome, such as <em><a href="http://bioconductor.org/packages/release/data/annotation/html/BSgenome.Hsapiens.NCBI.GRCh38.html">BSgenome.Hsapiens.NCBI.GRCh38</a></em>.</p>
<p>This analysis includes three major steps:</p>
<ol style="list-style-type: decimal">
<li>The <strong><em>GADEM {rGADEM}</em></strong> is called using given parameters to discover over-presented de novo motifs from a set of DNA sequences.</li>
<li>The motifs and their matches in the DNA sequences are summarized and visualized.</li>
<li>The motifs are searched again within the DNA sequences using the <strong><em>matchPWM {Biostrings}</em></strong> function and a p value cutoff, to identify report all matches and their matching scores.</li>
</ol>
</div>
<p> </p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>rGADAM wrapper</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.2</span> Analysis</h2>
<p>Search for DNA motifs around TSS of a few random genes.</p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="results" class="section level1">
<h1><span class="header-section-number">2</span> Results</h1>
<div id="summarize-sequences" class="section level2">
<h2><span class="header-section-number">2.1</span> Summarize sequences</h2>
<p><strong>GADEM</strong> discovers de novo motifs in a given set of DNA sequences. The length, number, base frequency, and complexity of the sequences could all impact the output of GADEM, such as the specificity of motifs and the significance of matches. It’s preferable to exclude outliers from the analysis, and strong overall bias of any characteristics could be a concern for follow-up analysis.</p>
<p><strong>81</strong> sequences are analyzed by the <strong><em>GADEM {rGADEM}</em></strong> function to discover de novo motifs, using the following parameters:</p>
<ul>
<li>Seed PWM: <em>Spwm</em> = no</li>
<li>Number of EM step: <em>numEM</em> = 80</li>
<li>P value cutoff of matches: <em>pValue</em> = 2e-04</li>
<li>Mask low-complexity sequences: <em>maskR</em> = yes</li>
<li>Maximal number of motifs allowed: <em>nmotifs</em> = 25</li>
</ul>
<p><strong>Table 1.</strong> Summary of DNA sequences.</p>
<ul>
<li><strong><em>Length (bp)</em></strong>: full length of DNA sequences</li>
<li><strong><em>GC (%)</em></strong>: frequency of G and C</li>
<li><strong><em>CpG (%)</em></strong>: frequency of CG or GC sites</li>
<li><strong><em>Duplex (%)</em></strong>: frequency of duplexes (AA, CC, GG, or TT)</li>
</ul>
<p>Click <a href="table/sequence_summary.html">here</a> to view details of individual sequences.</p>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:center;">
Min.
</th>
<th style="text-align:center;">
1st Qu.
</th>
<th style="text-align:center;">
Median
</th>
<th style="text-align:center;">
Mean
</th>
<th style="text-align:center;">
3rd Qu.
</th>
<th style="text-align:center;">
Max.
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Length (bp)
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
</tr>
<tr>
<td style="text-align:left;">
GC (%)
</td>
<td style="text-align:center;">
25.87
</td>
<td style="text-align:center;">
46.27
</td>
<td style="text-align:center;">
56.22
</td>
<td style="text-align:center;">
56.88
</td>
<td style="text-align:center;">
69.65
</td>
<td style="text-align:center;">
86.57
</td>
</tr>
<tr>
<td style="text-align:left;">
CpG (%)
</td>
<td style="text-align:center;">
1.50
</td>
<td style="text-align:center;">
7.00
</td>
<td style="text-align:center;">
12.00
</td>
<td style="text-align:center;">
14.51
</td>
<td style="text-align:center;">
22.00
</td>
<td style="text-align:center;">
34.50
</td>
</tr>
<tr>
<td style="text-align:left;">
Duplex (%)
</td>
<td style="text-align:center;">
20.00
</td>
<td style="text-align:center;">
27.50
</td>
<td style="text-align:center;">
31.00
</td>
<td style="text-align:center;">
31.16
</td>
<td style="text-align:center;">
34.00
</td>
<td style="text-align:center;">
49.00
</td>
</tr>
</tbody>
</table>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
<div id="discover-motifs" class="section level2">
<h2><span class="header-section-number">2.2</span> Discover motifs</h2>
<p><strong>GADEM</strong> is run using given parameters to discover motifs from DNA sequences. It reports the consensus sequence and PWM of each motif, as well as the most significant matches that the consensus and PWM are based upon.</p>
<p><strong>Table 2.</strong> Summary of motifs found from sequences. Click motif name to see full list of matches to each motif and their details. Click consensus sequence to see its PWM.</p>
<ul>
<li><strong><em>GC (%):</em></strong> average GC frequency at all position in PWM.</li>
<li><strong><em>Num_Base:</em></strong> length of the motif.</li>
<li><strong><em>Num_Match:</em></strong> total number of matches to the motif reported by GADEM in any sequences.</li>
<li><strong><em>Num_Seq:</em></strong> number of sequences including one or more matches to the motif</li>
<li><strong><em>RevComp (%):</em></strong> percent of matches on the reversed direction</li>
</ul>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Name
</th>
<th style="text-align:left;">
Consensus
</th>
<th style="text-align:right;">
GC (%)
</th>
<th style="text-align:right;">
Num_Base
</th>
<th style="text-align:right;">
Num_Match
</th>
<th style="text-align:center;">
Num_Seq
</th>
<th style="text-align:left;">
RevComp (%)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
<a href="table/Match_Motif_1.html">Motif_1</a>
</td>
<td style="text-align:left;">
<a href="table/PWM_Motif_1.html">sCCCsCsCCkCysCCnCCCn</a>
</td>
<td style="text-align:right;">
81.16
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
68
</td>
<td style="text-align:center;">
39
</td>
<td style="text-align:left;">
55.9
</td>
</tr>
<tr>
<td style="text-align:center;">
<a href="table/Match_Motif_2.html">Motif_2</a>
</td>
<td style="text-align:left;">
<a href="table/PWM_Motif_2.html">nTTTdTwAAwswnwwrnywrA</a>
</td>
<td style="text-align:right;">
27.98
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:center;">
29
</td>
<td style="text-align:left;">
40.0
</td>
</tr>
<tr>
<td style="text-align:center;">
<a href="table/Match_Motif_3.html">Motif_3</a>
</td>
<td style="text-align:left;">
<a href="table/PWM_Motif_3.html">ssyynCmnmrTCTrGrAwGTGAGGAGCsksTCykC</a>
</td>
<td style="text-align:right;">
58.26
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:center;">
7
</td>
<td style="text-align:left;">
0.0
</td>
</tr>
</tbody>
</table>
<p> </p>
<table>
<thead>
<tr>
<th style="text-align:center;">
PWM Logo
</th>
<th style="text-align:center;">
Matching position
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
<img src="figure/Logo_Motif_1.png" />
</td>
<td style="text-align:center;">
<img src="figure/Position_Motif_1.png" />
</td>
</tr>
<tr>
<td style="text-align:center;">
<img src="figure/Logo_Motif_2.png" />
</td>
<td style="text-align:center;">
<img src="figure/Position_Motif_2.png" />
</td>
</tr>
<tr>
<td style="text-align:center;">
<img src="figure/Logo_Motif_3.png" />
</td>
<td style="text-align:center;">
<img src="figure/Position_Motif_3.png" />
</td>
</tr>
</tbody>
</table>
<p><strong>Figure 1.</strong> Motif visualization. The right panel shows the logo of the consensus sequences based on their PWM. The y-axis represents the frequency of nucleotides at each position. The left panel shows the relative positions of the matches to the motif within the sequences. The positions are directional based on the strand of the matches and the sequences in genomne.</p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
<div id="search-motifs" class="section level2">
<h2><span class="header-section-number">2.3</span> Search motifs</h2>
<p>The following steps are applied to identify more matches to the motifs discovered by <strong>GADEM</strong> and properly evaluate their significance:</p>
<ul>
<li>Calculate overall base frequency in all DNA sequences.</li>
<li>For each motif, draw 100000 random sequences if the same length based on overall base frequency.</li>
<li>Get the background distribution of matching scores based on the agreement between all random sequences and the motif.</li>
<li>Obtain a cutoff matching score based on the background distribution with an empirical p value less than 0.005</li>
<li>Search and report matches to the motif in DNA sequences using the cutoff matching score, using the *<strong>matchPWM {Biostrings}</strong> function.</li>
</ul>
<p><strong>Table 3.</strong> Summary of matches to the motifs:</p>
<ul>
<li><strong><em>Motif:</em></strong> nanme of motifs, click to view full list of matches</li>
<li><strong><em>Len_Motif:</em></strong> length of motifs</li>
<li><strong><em>Match_Per_Kb:</em></strong> number of matches in each kilobase of DNA sequences</li>
<li><strong><em>Num_Match:</em></strong> total number of matches</li>
<li><strong><em>Num_Seq:</em></strong> number of sequences containing one or more matches</li>
<li><strong><em>RevComp (%):</em></strong> percent of matches on the opposite direction</li>
<li><strong><em>Min_Score, Mean_Score, Max_Score:</em></strong> range of matching scores (0 - 100)</li>
</ul>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Motif
</th>
<th style="text-align:right;">
Len_Motif
</th>
<th style="text-align:right;">
Match_Per_Kb
</th>
<th style="text-align:right;">
Num_Match
</th>
<th style="text-align:right;">
Num_Seq
</th>
<th style="text-align:right;">
RevComp (%)
</th>
<th style="text-align:right;">
Min_Score
</th>
<th style="text-align:right;">
Mean_Score
</th>
<th style="text-align:right;">
Max_Score
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="table/Rematch_Motif_1.html">Motif_1</a>
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
11.664
</td>
<td style="text-align:right;">
171
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
58.48
</td>
<td style="text-align:right;">
67.0
</td>
<td style="text-align:right;">
76.72
</td>
<td style="text-align:right;">
93.8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="table/Rematch_Motif_2.html">Motif_2</a>
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:right;">
10.082
</td>
<td style="text-align:right;">
147
</td>
<td style="text-align:right;">
57
</td>
<td style="text-align:right;">
46.26
</td>
<td style="text-align:right;">
66.1
</td>
<td style="text-align:right;">
74.06
</td>
<td style="text-align:right;">
92.9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="table/Rematch_Motif_3.html">Motif_3</a>
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
8.181
</td>
<td style="text-align:right;">
110
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
51.82
</td>
<td style="text-align:right;">
57.3
</td>
<td style="text-align:right;">
62.82
</td>
<td style="text-align:right;">
98.8
</td>
</tr>
</tbody>
</table>
<p> </p>
<div class="figure">
<img src="figure/rematch_position.png" />

</div>
<p><strong>Figure 2.</strong> Distribution of relative positions of motif matches within the DNA sequences. Grey bar is the number of matches within each 5% interval and blue star is the average Phred score (-log10 transformed p value) of matches within the interval.</p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="download" class="section level1">
<h1><span class="header-section-number">3</span> Download</h1>
<ul>
<li><strong>Motif_1:</strong> <a href="table/PWM_Motif_1.txt">PWM</a> <strong>|</strong> <a href="table/Match_Motif_1.txt">GADEM hits</a> <strong>|</strong> <a href="table/Rematch_Motif_1.txt">All matches</a></li>
<li><strong>Motif_2:</strong> <a href="table/PWM_Motif_2.txt">PWM</a> <strong>|</strong> <a href="table/Match_Motif_2.txt">GADEM hits</a> <strong>|</strong> <a href="table/Rematch_Motif_2.txt">All matches</a></li>
<li><strong>Motif_3:</strong> <a href="table/PWM_Motif_3.txt">PWM</a> <strong>|</strong> <a href="table/Match_Motif_3.txt">GADEM hits</a> <strong>|</strong> <a href="table/Rematch_Motif_3.txt">All matches</a></li>
</ul>
<hr />
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">4</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">4.1</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p>
<ul>
<li><em>output</em> : where you want to put the output files</li>
<li><em>home</em> : the URL if you have a home page for your project</li>
<li><em>analyst</em> : your name</li>
<li><em>description</em> : background information about your project, analysis, etc.</li>
<li><em>input</em> : where are your input data, read instruction for preparing them</li>
<li><em>parameter</em> : parameters for this analysis; read instruction about how to prepare input data</li>
</ul></li>
<li><p>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</p></li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">4.2</span> Session information</h2>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      stats4    parallel  stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] BSgenome.Hsapiens.NCBI.GRCh38_1.3.1000
##  [2] Agri_0.0.0.9000                       
##  [3] rGADEM_2.30.0                         
##  [4] seqLogo_1.48.0                        
##  [5] BSgenome_1.50.0                       
##  [6] rtracklayer_1.42.1                    
##  [7] Biostrings_2.50.1                     
##  [8] XVector_0.22.0                        
##  [9] GenomicFeatures_1.34.1                
## [10] AnnotationDbi_1.44.0                  
## [11] Biobase_2.42.0                        
## [12] GenomicRanges_1.34.0                  
## [13] GenomeInfoDb_1.18.1                   
## [14] IRanges_2.16.0                        
## [15] S4Vectors_0.20.1                      
## [16] BiocGenerics_0.28.0                   
## [17] htmlwidgets_1.3                       
## [18] DT_0.5                                
## [19] kableExtra_0.9.0                      
## [20] awsomics_0.0.0.9000                   
## [21] yaml_2.2.0                            
## [22] rmarkdown_1.10                        
## [23] knitr_1.20                            
## [24] RoCA_0.0.0.9000                       
## [25] RCurl_1.95-4.11                       
## [26] bitops_1.0-6                          
## [27] usethis_1.4.0                         
## [28] devtools_2.0.1                        
## 
## loaded via a namespace (and not attached):
##  [1] matrixStats_0.54.0          fs_1.2.6                   
##  [3] bit64_0.9-7                 progress_1.2.0             
##  [5] httr_1.3.1                  rprojroot_1.3-2            
##  [7] tools_3.5.1                 backports_1.1.2            
##  [9] R6_2.3.0                    DBI_1.0.0                  
## [11] colorspace_1.3-2            withr_2.1.2                
## [13] prettyunits_1.0.2           processx_3.2.0             
## [15] bit_1.1-14                  compiler_3.5.1             
## [17] cli_1.0.1                   rvest_0.3.2                
## [19] xml2_1.2.0                  desc_1.2.0                 
## [21] DelayedArray_0.8.0          scales_1.0.0               
## [23] readr_1.3.1                 callr_3.0.0                
## [25] stringr_1.3.1               digest_0.6.18              
## [27] Rsamtools_1.34.0            base64enc_0.1-3            
## [29] pkgconfig_2.0.2             htmltools_0.3.6            
## [31] sessioninfo_1.1.1           highr_0.7                  
## [33] rlang_0.4.0                 rstudioapi_0.8             
## [35] RSQLite_2.1.1               shiny_1.2.0                
## [37] jsonlite_1.5                crosstalk_1.0.0            
## [39] BiocParallel_1.16.2         magrittr_1.5               
## [41] GenomeInfoDbData_1.2.0      Matrix_1.2-15              
## [43] Rcpp_1.0.0                  munsell_0.5.0              
## [45] stringi_1.2.4               SummarizedExperiment_1.12.0
## [47] zlibbioc_1.28.0             pkgbuild_1.0.2             
## [49] blob_1.1.1                  promises_1.0.1             
## [51] crayon_1.3.4                lattice_0.20-38            
## [53] hms_0.4.2                   ps_1.2.1                   
## [55] pillar_1.3.0                biomaRt_2.38.0             
## [57] pkgload_1.0.2               XML_3.98-1.16              
## [59] glue_1.3.0                  evaluate_0.12              
## [61] remotes_2.0.2               httpuv_1.4.5               
## [63] testthat_2.2.1              assertthat_0.2.0           
## [65] mime_0.6                    xtable_1.8-3               
## [67] later_0.7.5                 viridisLite_0.3.0          
## [69] tibble_1.4.2                GenomicAlignments_1.18.0   
## [71] memoise_1.1.0</code></pre>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<hr />
<p><em>END OF DOCUMENT</em></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
