<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Zhe Zhang" />

<meta name="date" content="2016-06-27" />

<title>Gene expression correlated to a series of conditions</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="index_files/highlight/default.css"
      type="text/css" />
<script src="index_files/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="index_files/navigation-1.0/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gene expression correlated to a series of conditions</h1>
<h4 class="author"><em>Zhe Zhang</em></h4>
<h4 class="date"><em>2016-06-27</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p><strong>Introduction: </strong> The goal of this analysis is to find genes in a microarray or RNA-seq data set correlated to a series of sample feature, such as time and drug dosage. It first calculates the correlation between the sample feature and the expression level of each gene. Depending on the nature of the sample feature, the correlation could be linear or non-linear. The genes with the strongest correlation are then selected based on their correlation coefficient, p value, false discovery rate, and range across samples. The selected genes are further analyzed to identify pre-defined gene sets over-represented in these genes.</p>
</div>
<p> </p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>Perturbed rhythmic activation of signaling pathways in mice deficient for Sterol Carrier Protein 2-dependent diurnal lipid transport and metabolism. (<strong><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67426">GSE67426</a></strong>)</p>
</div>
<div id="pubmed" class="section level2">
<h2><span class="header-section-number">1.2</span> PubMed</h2>
<p>Jouffe C, Gobet C, Martin E, Métairon S et al. Perturbed rhythmic activation of signaling pathways in mice deficient for Sterol Carrier Protein 2-dependent diurnal lipid transport and metabolism. Sci Rep 2016 Apr 21;6:24631. PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/27097688">27097688</a>.</p>
</div>
<div id="experimental-design" class="section level2">
<h2><span class="header-section-number">1.3</span> Experimental design</h2>
<p>Comparison of liver mRNA expression from Scp2 KO and wild-type mice harvested every 2 hours during 3 consecutive days. Circadian rhythm in mouse liver, wild type mice only, 3 replicates every 2 hours (12 groups).</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.4</span> Analysis</h2>
<p>Each gene was adjusted to its 0 hour mean and rescaled to make its standard deviation equal to 0. This analysis looks for genes with continuous increased or decreased expression from 0 (midnight) to 12 hour (noon).</p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="data-and-samples" class="section level2">
<h2><span class="header-section-number">1.5</span> Data and samples</h2>
<ul>
<li>There are <strong>12065</strong> total genes.</li>
<li>There are <strong>21</strong> total <a href="table/sample.html">samples</a>.</li>
<li>There are <strong>7</strong> sample groups: <strong><em>WT_00Hr; WT_02Hr; WT_04Hr; WT_06Hr; WT_08Hr; WT_10Hr; WT_12Hr</em></strong>.</li>
</ul>
<p>The input data matrix was normalized using sample group <em>WT_00Hr</em> as control, so the data of each gene was substracted by control group mean and had SD equal to 1.0</p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="analysis-and-results" class="section level1">
<h1><span class="header-section-number">2</span> Analysis and results</h1>
<div id="selection-of-correlated-genes" class="section level2">
<h2><span class="header-section-number">2.1</span> Selection of correlated genes</h2>
<div style="color:darkblue; padding:0 2cm">
<strong>Table 1</strong> Selected genes positively or negatively correlated to time of the day. Click on the clicks veiw full gene list. Summary statistics are the maximums of p value and FDR and the minimums of correlation coefficient and data range of selected genes.
</div>
<div align="center" ,="" style="padding:0 2cm">
<table>
<thead>
<tr class="header">
<th align="left">Direction</th>
<th align="right">Num_Gene</th>
<th align="right">Min_Corr</th>
<th align="right">Max_Pval</th>
<th align="right">Max_FDR</th>
<th align="right">Min_Range</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="table/stat_positive.html">Positive</a></td>
<td align="right">303</td>
<td align="right">0.62</td>
<td align="right">0.0029</td>
<td align="right">0.05</td>
<td align="right">2.3320</td>
</tr>
<tr class="even">
<td align="left"><a href="table/stat_negative.html">Negative</a></td>
<td align="right">392</td>
<td align="right">0.62</td>
<td align="right">0.0029</td>
<td align="right">0.05</td>
<td align="right">2.7194</td>
</tr>
</tbody>
</table>
</div>
<div align="center">
<img src="figure/average_trend-1.png" title="plot of chunk average_trend" alt="plot of chunk average_trend" width="720px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 1.</strong> The averages of normalized expression level of selected genes positively (red) and negatively (blue) correlated to time of the day. The larger diamonds are the averages of sample groups and the smaller ones correspond to the individual samples.
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="gene-set-enrichment-analysis" class="section level2">
<h2><span class="header-section-number">2.2</span> Gene set enrichment analysis</h2>
<p>Find predefined gene sets over-represented in the two groups of genes correlated to time of the day.</p>
<div style="color:darkblue; padding:0 2cm">
<strong>Table 2</strong> Predefined gene sets significantly over-represented in the genes correlated to the time of the day. The test is run on the two groups of genes with positive and negative correlation separately and then jointly. Click on links to see full tables of statistical results.
</div>
<div align="center" ,="" style="padding:0 2cm">
<table>
<thead>
<tr class="header">
<th align="left">BioSystems</th>
<th align="left">KEGG</th>
<th align="left">MSigDb</th>
<th align="left">PubTator</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="ora/positive/BioSystems.html">743</a></td>
<td align="left"><a href="ora/positive/KEGG.html">63</a></td>
<td align="left"><a href="ora/positive/MSigDb.html">936</a></td>
<td align="left"><a href="ora/positive/PubTator.html">35</a></td>
</tr>
<tr class="even">
<td align="left"><a href="ora/negative/BioSystems.html">513</a></td>
<td align="left"><a href="ora/negative/KEGG.html">46</a></td>
<td align="left"><a href="ora/negative/MSigDb.html">883</a></td>
<td align="left"><a href="ora/negative/PubTator.html">8</a></td>
</tr>
<tr class="odd">
<td align="left"><a href="ora/both/BioSystems.html">909</a></td>
<td align="left"><a href="ora/both/KEGG.html">75</a></td>
<td align="left"><a href="ora/both/MSigDb.html">1385</a></td>
<td align="left"><a href="ora/both/PubTator.html">17</a></td>
</tr>
</tbody>
</table>
</div>
<div align="center">
<img src="figure/heatmap-1.png" title="plot of chunk heatmap" alt="plot of chunk heatmap" width="800px" />
</div>
<div style="color:darkblue; padding:0 1cm">
<strong>Figure 2.</strong> Heatmap of all genes correlated, positively or negatively, to time of the day. The colors represent the normalized expression levels (red = higher).
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
</div>
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">3</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">3.1</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p>
<ul>
<li><em>output</em> : where you want to put the output files</li>
<li><em>home</em> : the URL if you have a home page for your project</li>
<li><em>analyst</em> : your name</li>
<li><em>description</em> : background information about your project, analysis, etc.</li>
<li><em>input</em> : where are your input data, read instruction for preparing them</li>
<li><em>parameter</em> : parameters for this analysis; read instruction about how to prepare input data</li>
</ul></li>
<li><p>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</p></li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report. ## Session information</p>
<pre><code>## R version 3.2.2 (2015-08-14)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.10.5 (Yosemite)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] gplots_3.0.1        rchive_0.0.0.9000   htmlwidgets_0.6    
##  [4] DT_0.1              awsomics_0.0.0.9000 yaml_2.1.13        
##  [7] rmarkdown_0.9.6     knitr_1.13          RoCA_0.0.0.9000    
## [10] RCurl_1.95-4.8      bitops_1.0-6        devtools_1.12.0    
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.5        magrittr_1.5       stringr_1.0.0     
##  [4] highr_0.6          caTools_1.17.1     tools_3.2.2       
##  [7] KernSmooth_2.23-15 withr_1.0.2        htmltools_0.3.5   
## [10] gtools_3.5.0       digest_0.6.9       formatR_1.4       
## [13] memoise_1.0.0      evaluate_0.9       gdata_2.17.0      
## [16] stringi_1.1.1      jsonlite_0.9.22</code></pre>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
<p><em>END OF DOCUMENT</em></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
