---
title: "Summarize two pairwise comparisons of differential expression"
author: "Zhe Zhang"
date: "2016-06-30"
output:
  html_document:
    number_sections: yes
    self_contained: no
    toc: yes
    toc_float:
      collapsed: no
---

<div style="border:black 1px solid; padding: 0.5cm 0.5cm">

**Introduction**: This analysis is based on the outputs of pairwise comparisons of differential gene expression generated by [this](https://raw.githubusercontent.com/zhezhangsh/RoCA/master/template/de/pairwise_comparison/pairwise_comparison.Rmd) template. It uses results from 3 pairwise comparisons of 3 sample groups vs. their corresponding control groups and compares how these 3 sample groups are different from each other in terms of their sample-control differences (_delta-delta_). An example of such analysis is the different responses of 3 cell types to the treatment of the same drug. This analysis is focused on the overlapping of differentially expression at both gene and gene set levels.

</div>

&nbsp;





<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div> 

# Description
  
## Project


Transcriptome in immune cells of control-patient samples


## Data


Rna-seq data was generated from of 2 types of immune cells of 3 controls and 3 patients. Raw data was processed to get gene-level read counts. Pairwise comparisons were performed between controls and patients in each immune cell.


## Analysis


This is a demo.



## Pairwise comparisons

This report compares the results of the  following pairwise comparisons.  



<div style="color:darkblue; padding:0 1cm">
**Table 1.** Information about both comparisons, including comparison name, group names, group size, total number of genes, etc. 
</div>
  
<div align='center', style="padding:0 1cm">


|  Name  | Group1  | Group2 | Num1 | Num2 | Num_Gene | Test  | Paired | DEG_Higher | DEG_Lower |
|:------:|:-------:|:------:|:----:|:----:|:--------:|:-----:|:------:|:----------:|:---------:|
| B_Cell | Control |  SLE   |  3   |  3   |  23272   | EdgeR | FALSE  |    984     |   1853    |
| T_Cell | Control |  SLE   |  3   |  3   |  23272   | EdgeR | FALSE  |    693     |    671    |


</div>

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div> 

# Gene-level comparison



<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div>

## Global delta-delta correlation

Both comparisons reported the log ratio of 2 group means for each gene. The global agreement of log ratios of all genes indicates how much the results of these 2 comparisons are similar to or different from each other. Full table of gene-level statistics side-by-side is [here](table/gene_table.html). 

<div align='center'>
<img src="FIGURE/log_ratio-1.png" title="plot of chunk log_ratio" alt="plot of chunk log_ratio" width="640px" />
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 1.** This plot shows the global correlation (correlation coefficient = 0.244) between the 2 pairwise comparisons: **B_Cell** and **T_Cell**. Genes having p values less than 0.01 from both comparisons are highlighted. 
</div>

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div>

## Differentially expressed genes (DEGs)



Both comparisons identified DEGs between 2 compared groups. Overlapped DEGs identified by both comparisons are worthy of a closer look. 

<div style="color:darkblue; padding:0 3cm">
**Table 2.** Number of DEGs:
</div>

<div align='center', style="padding:0 3cm">


|       | Total_gene | P < 0.01 | DEG, > control | DEG, < control |
|:------|:----------:|:--------:|:--------------:|:--------------:|
|B_Cell |   23272    |   3235   |      984       |      1853      |
|T_Cell |   23272    |   2130   |      693       |      671       |


</div>

<div align='center'>
<img src="FIGURE/deg_overlap-1.png" title="plot of chunk deg_overlap" alt="plot of chunk deg_overlap" width="800px" />
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 2.** Overlapping of DEGs. All combinations of differential expression towards opposite directions are plotted and Fisher's exact test is performed to evaluate the significance of overlapping or lack of overlapping. 
Click links below to view overlapping DEGs:

  - [B_Cell::Higher_in_SLE and T_Cell::Higher_in_SLE](DEG/gene_up-up.html)
  - [B_Cell::Higher_in_SLE and T_Cell::Lower_in_SLE](DEG/gene_up-down.html)
  - [B_Cell::Lower_in_SLE and T_Cell::Higher_in_SLE](DEG/gene_down-up.html)
  - [B_Cell::Lower_in_SLE and T_Cell::Lower_in_SLE](DEG/gene_down-down.html)
</div>

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div>

## ANOVA



2-way ANOVA analysis is performed to identify genes responding to **_SLE_** differently in different **_Cell_**. The analysis reported 3 p values, corresponding to the effect of **_SLE_**, **_Cell_**, and their interaction. The analysis identified 1127 significant genes with interaction p values less than 0.01. The ANOVA results are summarized in a table [here](table/anova_table.html). 

<div align='center'>
<img src="FIGURE/aov_top-1.png" title="plot of chunk aov_top" alt="plot of chunk aov_top" width="800px" />
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 3.** Examples: the top 4 genes having the most significant interactive p value, among the genes with significant differential expression in at least one of the two pairwise comparisons. 
</div>

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div>

# Gene set-level comparison

Genes are often grouped into pre-defined gene sets according to their function, interaction, location, etc. Analysis then can be performed on genes in the same gene set as a unit instead of individual genes.

## Gene set average



Average differential expression of genes in the same gene set. The gene set-level statistics were fully summarized in this table [here](table/geneset_table.html).

<div align='center'>
<img src="FIGURE/geneset_average_plot-1.png" title="plot of chunk geneset_average_plot" alt="plot of chunk geneset_average_plot" width="800px" />
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 4.**  Each dot represents a gene set and the average log-ratio of all genes in this gene set. The averages were calculated with the log-ratio value of genes (_left panel_) and the absolute of the log-ratios (_right panel_). The correlation coefficients are 0.4668 and 0.4779 respectively. 
</div>

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div>

## Gene set over-representation analysis (ORA)



Each 2-group comparison performs gene set over-representation analysis (ORA) that identifies gene sets over-represented with differentially expressed genes. The results of ORA of both 2-group comparisons are summarized and compared here.  The ORA of each gene set reports an odds ratio and p value. These statistics from both comparisons were combined and listed side-by-side, as well as the difference of their odds ratios and ratio of their p values (p set to 0.5 when not available), in this table [here](table/ora_table.html)

<div style="color:darkblue; padding:0 3cm">
**Table 3.** Gene sets were broken down into subgroups by their sources. Click on the numbers of over-represented gene sets to see a full list.
</div>

<div align='center', style="padding:0 3cm">


|           |        B_Cell::Higher_in_Control         |          B_Cell::Higher_in_SLE           |        T_Cell::Higher_in_Control         |          T_Cell::Higher_in_SLE           |
|:----------|:----------------------------------------:|:----------------------------------------:|:----------------------------------------:|:----------------------------------------:|
|BioSystems | [438](ORA/BioSystems_Higher_in_SLE.html) | [3212](ORA/BioSystems_Lower_in_SLE.html) | [438](ORA/BioSystems_Higher_in_SLE.html) | [3212](ORA/BioSystems_Lower_in_SLE.html) |
|KEGG       |    [40](ORA/KEGG_Higher_in_SLE.html)     |    [319](ORA/KEGG_Lower_in_SLE.html)     |    [40](ORA/KEGG_Higher_in_SLE.html)     |    [319](ORA/KEGG_Lower_in_SLE.html)     |
|MSigDb     |   [857](ORA/MSigDb_Higher_in_SLE.html)   |   [4125](ORA/MSigDb_Lower_in_SLE.html)   |   [857](ORA/MSigDb_Higher_in_SLE.html)   |   [4125](ORA/MSigDb_Lower_in_SLE.html)   |
|OMIM       |     [0](ORA/OMIM_Higher_in_SLE.html)     |     [1](ORA/OMIM_Lower_in_SLE.html)      |     [0](ORA/OMIM_Higher_in_SLE.html)     |     [1](ORA/OMIM_Lower_in_SLE.html)      |
|PubTator   |  [123](ORA/PubTator_Higher_in_SLE.html)  |  [7634](ORA/PubTator_Lower_in_SLE.html)  |  [123](ORA/PubTator_Higher_in_SLE.html)  |  [7634](ORA/PubTator_Lower_in_SLE.html)  |


</div>

<div align='center'>
<img src="FIGURE/ora_overlap-1.png" title="plot of chunk ora_overlap" alt="plot of chunk ora_overlap" width="800px" />
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 5.** The overlapping of over-represented gene sets from both comparisons. Click links below to view tables of overlapping significant gene sets:

  - [B_Cell::Higher_in_SLE and T_Cell::Higher_in_SLE](ORA/ora_up-up.html)
  - [B_Cell::Higher_in_SLE and T_Cell::Lower_in_SLE](ORA/ora_up-down.html)
  - [B_Cell::Lower_in_SLE and T_Cell::Higher_in_SLE](ORA/ora_down-up.html)
  - [B_Cell::Lower_in_SLE and T_Cell::Lower_in_SLE](ORA/ora_down-down.html)
</div>

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div>

## Gene set enrichment analysis (GSEA) 



Each 2-group comparison performs gene set enrichment analysis (GSEA) on genes ranked by their differential expression. The results of GSEA of both 2-group comparisons are summarized and compared here. The GSEA of each gene set reports an enrichment score and p value. These statistics from both comparisons were combined and listed side-by-side in this table [here](table/gsea_table.html)

<div style="color:darkblue; padding:0 3cm">
**Table 4.** Gene sets were broken down into subgroups by collections. Click on the numbers of enriched gene sets to see a full list. 
</div>

<div align='center', style="padding:0 3cm">


|                                      |                      B_Cell::Higher_in_Control                      |                        B_Cell::Higher_in_SLE                         |                      T_Cell::Higher_in_Control                      |                        T_Cell::Higher_in_SLE                         |
|:-------------------------------------|:-------------------------------------------------------------------:|:--------------------------------------------------------------------:|:-------------------------------------------------------------------:|:--------------------------------------------------------------------:|
|C0_Hallmark                           |              [2](GSEA/C0_Hallmark_Higher_in_SLE.html)               |               [37](GSEA/C0_Hallmark_Lower_in_SLE.html)               |              [2](GSEA/C0_Hallmark_Higher_in_SLE.html)               |               [37](GSEA/C0_Hallmark_Lower_in_SLE.html)               |
|C1_Positional                         |             [13](GSEA/C1_Positional_Higher_in_SLE.html)             |              [26](GSEA/C1_Positional_Lower_in_SLE.html)              |             [13](GSEA/C1_Positional_Higher_in_SLE.html)             |              [26](GSEA/C1_Positional_Lower_in_SLE.html)              |
|C2_BioCarta_Pathways                  |          [1](GSEA/C2_BioCarta_Pathways_Higher_in_SLE.html)          |          [68](GSEA/C2_BioCarta_Pathways_Lower_in_SLE.html)           |          [1](GSEA/C2_BioCarta_Pathways_Higher_in_SLE.html)          |          [68](GSEA/C2_BioCarta_Pathways_Lower_in_SLE.html)           |
|C2_Chemical_and_genetic_perturbations | [36](GSEA/C2_Chemical_and_genetic_perturbations_Higher_in_SLE.html) | [1356](GSEA/C2_Chemical_and_genetic_perturbations_Lower_in_SLE.html) | [36](GSEA/C2_Chemical_and_genetic_perturbations_Higher_in_SLE.html) | [1356](GSEA/C2_Chemical_and_genetic_perturbations_Lower_in_SLE.html) |
|C3_MicroRNA_targets                   |          [0](GSEA/C3_MicroRNA_targets_Higher_in_SLE.html)           |           [51](GSEA/C3_MicroRNA_targets_Lower_in_SLE.html)           |          [0](GSEA/C3_MicroRNA_targets_Higher_in_SLE.html)           |           [51](GSEA/C3_MicroRNA_targets_Lower_in_SLE.html)           |
|C3_TF_targets                         |             [4](GSEA/C3_TF_targets_Higher_in_SLE.html)              |             [284](GSEA/C3_TF_targets_Lower_in_SLE.html)              |             [4](GSEA/C3_TF_targets_Higher_in_SLE.html)              |             [284](GSEA/C3_TF_targets_Lower_in_SLE.html)              |
|C4_Cancer_gene_neighborhoods          |     [42](GSEA/C4_Cancer_gene_neighborhoods_Higher_in_SLE.html)      |      [86](GSEA/C4_Cancer_gene_neighborhoods_Lower_in_SLE.html)       |     [42](GSEA/C4_Cancer_gene_neighborhoods_Higher_in_SLE.html)      |      [86](GSEA/C4_Cancer_gene_neighborhoods_Lower_in_SLE.html)       |
|C4_Cancer_modules                     |           [10](GSEA/C4_Cancer_modules_Higher_in_SLE.html)           |           [176](GSEA/C4_Cancer_modules_Lower_in_SLE.html)            |           [10](GSEA/C4_Cancer_modules_Higher_in_SLE.html)           |           [176](GSEA/C4_Cancer_modules_Lower_in_SLE.html)            |
|C6_Oncogenic_signatures               |        [2](GSEA/C6_Oncogenic_signatures_Higher_in_SLE.html)         |        [116](GSEA/C6_Oncogenic_signatures_Lower_in_SLE.html)         |        [2](GSEA/C6_Oncogenic_signatures_Higher_in_SLE.html)         |        [116](GSEA/C6_Oncogenic_signatures_Lower_in_SLE.html)         |
|C7_Immunologic_signatures             |       [58](GSEA/C7_Immunologic_signatures_Higher_in_SLE.html)       |       [922](GSEA/C7_Immunologic_signatures_Lower_in_SLE.html)        |       [58](GSEA/C7_Immunologic_signatures_Higher_in_SLE.html)       |       [922](GSEA/C7_Immunologic_signatures_Lower_in_SLE.html)        |
|GO_BP                                 |                [145](GSEA/GO_BP_Higher_in_SLE.html)                 |                 [2065](GSEA/GO_BP_Lower_in_SLE.html)                 |                [145](GSEA/GO_BP_Higher_in_SLE.html)                 |                 [2065](GSEA/GO_BP_Lower_in_SLE.html)                 |
|GO_CC                                 |                 [67](GSEA/GO_CC_Higher_in_SLE.html)                 |                 [159](GSEA/GO_CC_Lower_in_SLE.html)                  |                 [67](GSEA/GO_CC_Higher_in_SLE.html)                 |                 [159](GSEA/GO_CC_Lower_in_SLE.html)                  |
|GO_MF                                 |                 [44](GSEA/GO_MF_Higher_in_SLE.html)                 |                 [359](GSEA/GO_MF_Lower_in_SLE.html)                  |                 [44](GSEA/GO_MF_Higher_in_SLE.html)                 |                 [359](GSEA/GO_MF_Lower_in_SLE.html)                  |
|KEGG_compound                         |             [4](GSEA/KEGG_compound_Higher_in_SLE.html)              |             [126](GSEA/KEGG_compound_Lower_in_SLE.html)              |             [4](GSEA/KEGG_compound_Higher_in_SLE.html)              |             [126](GSEA/KEGG_compound_Lower_in_SLE.html)              |
|KEGG_enzyme                           |              [1](GSEA/KEGG_enzyme_Higher_in_SLE.html)               |               [1](GSEA/KEGG_enzyme_Lower_in_SLE.html)                |              [1](GSEA/KEGG_enzyme_Higher_in_SLE.html)               |               [1](GSEA/KEGG_enzyme_Lower_in_SLE.html)                |
|KEGG_module                           |              [11](GSEA/KEGG_module_Higher_in_SLE.html)              |               [13](GSEA/KEGG_module_Lower_in_SLE.html)               |              [11](GSEA/KEGG_module_Higher_in_SLE.html)              |               [13](GSEA/KEGG_module_Lower_in_SLE.html)               |
|KEGG_pathway                          |              [9](GSEA/KEGG_pathway_Higher_in_SLE.html)              |              [161](GSEA/KEGG_pathway_Lower_in_SLE.html)              |              [9](GSEA/KEGG_pathway_Higher_in_SLE.html)              |              [161](GSEA/KEGG_pathway_Lower_in_SLE.html)              |
|KEGG_reaction                         |             [2](GSEA/KEGG_reaction_Higher_in_SLE.html)              |              [35](GSEA/KEGG_reaction_Lower_in_SLE.html)              |             [2](GSEA/KEGG_reaction_Higher_in_SLE.html)              |              [35](GSEA/KEGG_reaction_Lower_in_SLE.html)              |
|OMIM_gene                             |               [1](GSEA/OMIM_gene_Higher_in_SLE.html)                |                [2](GSEA/OMIM_gene_Lower_in_SLE.html)                 |               [1](GSEA/OMIM_gene_Higher_in_SLE.html)                |                [2](GSEA/OMIM_gene_Lower_in_SLE.html)                 |
|REACTOME                              |               [92](GSEA/REACTOME_Higher_in_SLE.html)                |                [283](GSEA/REACTOME_Lower_in_SLE.html)                |               [92](GSEA/REACTOME_Higher_in_SLE.html)                |                [283](GSEA/REACTOME_Lower_in_SLE.html)                |
|WikiPathways                          |              [2](GSEA/WikiPathways_Higher_in_SLE.html)              |              [91](GSEA/WikiPathways_Lower_in_SLE.html)               |              [2](GSEA/WikiPathways_Higher_in_SLE.html)              |              [91](GSEA/WikiPathways_Lower_in_SLE.html)               |


</div>

<div align='center'>
<img src="FIGURE/nes-1.png" title="plot of chunk nes" alt="plot of chunk nes" width="640px" />
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 6.** Nominal enrichment scores from both comparisons. Each dot represents a gene set. Gene sets with p values less than 0.01 from both comparisons are highlighted. 
</div>

<div align='center'>
<img src="FIGURE/gsea_overlap-1.png" title="plot of chunk gsea_overlap" alt="plot of chunk gsea_overlap" width="800px" />
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 7.** The overlapping of over-represented gene sets from both comparisons. Click links to view tables of overlapping significant gene sets from GSEA:

  - [B_Cell::Higher_in_SLE and T_Cell::Higher_in_SLE](GSEA/gsea_up-up.html)
  - [B_Cell::Higher_in_SLE and T_Cell::Lower_in_SLE](GSEA/gsea_up-down.html)
  - [B_Cell::Lower_in_SLE and T_Cell::Higher_in_SLE](GSEA/gsea_down-up.html)
  - [B_Cell::Lower_in_SLE and T_Cell::Lower_in_SLE](GSEA/gsea_down-down.html)
</div>

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div>

## Gene clustering



The top 1000 genes with significant ANOVA p values (p <= 'r prms$geneset$cluster$panova') were used as seeds to perform a gene-gene clustering analysis and 5 clusters were identified. ORA was performed on the clusters to identify their functional association (see table below); 



<div style="color:darkblue; padding:0 2cm">
**Table 5.** This table lists the number of genes in each cluster (click the numbers to see gene lists), the average expression of all genes in a cluster of all sample groups, and then the gene sets over-represented in each cluster (click the numbers to see gene set lists). The gene expression levels were normalized so the mean of the control groups equals to 0 and the mean of the treatment groups is the number of standard deviations. </div>

<div align='center', style="padding:0 2cm">


|                     ID                     |              Size              | B_Cell::Control | B_Cell::SLE | T_Cell::Control | T_Cell::SLE |              Gene_set              |
|:------------------------------------------:|:------------------------------:|:---------------:|:-----------:|:---------------:|:-----------:|:----------------------------------:|
| [Cluster_1](CLUSTER/Heatmap_Cluster_1.pdf) | [895](CLUSTER/Cluster_1.html)  |        0        |   -1.6039   |        0        |   -1.5890   | [4328](CLUSTER/ORA_Cluster_1.html) |
| [Cluster_2](CLUSTER/Heatmap_Cluster_2.pdf) | [781](CLUSTER/Cluster_2.html)  |        0        |   1.3818    |        0        |   -1.3717   | [830](CLUSTER/ORA_Cluster_2.html)  |
| [Cluster_3](CLUSTER/Heatmap_Cluster_3.pdf) | [1000](CLUSTER/Cluster_3.html) |        0        |   1.6673    |        0        |   1.5826    | [2324](CLUSTER/ORA_Cluster_3.html) |
| [Cluster_4](CLUSTER/Heatmap_Cluster_4.pdf) |  [81](CLUSTER/Cluster_4.html)  |        0        |   0.2312    |        0        |   1.5631    | [1290](CLUSTER/ORA_Cluster_4.html) |
| [Cluster_5](CLUSTER/Heatmap_Cluster_5.pdf) | [396](CLUSTER/Cluster_5.html)  |        0        |   -1.4606   |        0        |   1.3153    | [3410](CLUSTER/ORA_Cluster_5.html) |


</div>

<div align='center'>
<img src="FIGURE/clustering_heatmap-1.png" title="plot of chunk clustering_heatmap" alt="plot of chunk clustering_heatmap" width="640px" />
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 8.**  This plot shows below the average expression levels of each cluster. Data was normalized before the analysis, so the mean of the control groups was zero and the standard deviation of all samples of each gene was 1.0. Values indicate number of standard deviation from mean of relative control group.
</div>

<div align='center'>
<img src="FIGURE/clustering_mean-1.png" title="plot of chunk clustering_mean" alt="plot of chunk clustering_mean" width="800px" />
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 9.** This plot summarizes the group means and standard errors of all clusters.
</div>

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div>

***

# Appendix 

Check out the **[RoCA home page](http://zhezhangsh.github.io/RoCA)** for more information.  

## Reproduce this report

To reproduce this report: 

  1. Find the data analysis template you want to use and an example of its pairing YAML file  [here](https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples) and download the YAML example to your working directory

  2. To generate a new report using your own input data and parameter, edit the following items in the YAML file:

    - _output_        : where you want to put the output files
    - _home_          : the URL if you have a home page for your project
    - _analyst_       : your name
    - _description_   : background information about your project, analysis, etc.
    - _input_         : where are your input data, read instruction for preparing them
    - _parameter_     : parameters for this analysis; read instruction about how to prepare input data

  3. Run the code below within ***R Console*** or ***RStudio***, preferablly with a new R session:


```r
if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis
```

If there is no complaint, go to the _output_ folder and open the ***index.html*** file to view report. 

## Session information


```
## R version 3.2.2 (2015-08-14)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.10.5 (Yosemite)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] DEGandMore_0.0.0.9000 snow_0.4-1            rchive_0.0.0.9000    
##  [4] gplots_3.0.1          MASS_7.3-45           htmlwidgets_0.6      
##  [7] DT_0.1                awsomics_0.0.0.9000   yaml_2.1.13          
## [10] rmarkdown_0.9.6       knitr_1.13            RoCA_0.0.0.9000      
## [13] RCurl_1.95-4.8        bitops_1.0-6          devtools_1.12.0      
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.5        magrittr_1.5       highr_0.6         
##  [4] stringr_1.0.0      caTools_1.17.1     tools_3.2.2       
##  [7] parallel_3.2.2     KernSmooth_2.23-15 withr_1.0.2       
## [10] htmltools_0.3.5    gtools_3.5.0       digest_0.6.9      
## [13] formatR_1.4        memoise_1.0.0      evaluate_0.9      
## [16] gdata_2.17.0       stringi_1.1.1      jsonlite_0.9.22
```

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div>

***
**END OF DOCUMENT**
