---
title: "Meta-analysis of differential gene expression"
author: "`r if (exists('yml')) yml$analyst else 'Jim Zhang'`"
date: "`r Sys.Date()`"
output:
  html_document:
    number_sections: yes
    self_contained: no
    toc: yes
    toc_float:
      collapsed: no
---

<div style="border:black 1px solid; padding: 0.5cm 0.5cm">

**Introduction**: This analysis is the summary of a set of pairwise comparisons. It performs meta-analysis on the results of differential gene expression generated by [this](https://raw.githubusercontent.com/zhezhangsh/RoCA/master/template/de/pairwise_comparison/pairwise_comparison.Rmd) template. Both gene and gene set-level results of individual comparisons are summarized through visualization and [Fisher's method](https://en.wikipedia.org/wiki/Fisher%27s_method) of p value meta-analysis. 

</div>

&nbsp;

```{r global_setup, include=FALSE}
name.yaml <- 'meta_analysis.yaml'; 
name.packages <- c('rmarkdown', 'knitr', 'yaml', 'DT', 'htmlwidgets', 'MASS', 'gplots', 'RoCA', 'DEGandMore', 'rchive'); 
name.subfolders <- c('path.input'='input', 'path.r'='R', 'path.fig'='figure', 'path.tbl'='table');

knitr::opts_chunk$set(eval=TRUE, dpi=300, fig.pos="H", fig.width=8, fig.height=6, echo=FALSE, warning=FALSE, message=FALSE, fig.path='FIGURE/');

## Default knitr parameters
knitr::opts_chunk$set(dpi=300, dev=c('png', 'pdf'), echo=FALSE, warning=FALSE, message=FALSE);

if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

## Load required R packages
loaded <- LoadPackage(name.packages);
if (length(loaded[!loaded])) stop('Error: failed to load package(s) ', paste(names(loaded[!loaded]), collapse=', '));

## By default, before knitting this R Markdown file, the YAML file pairing it has been loaded.
## But, the developer can also provide the option to load the YAML file on the fly.
## So, the template can be run using the "Knit HTML" button in RStudio
if (!exists('yml'))                   # if the 'yml' variable doesn't exist yet, create it by loading the YAML file
  if (file.exists(name.yaml))         # assume the pairing YAML file exists in the current folder with the same name
    yml<-yaml.load_file(name.yaml);   # rename the YAML file to fit this template

prms <- yml$parameter;

## Generate directory and sub-directories where the output files will be
f <- GenerateFolder(yml$output, name.subfolders);
path <- yml$output;
for (i in 1:length(name.subfolders)) assign(names(name.subfolders)[i], f[name.subfolders[i]]); 

## URL to project home
## Use this line to add a link to project home in the report: `r home.url`
home.url <- Link2Home(yml$home);
```

```{r load_data, include=FALSE}
cp <- ImportTable(DownloadFile(yml$input$comparison, path.input), rownames = FALSE, colnames = FALSE); 

NM <- as.vector(cp[[1]]); 
N  <- length(NM);

c <- as.vector(cp[, 2]); 
names(c) <- as.vector(cp[, 1]); 
res <- lapply(names(c), function(nm) {
  f <- c[[nm]]; 
  
  if (dir.exists(f)) {
    res <- ImportR(paste(f, 'R', 'result.rds', sep='/')); 
  } else {
    fn <- DownloadFile(f, path.input); 
    res <- ImportR(fn); 
  };
  
  saveRDS(res, paste(path.input, paste('result_', nm, '.rds', sep=''), sep='/')); 

  res; 
});

expr <- lapply(res, function(res) res$inputs$expr);
anno <- lapply(res, function(res) res$inputs$anno);
comp <- lapply(res, function(res) res$inputs$comparison);
de   <- lapply(res, function(res) res$de); 
ora  <- lapply(res, function(res) res$ora); 
gsea <- lapply(res, function(res) res$gsea); 
stat <- lapply(de, function(de) de$results$stat); 
deg  <- lapply(de, function(de) de$DEG); 
names(res) <- names(expr) <- names(anno) <- names(comp) <- names(de) <- NM;
names(ora) <- names(gsea) <- names(stat) <- names(deg) <- NM; 
```

`r home.url` 

# Description
  
`r WriteDescription(yml$description)`

`r home.url` 

## Individual comparisons

This report performs meta-analysis on the results of the following pairwise comparisons.  

```{r comparisons, include=FALSE}
nms <- sapply(comp, names); 
szs <- sapply(comp, function(c) sapply(c, length)); 
gns <- sapply(expr, nrow); 
mtd <- sub('^De', '', sapply(de, function(de) de$method));
prd <- sapply(de, function(de) de$parameter$paired);
nds <- sapply(deg, function(x) sapply(x, nrow)); 

tbl <- data.frame(Name=names(comp), Group1=nms[1, ], Group2=nms[2, ], Num1=szs[1,], Num2=szs[2,], Num_Gene=gns,
                  Test=mtd, Paired=prd, DEG_Higher=nds[1, ], DEG_Lower=nds[2, ], stringsAsFactors = FALSE); 

if (ncol(cp) >=3) tbl[, 1] <- paste('[', tbl[, 1], '](', as.vector(cp[, 3]), ')', sep=''); 
```

<div style="color:darkblue; padding:0 1cm">
**Table 1.** Information about individual comparisons, including comparison name, group names, group size, total number of genes, etc. 
</div>
  
<div align='center', style="padding:0 1cm">
`r kable(tbl, row.names=FALSE, align=rep('c', ncol(tbl)))`
</div>

`r home.url` 

# Gene-level analysis

```{r gene_meta_analysis, include=FALSE}
gid <- lapply(stat, rownames); 
if (prms$gene$union) gid <- Reduce('union', gid) else gid <- Reduce('intersect', gid);  

# Combined annotation
cnm  <- Reduce('intersect', lapply(anno, colnames)); 
anno <- lapply(anno, function(anno) anno[, cnm, drop=FALSE]); 
ids  <- unlist(lapply(anno, rownames), use.names=FALSE);
anno <- Reduce('rbind', anno); 
dup  <- duplicated(ids); 
anno <- anno[!dup, , drop=FALSE];
rownames(anno) <- ids[!dup];
anno <- anno[gid, , drop=FALSE]; 
rownames(anno) <- gid; 

# Combine p values
pvl <- lapply(de, function(de) {
  p <- de$results$stat[, 'Pvalue']; 
  m <- min(p[!is.na(p) & p>0]); 
  p[p==0 & !is.na(p)] <- m;
  names(p) <- rownames(de$results$stat); 
  p; 
}); 
pvl <- sapply(pvl, function(x) x[gid]); 

# Combine log2 fold change
l2r <- lapply(de, function(de) {
  l <- de$results$stat[, 'LogFC']; 
  names(l) <- rownames(de$results$stat); 
  l; 
}); 
l2r <- sapply(l2r, function(x) x[gid]); 

n  <- apply(pvl, 1, function(x) length(x[!is.na(x)])); 
sq <- -2*rowSums(log(pvl), na.rm=TRUE);
p0 <- exp(log(10)*pchisq(sq, df=2*n, lower.tail = FALSE, log.p = TRUE)); 
p1 <- apply(pvl, 1, function(x) max(x, na.rm=TRUE)); 
l0 <- rowMeans(abs(l2r), na.rm=TRUE); 
l1 <- apply(l2r, 1, function(x) min(abs(x), na.rm=TRUE));

# Rank genes
x <- pvl;
y <- l2r;
x[is.na(x)] <- 1; 
y[is.na(y)] <- 0; 
if (prms$gene$rank == 'fc') rnk <- y else 
  if (prms$gene$rank == 'p') rnk <- -sign(y)*log10(x) else
    rnk <- sign(y)*sqrt(abs(sign(y)*log10(x))); 

# Prepare table
meta <- cbind(N_Comp=n, Pval_Max=p1, Pval_Meta=p0, Abs_LogFC_Min=l1, Abs_LogFC_Avg=l0);
temp <- cbind(l2r, pvl)[, as.vector(rbind(1:N, N+(1:N)))]; 
colnames(temp) <- paste(rep(NM, 2), rep(c('Log2FC', 'Pval'), N), sep='_'); 

# Write out table
tbl <- tbl0 <- cbind(anno, meta, temp); 
CreateDatatable(tbl[order(tbl[, 'Pval_Meta']), , drop=FALSE], paste(path.tbl, 'gene_stat_all.html', sep='/')); 
for (i in 1:ncol(anno)) tbl[, i] <- CleanHtmlTags(anno[, i], remove.empty = FALSE); 
saveRDS(tbl, paste(path.r, 'gene_stat_all.rds', sep='/')); 
write.csv(tbl, paste(path.tbl, 'gene_stat_all.csv', sep='/')); 
```

```{r gene_text, included=FALSE}
if (prms$gene$union) l1 <- 'Genes presented in any of the comparisons are included.' else 
  l1 <- 'Only genes presented in all of the comparisons are included.'

x <- tolower(prms$gene$rank)[1];
if (x == 'fc') l2 <- 'The log2 of fold change between 2 compared groups' else if (x == 'p')
  l2 <- 'The log10 of p value signed by direction of change' else 
    l2 <- 'The joint of p value and fold change, _sign(log2FC)*sqrt(abs(log10(p)*logFC))_,';

l2 <- paste(l2, 'is used as a representative statistic of the differential expression of each gene in each comparisons.'); 
```

Gene-level statistics, log2 of fold change and p value, are retrieved from all individual comparisons. `r l1` [Fisher's method](https://en.wikipedia.org/wiki/Fisher%27s_method) of meta-analysis is performed on the p values of individual comparisons to get a combined p value for each gene. Other summary statistics of each gene include its maximum p value from all comparisons, and the average/minimum of _|log2 of fold change|_ from all comparisons. `r l2`

<div align='center'>
```{r gene_cluster, include=TRUE, fig.width=min(6, 0.5*N+2), fig.height=min(6, 0.5*N+2), out.width='640px'}
corr <- cor(rnk, use='pair'); 
par(omi=c(1, 0, 0, 1)); 
wid <- 0.8*max(strwidth(NM, unit='inch'));
for (i in 1:N) corr[i, i] <- NA;
heatmap(1-corr, cexRow=1/wid, cexCol=1/wid, scale='none'); 
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 1.** Clustering of all the pairwise comparisons. `r l2` The representative statistics of all genes are then used for the clustering.  
</div>

`r home.url` 

## Top genes

Selection of the top genes with the most prominent meta-analysis results. 

<div align='center'>
```{r gene_top_mean, include=TRUE, fig.width=max(4, 0.5*(N+1)), fig.height=min(8, 0.1*prms$gene$top+1), out.width=paste(100*min(6, 0.5*N+2), 'px', sep='')}
x <- rev(sort(rowMeans(abs(rnk))))[1:min(length(p0), prms$gene$top)]; 
x <- rnk[names(x), , drop=FALSE]; 
x <- x[hclust(dist(x))$order, , drop=FALSE]; 
y <- round(max(abs(x), na.rm=TRUE), 1); 
PlotColoredBlock(x, groups=NM, min=-y, max=y, key='Differential expression'); 

tbl <- tbl0[rownames(x), ]; 
CreateDatatable(tbl, paste(path.tbl, 'gene_top_mean.html', sep='/')) -> x; 
for (i in 1:ncol(anno)) tbl[, i] <- CleanHtmlTags(tbl[, i], remove.empty = FALSE); 
write.csv(tbl, paste(path.tbl, 'gene_top_mean.csv', sep='/')); 
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 2A.** The [top `r nrow(x)` genes](table/gene_top_mean.html) with the most overall differential expression across all comparisons, based on their representative statistics. 
</div>

&nbsp;

<div align='center'>
```{r gene_top_diff, include=TRUE, fig.width=max(4, 0.5*(N+1)), fig.height=min(8, 0.1*prms$gene$top+1), out.width=paste(100*min(6, 0.5*N+2), 'px', sep='')}
x <- apply(rnk, 1, sd)
x <- rev(sort(x))[1:min(length(p0), prms$gene$top)]; 
x <- rnk[names(x), , drop=FALSE]; 
x <- x[hclust(dist(x))$order, , drop=FALSE]; 
y <- round(max(abs(x), na.rm=TRUE), 1); 
PlotColoredBlock(x, groups=NM, min=-y, max=y, key='Differential expression');  

tbl <- tbl0[rownames(x), ]; 
CreateDatatable(tbl, paste(path.tbl, 'gene_top_diff.html', sep='/')) -> x; 
for (i in 1:ncol(anno)) tbl[, i] <- CleanHtmlTags(tbl[, i], remove.empty = FALSE); 
write.csv(tbl, paste(path.tbl, 'gene_top_diff.csv', sep='/')); 
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 2B.** The [top `r nrow(x)` genes](table/gene_top_diff.html) with the most different results of differential expression between the comparisons, based on their representative statistics. 
</div>

`r home.url` 

# Gene set-level analysis

```{r geneset_meta_analysis, include=FALSE}
ss <- lapply(gsea, function(x) {
  s <- x$stat;
  rownames(s) <- paste(s[, 1], s[, 2], sep=':::');
  s; 
}); 
gid <- lapply(ss, rownames); 
if (prms$geneset$union) gid <- Reduce('union', gid) else gid <- Reduce('intersect', gid);  

# Gene set size
ngs <- unlist(lapply(ss, function(ss) ss[, 3]), use.names=FALSE); 
names(ngs) <- unlist(lapply(ss, rownames), use.names=FALSE); 
ngs <- ngs[gid]; 
nms <- strsplit(names(ngs), ':::'); 
cll <- sapply(nms, function(x) x[1]); 
nms <- sapply(nms, function(x) x[2]); 

# Combine p values
pvl <- lapply(ss, function(ss) {
  p <- ss[, 'PValue']; 
  m <- floor(log10(min(p[!is.na(p) & p>0]))); 
  p[p==0 & !is.na(p)] <- 10^m;
  names(p) <- rownames(ss); 
  p; 
}); 
pvl <- sapply(pvl, function(x) x[gid]); 

# Combine NES
nes <- lapply(ss, function(ss) {
  s <- ss[, 'NES']; 
  names(s) <- rownames(ss); 
  s; 
}); 
nes <- sapply(nes, function(x) x[gid]); 

n  <- apply(pvl, 1, function(x) length(x[!is.na(x)])); 
sq <- -2*rowSums(log(pvl), na.rm=TRUE);
p0 <- exp(log(10)*pchisq(sq, df=2*n, lower.tail = FALSE, log.p = TRUE)); 
p1 <- apply(pvl, 1, function(x) max(x, na.rm=TRUE)); 
s0 <- rowMeans(abs(nes), na.rm=TRUE); 
s1 <- apply(nes, 1, function(x) min(abs(x), na.rm=TRUE));

# Prepare table
meta <- cbind(N_Comp=n, Pval_Max=p1, Pval_Meta=p0, Abs_NES_Min=s1, Abs_NES_Avg=s0);
temp <- cbind(nes, pvl)[, as.vector(rbind(1:N, N+(1:N)))]; 
colnames(temp) <- paste(rep(NM, 2), rep(c('NES', 'Pval'), N), sep='_'); 

# Write out table
tbl <- tbl0 <- data.frame(Collection=as.vector(cll), Gene_Set=as.vector(nms), Size=as.vector(ngs), meta, temp); 
rownames(tbl) <- rownames(tbl0) <- 1:nrow(tbl); 
CreateDatatable(tbl[order(tbl[, 'Pval_Meta']), , drop=FALSE], paste(path.tbl, 'gsea_stat_all.html', sep='/')); 
saveRDS(tbl, paste(path.r, 'gsea_stat_all.rds', sep='/')); 
write.csv(tbl, paste(path.tbl, 'gsea_stat_all.csv', sep='/')); 
```

```{r geneset_text, included=FALSE}
if (prms$geneset$union) l1 <- 'Gene sets presented in any of the comparisons are included.' else 
  l1 <- 'Only gene sets presented in all of the comparisons are included.'

x <- prms$geneset$heatmap$sort$comparison[1]
if (x>=1 & x<=N) l2 <- paste('with the highest _|NES|_ from comparison ', NM[x], '.', sep='') else 
  l2 <- 'with the highest average _|NES|_ of all comparisons.'
```

Statisitcs of gene set enrichment analysis, p value and normalized enriched score (NES), are retrieved from individual comparisons. `r l1` [Fisher's method](https://en.wikipedia.org/wiki/Fisher%27s_method) of meta-analysis is performed on the GSEA p values of individual comparisons to get a combined p value for each gene set. For the meta-analysis, p values of 0 from the original GSEA are rounded to `r min(pvl)`. Other summary statistics of each gene set include its maximum p value from all comparisons, and the average/minimum of _|NES|_ from all comparisons.

<div align='center'>
```{r gsea_cluster, include=TRUE, fig.width=min(6, 0.5*N+2), fig.height=min(6, 0.5*N+2), out.width='640px'}
corr <- cor(nes, use='pair'); 
par(omi=c(1, 0, 0, 1)); 
wid <- 0.8*max(strwidth(NM, unit='inch'));
for (i in 1:N) corr[i, i] <- NA;
heatmap(1-corr, cexRow=1/wid, cexCol=1/wid, scale='none'); 
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 3.** Clustering of all the pairwise comparisons, using NES of all gene sets.
</div>

`r home.url` 

## Top gene sets

Selection of the top gene sets with the most significant meta-analysis results. 

```{r sort_nes, include=FALSE}
sortNES <- function(x, top, ind, ord) {
  if (ind>=1 & ind <=ncol(x)) {
    x <- x[rev(order(abs(x[, ind]))), , drop=FALSE];
    x <- x[1:min(nrow(x), top), , drop=FALSE]; 
    o <- order(x[, ind]); 
  } else {
    x <- x[rev(order(rowMeans(abs(x), na.rm=TRUE))), , drop=FALSE]; 
    x <- x[1:min(nrow(x), top), , drop=FALSE]; 
    o <- order(rowMeans(x, na.rm=TRUE));
  }
  
  if (ord == -1) x <- x[o, , drop=FALSE] else if (ord == 1) x <- x[rev(o), , drop=FALSE] else
    x <- x[hclust(dist(x))$order, , drop=FALSE]; 

  x; 
}; 
```

<div align='center'>
```{r gsea_top_nes, include=TRUE, fig.width=max(4, 0.5*(N+4)), fig.height=min(8, 0.1*prms$gene$top+1), out.width='600px'}
rownames(nes) <- rownames(pvl) <- nms;
x <- y <- sortNES(nes, prms$geneset$heatmap$top$overall, prms$geneset$heatmap$sort$comparison, prms$geneset$heatmap$sort$order); 
rownames(y) <- substr(rownames(y), 1, 200);
PlotColoredBlock(y, groups=NM, min=-3, max=3, key='Enrichment score'); 

tbl <- tbl0[tbl0[, 2] %in% rownames(x), , drop=FALSE]; 
CreateDatatable(tbl, paste(path.tbl, 'gsea_top_nes.html', sep='/'), rownames = FALSE) -> x; 
write.csv(tbl, paste(path.tbl, 'gsea_top_nes.csv', sep='/'), row.names = FALSE); 
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 4A.** The [top `r nrow(x)` gene sets](table/gsea_top_nes.html) `r l2`
</div>

&nbsp;

<div align='center'>
```{r gsea_top_diff, include=TRUE, fig.width=max(4, 0.5*(N+4)), fig.height=min(8, 0.1*prms$gene$top+1), out.width='600px'}
x <- nes[rev(order(apply(nes, 1, sd))), , drop=FALSE]; 
x <- x[1:min(length(p0), prms$geneset$heatmap$top$overall), , drop=FALSE]; 
x <- y <- sortNES(x, prms$geneset$heatmap$top$overall, prms$geneset$heatmap$sort$comparison, prms$geneset$heatmap$sort$order); 
rownames(y) <- substr(rownames(y), 1, 200);
PlotColoredBlock(y, groups=NM, min=-3, max=3, key='Enrichment score'); 

tbl <- tbl0[tbl0[, 2] %in% rownames(x), , drop=FALSE]; 
CreateDatatable(tbl, paste(path.tbl, 'gsea_top_diff.html', sep='/'), rownames = FALSE) -> x; 
write.csv(tbl, paste(path.tbl, 'gsea_top_diff.csv', sep='/'), row.names = FALSE); 
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 4B.** The [top `r nrow(x)` gene sets](table/gsea_top_diff.html) with the most different NES values among all comparisons.
</div>

`r home.url` 

## Gene set collections

```{r gene_set_collection, include=FALSE}
c <- as.vector(tbl0[, 1]);
a <- sort(unique(c));
coll <- lapply(a, function(nm) {
  t <- tbl0[c==nm, -1, drop=FALSE]; 
  s <- nes[c==nm, , drop=FALSE]; 
  
  fn1 <- paste('collection_', nm, '.html', sep=''); 
  CreateDatatable(t, paste(path.tbl, fn1, sep='/'), rownames = FALSE); 

  x <- sortNES(s, prms$geneset$heatmap$top$collection, prms$geneset$heatmap$sort$comparison, prms$geneset$heatmap$sort$order); 
  rownames(x) <- substr(rownames(x), 1, 200); 
  sz <- CalculateColoredBlockSize(x); 
  fn2 <- paste('collection_', nm, '.png', sep=''); 
  png(paste(path.fig, fn2, sep='/'), w=sz[1], h=sz[2], units='in', res=150); 
  PlotColoredBlock(x, min=-3, max=3, num.breaks = 63, key='Enrichment score', groups=colnames(nes)); 
  dev.off(); 
  pdf(paste(path.fig, sub('.png$', '.pdf', fn2), sep='/'), w=sz[1], h=sz[2]); 
  PlotColoredBlock(x, min=-3, max=3, num.breaks = 63, key='Enrichment score', groups=colnames(nes)); 
  dev.off(); 
  
  list(nrow(t), fn1, fn2); 
});

x <- sapply(coll, function(x) x[[1]]); 
y <- sapply(coll, function(x) x[[2]]); 
z <- sapply(coll, function(x) x[[3]]); 
l <- paste('  - [', a, '](figure/', z, ') :   [', x, ' gene sets](table/', y, ')', sep='');

lns <- paste(l, collapse='\n'); 
```

Gene sets are split into different collections. Click on the collection names to view a plot of NES values of top gene sets and the numbers to see a full table of all gene sets:

`r lns`

`r home.url` 

***

# Appendix 

Check out the **[RoCA home page](http://zhezhangsh.github.io/RoCA)** for more information.  

## Reproduce this report

To reproduce this report: 

  1. Find the data analysis template you want to use and an example of its pairing YAML file  [here](https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples) and download the YAML example to your working directory

  2. To generate a new report using your own input data and parameter, edit the following items in the YAML file:

    - _output_        : where you want to put the output files
    - _home_          : the URL if you have a home page for your project
    - _analyst_       : your name
    - _description_   : background information about your project, analysis, etc.
    - _input_         : where are your input data, read instruction for preparing them
    - _parameter_     : parameters for this analysis; read instruction about how to prepare input data

  3. Run the code below within ***R Console*** or ***RStudio***, preferablly with a new R session:

```{r reproduce_report, eval=FALSE, echo=TRUE}
if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis
```

If there is no complaint, go to the _output_ folder and open the ***index.html*** file to view report. 

## Session information

```{r session_info, echo=FALSE}
sessionInfo(); 
```

`r home.url`

***
**END OF DOCUMENT**
