---
title: "Analysis of RNA-seq samples using read count matrix"
author: "`r if (exists('yml')) yml$analyst else 'Jim Zhang'`"
date: "`r Sys.Date()`"
output:
  html_document:
    number_sections: yes
    self_contained: no
    toc: yes
    toc_float:
      collapsed: no
---

<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
This procedure analyzes the read counts of multiple RNA-seq samples for quality control purposes. It ususally takes an integer matrix of gene-level read counts as input. It is a simplified version of the [rnaseq_sample](https://raw.githubusercontent.com/zhezhangsh/RoCA/master/template/qc/rnaseq_sample/rnaseq_sample.Rmd) procedure, which also performs sample analysis based on the read count matrix. Instead, this procedure focuses only on the contents of the read count matrix itself.
</div>

&nbsp;

```{r global_setup, eval=TRUE, include=FALSE}
name.yaml <- 'rnaseq_count.yaml'; 
name.packages <- c('rmarkdown', 'knitr', 'kableExtra', 'yaml', 'DT', 'htmlwidgets', 'vioplot', 
                   'GenomicRanges', 'RoCA', 'Rnaseq', 'awsomics', 'DEGandMore'); 
name.subfolders <- c('path.input'='input', 'path.r'='R', 'path.fig'='figure', 'path.tbl'='table'); # <full.path.to.subfolder>=<subfolder.name>

## Default knitr parameters
knitr::opts_chunk$set(dpi=300, dev=c('png', 'pdf'), echo=FALSE, warning=FALSE, message=FALSE, fig.path = 'figure/');
if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

## Load required R packages
loaded <- LoadPackage(name.packages);
if (length(loaded[!loaded])) stop('Error: failed to load package(s) ', paste(names(loaded[!loaded]), collapse=', '));

if (!exists('yml'))                   # if the 'yml' variable doesn't exist yet, create it by loading the YAML file
  if (file.exists(name.yaml))         # assume the pairing YAML file exists in the current folder with the same name
    yml<-yaml.load_file(name.yaml);   # rename the YAML file to fit this template

prms <- yml$parameter;

## Generate directory and sub-directories where the output files will be
f <- GenerateFolder(yml$output, name.subfolders);
path <- yml$output;
for (i in 1:length(name.subfolders)) assign(names(name.subfolders)[i], f[name.subfolders[i]]); 

## Automatical figure/table ordering
OrderFigure(reset=TRUE);
OrderTable(reset=TRUE);

## URL to project home
## Use this line to add a link to project home in the report: `r home.url`
home.url <- Link2Home(yml$home);
```

`r home.url` 

# Description {.tabset}

`r WriteDescription(yml$description)`

```{r load_data, include=FALSE}
cnt <- as.matrix(ImportTable(DownloadFile(yml$input$count, path.input)));
# anno <- ImportTable(DownloadFile(yml$input$annotation, path.input));
# smpl <- ImportTable(DownloadFile(yml$input$sample, path.input));

if (nrow(cnt) < 3) stop("Error: cannot proceed with this analysis with less than 3 genes\n"); 
if (ncol(cnt) < 3) stop("Error: cannot proceed with this analysis with less than 3 samples\n"); 
```

`r home.url`

# Results

- Number of samples: ***`r ncol(cnt)`***
- Number of genes: ***`r format(nrow(cnt), big.mark=',')`***
- Average number of total reads per gene: ***`r format(mean(rowSums(cnt)), big.mark=',')`***
- Average number of total reads per sample: ***`r format(mean(colSums(cnt)), big.mark=',')`***

```{r sample, include=FALSE}
ttl <- colSums(cnt)/10^6;
p.shapiro <- shapiro.test(ttl)$p.value; 

nsd <- sapply(1:length(ttl), function(i) (ttl[i]-mean(ttl[-i]))/sd(ttl)); 
hi <- names(ttl)[nsd >= 3];
if (length(hi) == 0) hi <- 'none' else hi <- paste(hi, collapse='; '); 
lo <- names(ttl)[nsd <=-3];
if (length(lo) == 0) lo <- 'none' else lo <- paste(lo, collapse='; '); 
```

The total read counts of all genes are compared between RNA-seq samples. _Shapiro-Wilk_ normality test shows that the total read counts of all samples **`r if (p.shapiro>0.05) 'is' else 'is not'` normally distributed** (p = `r format(p.shapiro, digit=2)`) in this data set. 

  - Outlier samples with low total read count: ***`r lo`***
  - Outlier samples with high total read count: ***`r hi`***

```{r, sample_hist, include=TRUE, fig.width=6, fig.height=3.6, out.width='600px'}
par(mar=c(5,5,2,2));
hist(ttl, xlab='Total read count of all genes (millions)', ylab='Number of samples', 
     main='', cex.lab=1.5, cex.main=1.5, col='#F7DC6F', yaxs='i');
```

`r OrderFigure()` Distribution of total read counts of all samples.

`r home.url`

## Gene read counts

As genes have different length and expression level, reads are unevenly distributed across genes. Usually, a small portion of the genes contribute most of the total reads, while a large portion of the genes have no or few reads mapped to them

```{r, gene_cumulative_total, include=TRUE, fig.width=7.2, fig.height=5.4, out.width='720px'}
gene.ct  <- sort(rowSums(cnt), decreasing = TRUE); 
gene.pct <- 100*gene.ct/sum(gene.ct);
cum.pct  <- cumsum(gene.pct); # Cumulative percent of total reads

par(mar=c(5, 5, 2, 2)); 
plot(1:nrow(cnt), cum.pct, type='l', col='white', log='x', xlab='Number of genes', ylab='Cumulative percent of total reads (%)', 
     cex.lab=1.75, xaxs='i', yaxs='i', ylim=c(0, 100), yaxt='n');
axis(2, at=seq(0, 100, 25));
abline(h=seq(25, 75, 25), lty=2, col='lightgrey');
polygon(c(1, 1:nrow(cnt), nrow(cnt)), c(0, cum.pct, 0), border=NA, col='#E6B0AA');
# lines(1:nrow(cnt), cum.pct, lwd=2, col='#BB8FCE');
pct <- c(5, 10, 25, 50, 75, 90, 95);
ind <- as.vector(sapply(pct, function(pct) which(cum.pct>pct)[1]));
text(ind, cum.pct[ind], label=ind, col='#2E86C1', pos=4, cex=1.25);
points(ind, cum.pct[ind], pch='*', cex=1.5);
legend('topleft', bty='n', pch='*', legend=paste(pct, '% total reads: ', ind, ' gene(s)', sep=''), cex=2/3);
box();
```

`r OrderFigure()` Genes are sorted by their total read counts of all samples, along the X-axis. The cumulative percentages of total reads contributed by the top genes are plotted along the Y-axis. 

&nbsp;

```{r, gene_cumulative_single, include=TRUE, fig.width=7.2, fig.height=5.4, out.width='720px'}
gene.ct  <- sort(rowSums(cnt), decreasing = TRUE); 
gene.pct <- 100*gene.ct/sum(gene.ct);
cum.pct  <- cumsum(gene.pct); # Cumulative percent of total reads

pct <- apply(cnt, 2, function(c) cumsum(100*rev(sort(c))/sum(c)));
col <- rainbow(ncol(cnt));
var <- apply(pct, 1, sd);
ind <- which(var==max(var));

par(mar=c(5, 5, 2, 2)); 
plot(1, 1, type='n', col='white', log='x', xlab='Number of genes', ylab='Cumulative percent of total reads (%)', 
     cex.lab=1.75, xaxs='i', yaxs='i', xlim=c(1, nrow(cnt)), ylim=c(0, 100), yaxt='n');
axis(2, at=seq(0, 100, 25));
abline(h=seq(25, 75, 25), lty=2, col='lightgrey');
for (i in 1:ncol(cnt)) lines(c(1, 1:nrow(cnt)), c(0, pct[, i]), col=col[i], lwd=2/3); 
text(ind, pct[ind, ], label=colnames(cnt), col='black', cex=0.4);
box();
```

`r OrderFigure()` Same plot as the last one, but showing individual samples and their cumulative percent of read counts contributed by sorted genes.

&nbsp;

```{r gene_top, include=TRUE, fig.width=7.2, fig.height=5.4, out.width='720px'}
top10 <- apply(cnt, 2, function(x) {
x <- rev(sort(x)); 
y <- cumsum(x[1:min(length(x), 10)]); 
y/sum(x)*100; 
}); 
top10 <- top10[, order(top10[nrow(top10), , drop=FALSE])]; 
CreateDatatable(top10, paste(path.tbl, 'top_gene_count.html', sep='/')) -> x; 

ids10 <- apply(cnt, 2, function(x) rownames(cnt)[rev(order(x))][1:10]);
CreateDatatable(ids10, paste(path.tbl, 'top_gene_id.html', sep='/')) -> x; 

par(mar=c(8, 5, 3, 2)); 
plot(0, type='n', ylim=c(min(top10), max(top10)), xlim=c(1, ncol(top10)), xaxs='i', xaxt='n', 
     xlab='', ylab='Cumulative percent (%)', cex.lab=2, 
main=paste('Reads contributed by the top', nrow(top10), 'genes'), cex.main=1.5);
abline(v=1:ncol(top10), lwd=0.5, col='lightgrey', lty=2);
for (i in nrow(top10):1) {
  polygon(c(1, 1:ncol(top10), ncol(top10)), c(0, top10[i, ], 0), border=NA, col=terrain.colors(nrow(top10))[i]);
  lines(top10[i, ], lwd=0.5, lty=1, pch=20, type='b', col='lightgrey', cex=0.5);
}
axis(1, at=c(1:ncol(top10)), tick=TRUE, label=colnames(top10)[c(1:ncol(top10))], las=3, 
     cex.axis=min(min(1, 32/ncol(cnt)), 12/max(nchar(colnames(cnt))))); 
legend('topleft', bty='n', lty=2, pch=20, col=rev(terrain.colors(nrow(top10))), cex=2/3,
       legend=paste('Top ', nrow(top10):1, ', ', rev(round(rowMeans(top10), 2)), '%', sep='')); 

nsd <- sapply(1:ncol(top10), function(i)
(top10[nrow(top10), i]-mean(top10[nrow(top10), -i]))/sd(top10[nrow(top10), -i]))
hi <- colnames(top10)[nsd >= 3];
if (length(hi) == 0) hi <- 'none' else hi <- paste(hi, collapse='; ');
lo <- colnames(top10)[nsd <=-3];
if (length(lo) == 0) lo <- 'none' else lo <- paste(lo, collapse='; ');
```

`r OrderFigure()` The percent of total read counts per sample contributed by its [top `r nrow(top10)` genes](table/top_gene_id.html). Samples are sorted by the percents from low to high along the X-axis. The cumulative percents of all samples are then compared to each other to identify samples with extremely lower or higher percents than the other samples: 

- Outlier samples with low percent of reads contributed by the top `r nrow(top10)` genes: ***`r lo`***
- Outlier samples with high percent of reads contributed by the top `r nrow(top10)` genes: ***`r hi`***

`r home.url`

## Between-sample dispersion

Gene-specific dispersion of RNA-seq read counts is commonly used to evaluate between-sample variance of a data set. Here, the dispersion is estimated by the overall pattern of coefficient of variation (standard deviation devided by average read count) of all genes. 

```{r dispersion, include=TRUE, fig.width=7.2, fig.height=4.8, out.width='720px'}
par(mar=c(5,5,2,2)); 

logged <- log10(cnt);
logged[logged==-Inf] <- 0;
mn <- rowMeans(logged);
cv <- apply(logged, 1, sd)/mn;

plot(mn, cv, pch=19, cex=.5, xaxt='n', cex.lab=1.5, col='#88888888', xaxs='i', yaxs='i',
     xlab='Average read count of all samples', ylab='Coefficient of variation'); 
axis(1, at=0:ceiling(max(mn)), label=10^(0:ceiling(max(mn)))); 
# lines(lowess(cv~mn), col=4);
# abline(v = dens$x[dens.y==max(dens.y)], lty=3, col=3); 
```

`r OrderFigure()` The dispersion of gene expression level between samples, measured as the coefficient of variation (CV). Genes with lower read counts have generally higher CV.  

`r home.url`

## Sample-sample correlation

Similarity between samples can be evaluated by their correlation of read counts. 

```{r correlation, include=TRUE, fig.width=6, fig.height=6, out.width='600px'}
crr <- cor(log2(cnt+1));
for (i in 1:ncol(cnt)) crr[i, i] <- NA;

srt <- lapply(1:nrow(crr), function(i) sort(crr[i, ]));
names(srt) <- rownames(crr);

mx <- sapply(srt, function(x) rev(x)[1]);
mn <- sapply(srt, function(x) x[1]);

p1 <- names(srt)[which(mx==max(mx))];
p2 <- names(srt)[which(mn==min(mn))];

PlotColoredBlock(crr, groups=colnames(cnt), key='Correlation coefficient');
```

`r OrderFigure()` Correlation between all pairs of RNA-seq samples.

```{r scatter, include=TRUE, fig.width=12, fig.height=6, out.width='800px'}
par(mfrow=c(1, 2));

PlotLogScatter(cnt[, p1[1]], cnt[, p1[2]], xlab=p1[1], ylab=p1[2], col="#E74C3C",
               main=paste('Highest Correlation =', round(crr[p1[1], p1[2]], 4)));
abline(0, 1);
PlotLogScatter(cnt[, p2[1]], cnt[, p2[2]], xlab=p1[1], ylab=p1[2], col="#1ABC9C",
               main=paste('Lowest Correlation =', round(crr[p2[1], p2[2]], 4)));
abline(0, 1);
```

`r OrderFigure()` The two pairs of samples with the best and worst correlation of read counts.



## Summary

```{r summary, include=FALSE}

nnz <- apply(cnt, 2, function(c) length(c[c>0])); 

smm <- data.frame(Total=colSums(cnt), Mean=round(colMeans(cnt), 2), Median=apply(cnt, 2, median),
                  ZeroRead=apply(cnt, 2, function(c) length(c[c==0])), 
                  Top10_Pct=round(top10[10, ], 1), Top_Gene=ids10[1, ], stringsAsFactors = FALSE);

dir.create(paste(path.fig, 'scatter', sep='/'), showWarnings = FALSE);
ngh <- sapply(rownames(smm), function(id) {
  mx <- names(rev(srt[[id]]))[1];
  f1 <- paste0('scatter/', id, '_', mx, '.png');
  png(paste(path.fig, f1, sep='/'), width = 800, height = 800);
  PlotLogScatter(cnt[, id], cnt[, mx], xlab=id, ylab=mx, col="#1ABC9C",
               main=paste('Correlation =', round(crr[id, mx], 4)));
  abline(0, 1);
  try(dev.off());
  
  mn <- names(srt[[id]])[1];
  f2 <- paste0('scatter/', id, '_', mn, '.png');
  png(paste(path.fig, f2, sep='/'), width = 800, height = 800);
  PlotLogScatter(cnt[, id], cnt[, mn], xlab=id, ylab=mx, col="#E74C3C",
               main=paste('Correlation =', round(crr[id, mn], 4)));
  abline(0, 1);
  try(dev.off());
  
  cbind(paste0('[', mx, '](figure/', f1, ')'), paste0('[', mx, '](figure/', f2, ')'));
});
smm$Nearest  <- ngh[1, ];
smm$Farthest <- ngh[2, ];

write.csv(cnt, paste(path.tbl, 'count.csv', sep='/'));
write.csv(smm, paste(path.tbl, 'summary.csv', sep='/'));
write.csv(crr, paste(path.tbl, 'correlation.csv', sep='/'));
```

`r OrderTable()` Sample summary.

`r kable(smm) %>% kable_styling(
    bootstrap_options = c("striped", "hover", "condensed", "responsive"), full_width=FALSE)`

# Download

> [Sample summary](table/summary.csv)

> [Read count matrix](table/count.csv)

> [Sample-sample correlation](table/correlation.csv)

`r home.url`

# Appendix 

Check out the **[RoCA home page](http://zhezhangsh.github.io/RoCA)** for more information.  

## Reproduce this report

To reproduce this report: 

1. Find the data analysis template you want to use and an example of its pairing YAML file  [here](https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples) and download the YAML example to your working directory

2. To generate a new report using your own input data and parameter, edit the following items in the YAML file:

- _output_        : where you want to put the output files
- _home_          : the URL if you have a home page for your project
- _analyst_       : your name
- _description_   : background information about your project, analysis, etc.
- _input_         : where are your input data, read instruction for preparing them
- _parameter_     : parameters for this analysis; read instruction about how to prepare input data

3. Run the code below within ***R Console*** or ***RStudio***, preferablly with a new R session:

```{r reproduce_report, eval=FALSE, echo=TRUE}
if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis
```

If there is no complaint, go to the _output_ folder and open the ***index.html*** file to view report. 

## Session information

```{r session_info, echo=FALSE}
sessionInfo(); 
```

`r home.url`

***
_END OF DOCUMENT_
