---
title: "Summarize three pairwise comparisons of differential expression"
author: "`r if (exists('yml')) yml$analyst else 'Jim Zhang'`"
date: "`r Sys.Date()`"
output:
  html_document:
    number_sections: yes
    self_contained: no
    toc: yes
    toc_float:
      collapsed: no
---

<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
**Introduction**: This analysis is based on the outputs of pairwise comparisons of differential gene expression generated by [this](https://raw.githubusercontent.com/zhezhangsh/RoCA/master/template/de/pairwise_comparison/pairwise_comparison.Rmd) template. It uses results from 3 pairwise comparisons of 3 sample groups vs. their corresponding control groups and compares how these 3 sample groups are different from each other in terms of their sample-control differences (_delta-delta_). An example of such analysis is the different responses of 3 cell types to the treatment of the same drug. This analysis is focused on the overlapping of differentially expression at both gene and gene set levels.

</div>

&nbsp;

```{r global_setup, include=FALSE}
name.yaml <- 'summarize_two.yaml'; 
name.packages <- c('rmarkdown', 'knitr', 'yaml', 'DT', 'htmlwidgets', 'MASS', 'gplots', 'RoCA', 'awsomics', 'rchive', 'DEGandMore'); 
name.subfolders <- c('path.input'='input', 'path.r'='R', 'path.fig'='figure', 'path.tbl'='table');

knitr::opts_chunk$set(eval=TRUE, dpi=300, fig.pos="H", fig.width=8, fig.height=6, echo=FALSE, warning=FALSE, message=FALSE, fig.path='FIGURE/');

## Default knitr parameters
knitr::opts_chunk$set(dpi=300, dev=c('png', 'pdf'), echo=FALSE, warning=FALSE, message=FALSE);

if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

## Load required R packages
loaded <- LoadPackage(name.packages);
if (length(loaded[!loaded])) stop('Error: failed to load package(s) ', paste(names(loaded[!loaded]), collapse=', '));

## By default, before knitting this R Markdown file, the YAML file pairing it has been loaded.
## But, the developer can also provide the option to load the YAML file on the fly.
## So, the template can be run using the "Knit HTML" button in RStudio
if (!exists('yml'))                   # if the 'yml' variable doesn't exist yet, create it by loading the YAML file
  if (file.exists(name.yaml))         # assume the pairing YAML file exists in the current folder with the same name
    yml<-yaml.load_file(name.yaml);   # rename the YAML file to fit this template

prms <- yml$parameter;

## Generate directory and sub-directories where the output files will be
f <- GenerateFolder(yml$output, name.subfolders);
path <- yml$output;
for (i in 1:length(name.subfolders)) assign(names(name.subfolders)[i], f[name.subfolders[i]]); 

## URL to project home
## Use this line to add a link to project home in the report: `r home.url`
home.url <- Link2Home(yml$home);
```

```{r load_data, include=FALSE}
# Loading results from pairwsie DEG analysis
yml$input$comparison <- lapply(yml$input$comparison, function(c) {
  if (is.list(c)) c else {
    if (is.character(c) & dir.exists(c[1])) 
      list(result=paste(c, 'R/result.rds', sep='/'), index=paste(c, 'index.html', sep='/')) else
        stop('Error: comparison results not found\n'); 
  }
})

res <- lapply(yml$input$comparison, function(f) ImportR(DownloadFile(f$result, path.input))); 
gset <- ImportR(DownloadFile(yml$input$geneset, path.input)); 
smpl <- ImportR(DownloadFile(yml$input$sample, path.input));
comp <- names(yml$input$comparison); 

grp <- lapply(res, function(res) names(res$inputs$comparison));
smp <- lapply(res, function(x) x$inputs$comparison);
names(res) <- names(grp) <- names(smp) <- names(yml$input$comparison);

grps<-do.call('c', lapply(res, function(res) res$inputs$comparison)); 
smpl <- smpl[unlist(smp, use.names=FALSE), ]; 

s0 <- setdiff(unlist(smp), rownames(smpl));
if (length(s0) > 0) stop('Error: missing metadata of sample(s): ', paste(s0, collapse='; '), '\n');

de<-lapply(res, function(res) res$de); 
ora<-lapply(res, function(res) res$ora); 
gsea<-lapply(res, function(res) res$gsea); 

# Combined matrix
expr.all <- lapply(res, function(res) {
  input<-res$inputs;
  s<-c(input$comparison[[1]], input$comparison[[2]]); 
  sapply(s, function(s) input$expr[, s]); 
}); 
g<-Reduce('intersect', lapply(expr.all, rownames)); 
expr.all<-do.call('cbind', lapply(expr.all, function(d) d[g, ])); 
```

`r home.url` 

# Description
  
`r WriteDescription(yml$description)`

## Pairwise comparisons

This report compares the results of the  following pairwise comparisons.  

```{r link_lines, include=FALSE}
lns <- sapply(names(yml$input$comparison), function(nm) {
  c <- yml$input$comparison[[nm]];
  ind <- c$index[1];
  l <- paste(grp[[nm]], collapse=' vs. ', sep='');
  if (is.null(ind) | is.na(ind) | ind=='') l else paste('[', l, '](', yml$input$comparison[[1]]$index, ')', sep=''); 
});
lns <- as.vector(lns); 
```
  - **`r comp[1]`**: `r lns[1]`
  - **`r comp[2]`**: `r lns[2]`

`r home.url` 

# Gene-level comparison

```{r gene_level, include=TRUE}
de<-lapply(res, function(res) res$de); 
stat<-lapply(de, function(de) de$result$stat); 
deg<-lapply(de, function(de) de$DEG); 
anno<-lapply(res, function(res) res$input$anno); 
gid<-lapply(stat, rownames); 
if (setequal(rownames(stat[[1]]), rownames(stat[[2]]))) gid<-rownames(stat[[1]]) else {
  if (prms$gene$union) gid<-Reduce('union', lapply(stat, rownames)) else gid<-Reduce('intersect', lapply(stat, rownames)) 
}
anno[[2]]<-anno[[2]][!(rownames(anno[[2]]) %in% rownames(anno[[1]])), , drop=FALSE];
anno<-rbind(anno[[1]], anno[[2]])[gid, , drop=FALSE]; 
names(stat)<-names(de)<-comp; 

l2r <- sapply(stat, function(stat) as.matrix(stat)[, 'LogFC'][gid]); 
pvl <- sapply(stat, function(stat) as.matrix(stat)[, 'Pvalue'][gid]); 
fdr <- sapply(stat, function(stat) as.matrix(stat)[, 'FDR'][gid]); 

l2r[is.na(l2r)] <- 0;
pvl[is.na(pvl)] <- 1;
fdr[is.na(fdr)] <- 1;
rnk <- apply(-sign(l2r)*log10(pvl), 2, rank);
rownames(l2r) <- rownames(pvl) <- rownames(fdr) <- rownames(rnk) <- gid; 

corr <- round(cor(l2r[, 1], l2r[, 2]), 3);
p0 <- prms$gene$pvalue;

tbl <- lapply(stat, function(s) s[gid, c(1,2,4,5,6)]); 
cnm <- lapply(tbl, colnames); 
tbl <- do.call('cbind', tbl);
rownames(tbl) <- gid; 
tbl <- cbind(anno[rownames(tbl), ], FormatNumeric(tbl)); 
rownames(tbl) <- gid; 
colnames(tbl)[ncol(anno)+c(1:10)]<-paste(rep(names(grp), each=5), unlist(cnm), sep='::'); 
tbl[, 1] <- CleanHtmlTags(tbl[, 1], FALSE); 
tbl0 <- tbl;
write.csv(tbl0, paste(path.tbl, 'gene_stat.csv', sep='/')); 
saveRDS(tbl0, paste(path.r, 'gene_stat.rds', sep='/')); 
tbl[, 1] <- AddHref(tbl[, 1], UrlEntrezGene(rownames(tbl))); 
CreateDatatable(tbl, paste(path.tbl, 'gene_table.html', sep='/'), 
                caption='Gene-level differential expression of both pairwise comparisons')->fn; 

corr <- round(cor(l2r[, 1], l2r[, 2], use='pair'), 4); 
lns <- paste('  - Corr(', comp[1], ':', comp[2], ') = ', corr, sep='');
lns <- paste(lns, collapse='\n'); 
```

`r home.url`

## Global delta-delta correlation

Both comparisons reported the log ratio of 2 group means for each gene. The global agreement of log ratios of all genes indicates how much the results of these 2 comparisons are similar to or different from each other. Full table of gene-level statistics side-by-side is [here](table/gene_table.html). 

<div align='center'>
```{r log_ratio, include=TRUE, fig.width=6, fig.height=6, out.width='640px'}
par(mar=c(5,5,2,2)); 
labs<-sapply(names(grp), function(nm) paste(nm, paste(grp[[nm]], collapse=' vs.'), sep=', '));
rng<-range(l2r);
plot(l2r, cex=0.6, col='#88888888', pch=19, xlim=rng, ylim=rng, xlab=labs[1], ylab=labs[2], cex.lab=1.25);
abline(h=0, v=0, lty=2);
legend('topleft', bty='n', pch=19, col='#FF3333', legend=paste('DEG, p<', p0, sep=''));
d<-Reduce('intersect', lapply(deg, function(deg) unlist(lapply(deg, rownames))));
points(l2r[d, , drop=FALSE], cex=0.6, col='#FF3333', pch=19);
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 1.** This plot shows the global correlation (correlation coefficient = `r corr`) between the 2 pairwise comparisons: **`r comp[1]`** and **`r comp[2]`**. Genes having p values less than `r p0` from both comparisons are highlighted. 
</div>

`r home.url`

## Differentially expressed genes (DEGs)

```{r deg, include=TRUE}
deg.n<-sapply(deg, function(deg) sapply(deg, nrow)); 
g<-paste(sapply(grp, function(x) x[2]), collapse='/'); 
rownames(deg.n)<-paste(c('Higher', 'Lower'), 'in the 2nd group');
```

Both comparisons identified DEGs between 2 compared groups. Overlapped DEGs identified by both comparisons are worthy of a closer look. 

<div style="color:darkblue; padding:0 3cm">
**Table 1.** Number of DEGs:
</div>

<div align='center', style="padding:0 3cm">
`r kable(deg.n, align=c('c', 'c'))`
</div>

<div align='center'>
```{r deg_overlap, include=TRUE, fig.width=8, fig.height=6, out.width='800px'}
path.deg<-paste(path, 'DEG', sep='/');
if (!file.exists(path.deg)) dir.create(path.deg, recursive = TRUE); 

deg1<-lapply(deg[[1]], rownames);
deg2<-lapply(deg[[2]], rownames);
ind<-cbind(c(1, 1, 2, 2), c(1, 2, 1, 2));
rownames(ind)<-c('up-up', 'up-down', 'down-up', 'down-down');
par(mfrow=c(2,2));
apply(ind, 1, function(i) 
  PlotVenn(deg1[[i[1]]], deg2[[i[2]]], 
           paste(names(grp), c(names(deg1)[i[1]], names(deg2)[i[2]]), sep='::'), gid))->vnn;

ol <- sapply(rownames(ind), function(nm) {
  i <- ind[nm, ];
  g <- intersect(deg1[[i[1]]], deg2[[i[2]]]); 
  f <- paste('DEG/', 'gene_', nm, '.html', sep='');
  t <- tbl[g, , drop=FALSE];
  c <- paste(paste(comp, c(names(deg1)[i[1]], names(deg2)[i[2]]), sep='::'), collapse=' and ')
  fn <- CreateDatatable(t, paste(path, f, sep='/'), caption=c); 
  list(c, f); 
});

lns <- paste('  - [', ol[1, ], ']', sep=''); 
lns <- paste(lns, '(', ol[2, ], ')', sep=''); 
lns <- paste(lns, collapse='\n'); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 2.** Overlapping of DEGs. All combinations of differential expression towards opposite directions are plotted and Fisher's exact test is performed to evaluate the significance of overlapping or lack of overlapping. 
Click links below to view overlapping DEGs:

`r lns`
</div>

`r home.url`

## ANOVA

```{r anova, include=FALSE}
d <- expr.all;

# Prepare for ANOVA
s<-smpl[colnames(d), ]
for (i in 1:ncol(s)) s[[i]]<-as.factor(s[[i]]);
s$d<-d[1, ];
d0 <- d;
if (!prms$gene$processing$logged) d0 <- log2(d0+1);
if (!prms$gene$processing$normalized) d0 <- NormLoess(d0); 
d.anova<-lapply(1:nrow(d0), function(i) d0[i, ]);
f<-formula(prms$gene$anova$formula);

# Run ANOVA
paov <- parallel::mclapply(d.anova, function(x) {
  s$d<-x;
  smm<-summary(aov(f, data=s))[[1]];
  if (length(smm)==1) smm[[1]] else smm;
}, mc.cores=prms$gene$anova$core);

# sort column names
cnm<-rownames(paov[[1]]); 
cnm<-gsub(' ', '', cnm[-length(cnm)]); 
cnm0<-c(prms$gene$anova$f1, prms$gene$anova$f2, paste(prms$gene$anova$f1, prms$gene$anova$f2, sep=':'));

# p value table
paov<-t(sapply(paov, function(x) x[-nrow(x), 5])); 
dimnames(paov)<-list(rownames(d), cnm); 
paov<-paov[, c(cnm0, setdiff(cnm, cnm0))]; 
paov<-paov[order(paov[, 3]), ];
colnames(paov)<-paste('P', colnames(paov), sep='_'); 
sig.aov<-paov[paov[, 3]<=prms$gene$anova$p, , drop=FALSE]; 

# full table
m <- sapply(grps, function(g) exp(log(2)*rowMeans(d0[, g, drop=FALSE]))); 
aov.stat <- cbind(m[rownames(paov), ], paov); 
cnm <- c(colnames(anno), colnames(aov.stat));
aov.tbl <- data.frame(tbl[rownames(paov), colnames(anno)], FormatNumeric(aov.stat), stringsAsFactors = FALSE);
colnames(aov.tbl) <- cnm; 
CreateDatatable(aov.tbl, paste(path.tbl, 'anova_table.html', sep='/'), caption='ANOVA results'); 

# Write out
saveRDS(d, paste(path.r, 'expr.rds', sep='/'));
saveRDS(anno[rownames(d), ], paste(path.r, 'anno.rds', sep='/'));
saveRDS(aov.stat, paste(path.r, 'anova_stat.rds', sep='/'));
write.csv(d, paste(path.tbl, 'expr.csv', sep='/'));
write.csv(anno[rownames(d), ], paste(path.tbl, 'anno.csv', sep='/'));
write.csv(aov.stat, paste(path.tbl, 'anova_stat.csv', sep='/'));
```

2-way ANOVA analysis is performed to identify genes responding to **_`r prms$gene$anova$f2`_** differently in different **_`r prms$gene$anova$f1`_**. The analysis reported 3 p values, corresponding to the effect of **_`r prms$gene$anova$f2`_**, **_`r prms$gene$anova$f1`_**, and their interaction. The analysis identified `r nrow(sig.aov)` significant genes with interaction p values less than `r prms$gene$anova$p`. The ANOVA results are summarized in a table [here](table/anova_table.html). 

<div align='center'>
```{r aov_top, out.width='800px'}
mn <- apply(pvl, 1, min); 
sg <- paov[rownames(paov) %in% rownames(pvl)[mn < 0.05], , drop=FALSE]; 
id <- rownames(sg)[1:min(4, nrow(sg))];
ttl <- sapply(id, function(id) if (is.na(anno[id, 1])) id else paste(id, CleanHtmlTags(anno[id, 1]), sep=': ')); 

par(mar=c(8, 2, 2, 2), mfrow=c(2,2), omi=c(0, 1, 0, 0)); 
c<-rep(c("#FF666633", "#FF000088", "#66FF6633", "#00FF0088"), sapply(grps, length)); 
for (i in 1:length(id)) {
  barplot(d0[id[i], ], col=c, las=3, ylab='Normalized expression level', cex.lab=1.5, main=ttl[i], cex.main=1.5);
  if (i==1) legend('topleft', legend=names(grps), pch=15, col=unique(c), bty='n', cex = 0.5);
}
title(ylab='Normalized expression level', outer=TRUE, line=1, cex.lab=2); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 3.** Examples: the top 4 genes having the most significant interactive p value, among the genes with significant differential expression in at least one of the two pairwise comparisons. 
</div>

`r home.url`

# Gene set-level comparison

Genes are often grouped into pre-defined gene sets according to their function, interaction, location, etc. Analysis then can be performed on genes in the same gene set as a unit instead of individual genes.

## Gene set average

```{r geneset_average, include=FALSE}
# Gene set anno
lst<-gset$list;
gns<-unlist(lst, use.names=FALSE); 
gst<-rep(names(lst), sapply(lst, length));
lst<-split(gns[gns %in% gid], gst[gns %in% gid]);
sz<-sapply(lst, length);
lst<-lst[sz>=prms$geneset$min & sz<=prms$geneset$max];
sz<-sapply(lst, length); 

# split gene log-ratio into gene sets
gns<-unlist(lst, use.names=FALSE);
gst<-rep(names(lst), sapply(lst, length));
lr1<-l2r[gns, ];
splt1<-apply(lr1, 2, function(lr) split(lr, gst)); 
mean.gs1<-sapply(splt1, function(x) sapply(x, mean));
lr2<-abs(lr1);
splt2<-apply(lr2, 2, function(lr) split(lr, gst)); 
mean.gs2<-sapply(splt2, function(x) sapply(x, mean));

# statistics of log-ratio means
splt<-cbind(splt1[[1]], splt1[[2]]); 
pprs<-apply(splt, 1, function(x) t.test(x[[1]], x[[2]], paired=TRUE)$p.value[[1]]); 
dprs<-mean.gs1[, 2]-mean.gs1[, 1];

# ranked log-ratio and rank sum test
rk1<-rnk[gns, ];
splt3<-apply(rk1, 2, function(x) split(x, gst)); 
prnk<-apply(cbind(splt3[[1]], splt3[[2]]), 1, function(x) wilcox.test(x[[1]], x[[2]], paired = TRUE)$p.value[[1]]);
rnk.mn<-sapply(splt3, function(x) sapply(x, min));
rnk.mx<-sapply(splt3, function(x) sapply(x, max)); 
mrnk<-sapply(splt3, function(x) sapply(x, mean)); 
stat.rnk<-cbind(rnk.mn, rnk.mx); 

mean.gs<-list("Mean of log-ratio"=mean.gs1, "Mean of absolute(log-ratio)"=mean.gs2); 

stat.gs<-cbind(Size=sapply(splt[, 1], length), mean.gs1, Diff=mean.gs1[,2]-mean.gs1[,1], P_Ttest=pprs, FDR_Ttest=p.adjust(pprs, method='BH'), P_RST=prnk, FDR_RST=p.adjust(prnk, method='BH'));
stat.gs<-stat.gs[order(stat.gs[, 'P_RST']), ];
gs.tbl<-gset[[1]][rownames(stat.gs), ];
gs.tbl$Name<-AddHref(gs.tbl$Name, gs.tbl$URL);
gs.tbl<-cbind(gs.tbl[, 1:3], FormatNumeric(stat.gs)); 

CreateDatatable(gs.tbl, paste(path.tbl, 'geneset_table.html', sep='/'), rownames = FALSE, 
                caption='Gene set average'); 
saveRDS(gs.tbl, paste(path.r, 'geneset_stat.rds', sep='/'));
write.csv(gs.tbl, paste(path.tbl, 'geneset_stat.csv', sep='/')); 
```

Average differential expression of genes in the same gene set. The gene set-level statistics were fully summarized in this table [here](table/geneset_table.html).

<div align='center'>
```{r geneset_average_plot, include=TRUE, fig.width=8, fig.height=4, out.width='800px'}
par(mfrow=c(1, 2), mar=c(5,5,2,2)); 
fn<-sapply(names(mean.gs), function(mn) {
  x<-mean.gs[[mn]]; 
  rng<-range(x); 
  plot(x, pch=19, cex=0.6, col='#88888888', xlim=rng, ylim=rng, xlab=labs[1], ylab=labs[2], main=mn); 
  abline(0, 1, h=0, v=0, col='#0000FF88', lty=2, lwd=0.5); 
});
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 4.**  Each dot represents a gene set and the average log-ratio of all genes in this gene set. The averages were calculated with the log-ratio value of genes (_left panel_) and the absolute of the log-ratios (_right panel_). The correlation coefficients are `r round(cor(mean.gs[[1]][, 1], mean.gs[[1]][, 2]), 4)` and `r round(cor(mean.gs[[2]][, 1], mean.gs[[2]][, 2]), 4)` respectively. 
</div>

`r home.url`

## Gene set over-representation analysis (ORA)

```{r ora, include=FALSE}
path.ora<-paste(path, 'ORA', sep='/');
if (!file.exists(path.ora)) dir.create(path.ora, recursive = TRUE); 

ora<-lapply(res, function(x) x$ora); 
ora.stat<-lapply(ora, function(x) lapply(x[1:2], function(x) x[[1]])); 

ora.sets<-lapply(ora.stat, function(x) lapply(x, rownames)); 
ora.univ<-lapply(ora, function(x) lapply(x[1:2], function(x) rownames(x[[3]]))); 
ora.univ<-Reduce('union', c(ora.univ[[1]], ora.univ[[2]])); 

# Combine stats
ora.n<-lapply(ora, function(o) sapply(o[1:2], function(o) o[[3]][, 'n11'][ora.univ])); 
ora.n<-cbind(ora.n[[1]], ora.n[[2]]); 
ora.n[is.na(ora.n)]<-0; 
rownames(ora.n)<-ora.univ;
colnames(ora.n)<-paste('N', colnames(ora.n), sep='_'); 
ora.or<-lapply(ora, function(o) sapply(o[1:2], function(o) {
  c<-o[[3]]; 
  c<-(c[, 1]/c[, 2])*(c[, 4]/c[, 3]); 
  c<-c[ora.univ];
  c;
})); 
ora.or<-cbind(ora.or[[1]], ora.or[[2]]); 
ora.or[is.na(ora.or)]<-1; 
rownames(ora.or)<-ora.univ;
colnames(ora.or)<-paste('OR', colnames(ora.or), sep='_'); 
ora.or<-cbind(ora.or[, c(1, 3)], OR_High_Diff=ora.or[, 3]-ora.or[, 1], ora.or[, c(2, 4)], OR_Low_Diff=ora.or[, 4]-ora.or[, 2]);
ora.p<-lapply(ora.stat, function(o) sapply(o, function(o) o[, 'P_HyperGeo'][ora.univ]));
ora.p<-cbind(ora.p[[1]], ora.p[[2]]); 
rownames(ora.p)<-ora.univ;
colnames(ora.p)<-paste('P', colnames(ora.p), sep='_'); 
ora.p[is.na(ora.p)]<-0.5;
ora.p<-cbind(ora.p[, c(1, 3)], P_High_Ratio=ora.p[, 3]/ora.p[, 1], ora.p[, c(2, 4)], P_Low_ratio=ora.p[, 4]/ora.p[, 2]);

ora.tbl<-FormatNumeric(cbind(ora.n, ora.or, ora.p)); 
ora.tbl<-cbind(gset[[1]][rownames(ora.tbl), ], ora.tbl); 
CreateDatatable(ora.tbl, paste(path.tbl, 'ora_table.html', sep='/'), caption='Combined table of ORA statistics'); 

saveRDS(ora.tbl, paste(path.r, 'ora_stat.rds', sep='/'));
write.csv(ora.tbl, paste(path.tbl, 'ora_stat.csv', sep='/')); 

gset.src<-sort(unique(gset[[1]][, 'Source'])); 
ora.s<-c(ora.stat[[1]], ora.stat[[2]]); 
fn.tbl<-lapply(names(ora.s), function(nm) {
  s<-ora.s[[nm]]; 
  lapply(gset.src, function(src) {
    a<-gset[[1]][rownames(s), ];
    a$Name<-AddHref(a$Name, a$URL)
    a<-a[, 1:3]; 
    t<-cbind(a[a$Source==src, , drop=FALSE], FormatNumeric(s[a$Source==src, , drop=FALSE]))[, -1]; 
    f<-paste('ORA/', src, '_', nm, '.html', sep=''); 
    CreateDatatable(t, paste(path, f, sep='/'), rownames = FALSE, caption=paste(src, nm, sep=': '));
    c<-nrow(t); 
    names(c)<-f;
    c;
  });
}); 
lnk<-sapply(fn.tbl, function(fn) {
  c<-as.vector(unlist(fn));
  f<-sapply(fn, names); 
  paste('[', c, '](', f, ')', sep='');
});
colnames(lnk) <- paste(rep(comp, each=2), paste('Higher_in_', unlist(grp), sep=''), sep='::'); 
rownames(lnk) <- gset.src;
```

Each 2-group comparison performs gene set over-representation analysis (ORA) that identifies gene sets over-represented with differentially expressed genes. The results of ORA of both 2-group comparisons are summarized and compared here.  The ORA of each gene set reports an odds ratio and p value. These statistics from both comparisons were combined and listed side-by-side, as well as the difference of their odds ratios and ratio of their p values (p set to 0.5 when not available), in this table [here](table/ora_table.html)

<div style="color:darkblue; padding:0 3cm">
**Table 2.** Gene sets were broken down into subgroups by their sources. Click on the numbers of over-represented gene sets to see a full list.
</div>

<div align='center', style="padding:0 3cm">
`r kable(lnk, align='c')`
</div>

<div align='center'>
```{r ora_overlap, include=TRUE, fig.width=8, fig.height=6, out.width='800px'}
ind<-cbind(c(1, 1, 2, 2), c(1, 2, 1, 2));
rownames(ind)<-c('up-up', 'up-down', 'down-up', 'down-down');
par(mfrow=c(2,2));
apply(ind, 1, function(i) 
  PlotVenn(ora.sets[[1]][[i[1]]], ora.sets[[2]][[i[2]]], 
           c(paste(comp[1], names(ora.sets[[1]])[i[1]], sep='::'), 
             paste(comp[2], names(ora.sets[[2]])[i[2]], sep='::')), 
           u=ora.univ))->vnn;

ol<-sapply(rownames(ind), function(nm) {
  i<-ind[nm, ];
  g<-intersect(ora.sets[[1]][[i[1]]], ora.sets[[2]][[i[2]]]); 
  f<-paste('ORA/', 'ora_', nm, '.html', sep='');
  c <- paste(paste(comp, c(names(deg1)[i[1]], names(deg2)[i[2]]), sep='::'), collapse=' and ')
  t<-gset[[1]][g, ]; 
  t$Name<-AddHref(t$Name, t$URL); 
  t<-t[, 1:3]; 
  t<-cbind(t, ora.stat[[1]][[i[1]]][g, c(2, 4, 5, 6)], ora.stat[[2]][[i[2]]][g, c(2, 4, 5, 6)]); 
  colnames(t)[4:11]<-paste(colnames(t)[4:11], rep(names(grp), each=4), sep='_'); 
  fn<-CreateDatatable(t, paste(path, f, sep='/'), rownames = FALSE, caption=c); 
  list(c, f); 
});

lns<-paste('  - [', ol[1, ], ']', sep=''); 
lns<-paste(lns, '(', ol[2, ], ')', sep=''); 
lns<-paste(lns, collapse='\n'); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 5.** The overlapping of over-represented gene sets from both comparisons. Click links below to view tables of overlapping significant gene sets:

`r lns`
</div>

`r home.url`

## Gene set enrichment analysis (GSEA) 

```{r gsea, include=FALSE}
path.gsea<-paste(path, 'GSEA', sep='/');
if (!file.exists(path.gsea)) dir.create(path.gsea, recursive = TRUE); 

gsea<-lapply(res, function(x) x$gsea); 
gsea.stat<-lapply(gsea, function(x) x$stat); 
gsea.stat<-lapply(gsea.stat, function(x) {
  rownames(x)<-paste(x[[1]], x[[2]], sep=':'); 
  x;
})
gsea.univ<-Reduce('union', lapply(gsea.stat, rownames)); 
gsea.stat<-lapply(gsea.stat, function(x) x[gsea.univ, ]); 
gsea.tbl<-cbind(gsea.stat[[1]][, 1:3], gsea.stat[[1]][gsea.univ, 4:7], gsea.stat[[2]][gsea.univ, 4:7]); 
colnames(gsea.tbl)[c(4:6, 8:10)]<-paste(colnames(gsea.tbl)[c(4:6, 8:10)], rep(names(grp), each=3), sep='_'); 
gsea.tbl<-FormatNumeric(gsea.tbl); 
rownames(gsea.tbl)<-1:nrow(gsea.tbl); 
CreateDatatable(gsea.tbl, paste(path.tbl, 'gsea_table.html', sep='/'), caption='GSEA statistics'); 

saveRDS(gsea.tbl, paste(path.r, 'gsea_stat.rds', sep='/'));
write.csv(gsea.tbl, paste(path.tbl, 'gsea_stat.csv', sep='/')); 

gsea.st<-lapply(gsea.stat, function(x) split(x[x[, 'PValue']<=0.05, -ncol(x)], x[x[, 'PValue']<=0.05, ncol(x)])[2:1]); 
gsea.s<-c(gsea.st[[1]], gsea.st[[2]]);
names(gsea.s)<-names(ora.s); 
gset.cll<-sort(unique(unlist(lapply(gsea.s, function(x) x$Collection), use.names=FALSE)));
fn.tbl<-lapply(names(gsea.s), function(nm) {
  s<-gsea.s[[nm]]; 
  lapply(gset.cll, function(cll) {
    t<-s[s$Collection==cll, , drop=FALSE]; 
    t<-FormatNumeric(t); 
    f<-paste('GSEA/', cll, '_', nm, '.html', sep=''); 
    CreateDatatable(t, paste(path, f, sep='/'), rownames = FALSE, caption=paste(cll, nm, sep=': '));
    c<-nrow(t); 
    names(c)<-f;
    c;
  });
}); 
lnk<-sapply(fn.tbl, function(fn) {
  c<-as.vector(unlist(fn));
  f<-sapply(fn, names); 
  paste('[', c, '](', f, ')', sep='');
});
colnames(lnk) <- paste(rep(comp, each=2), paste('Higher_in_', unlist(grp), sep=''), sep='::'); 
rownames(lnk) <- gset.cll;
```

Each 2-group comparison performs gene set enrichment analysis (GSEA) on genes ranked by their differential expression. The results of GSEA of both 2-group comparisons are summarized and compared here. The GSEA of each gene set reports an enrichment score and p value. These statistics from both comparisons were combined and listed side-by-side in this table [here](table/gsea_table.html)

<div style="color:darkblue; padding:0 3cm">
**Table 3.** Gene sets were broken down into subgroups by collections. Click on the numbers of enriched gene sets to see a full list. 
</div>

<div align='center', style="padding:0 3cm">
`r kable(lnk, align='c')`
</div>

<div align='center'>
```{r nes, fig.width=6, fig.height=6, out.width='640px'}
par(mar=c(5,5,2,2)); 
gsea.tbl[is.na(gsea.tbl[,4]), 4]<-0; 
gsea.tbl[is.na(gsea.tbl[,8]), 8]<-0; 
rng<-range(gsea.tbl[, c(4, 8)]); 
plot(gsea.tbl[, c(4, 8)], col='#88888888', cex=0.6, pch=19, xlim=rng, ylim=rng, main='Enrichment score', xlab=labs[1], ylab=labs[2]); 
abline(h=0, v=0, col='blue', lty=2);
points(gsea.tbl[gsea.tbl[,5]<=0.05&gsea.tbl[,9]<=0.05, c(4, 8)], col='red', cex=0.6, pch=19); 
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 6.** Nominal enrichment scores from both comparisons. Each dot represents a gene set. Gene sets with p values less than 0.01 from both comparisons are highlighted. 
</div>

<div align='center'>
```{r gsea_overlap, include=TRUE, fig.width=8, fig.height=6, out.width='800px'}
gsea.st<-lapply(gsea.s, rownames);
gsea.st<-list(gsea.st[1:2], gsea.st[3:4]); 
names(gsea.st)<-names(grp);
names(gsea.st[[1]])<-names(ora.sets[[1]]);
names(gsea.st[[2]])<-names(ora.sets[[2]]);
ind<-cbind(c(1, 1, 2, 2), c(1, 2, 1, 2));
rownames(ind)<-c('up-up', 'up-down', 'down-up', 'down-down');
par(mfrow=c(2,2));
apply(ind, 1, function(i) 
  PlotVenn(gsea.st[[1]][[i[1]]], gsea.st[[2]][[i[2]]], 
           paste(names(grp), c(names(deg1)[i[1]], names(deg2)[i[2]]), sep='::'), 
           u=gsea.univ))->vnn;

ol<-sapply(rownames(ind), function(nm) {
  i<-ind[nm, ];
  g<-intersect(gsea.st[[1]][[i[1]]], gsea.st[[2]][[i[2]]]); 
  f<-paste('GSEA/', 'gsea_', nm, '.html', sep='');
  c <- paste(paste(comp, c(names(deg1)[i[1]], names(deg2)[i[2]]), sep='::'), collapse=' and ')
  t<-gsea.stat[[1]][g, 1:3]; 
  t<-t[, 1:3]; 
  t<-cbind(t, gsea.stat[[1]][g, c(4, 5, 6)], gsea.stat[[2]][g, c(4, 5, 6)]); 
  colnames(t)[4:9]<-paste(colnames(t)[4:9], rep(names(grp), each=3), sep='_'); 
  fn<-CreateDatatable(FormatNumeric(t), paste(path, f, sep='/'), rownames=FALSE, caption=c); 
  list(c, f); 
});

lns<-paste('  - [', ol[1, ], ']', sep=''); 
lns<-paste(lns, '(', ol[2, ], ')', sep=''); 
lns<-paste(lns, collapse='\n'); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 7.** The overlapping of over-represented gene sets from both comparisons. Click links to view tables of overlapping significant gene sets from GSEA:

`r lns`
</div>

`r home.url`

## Gene clustering

```{r clustering, include=FALSE}
path.cl<-paste(path, 'CLUSTER', sep='/');
if (!file.exists(path.cl)) dir.create(path.cl, recursive = TRUE);

# normalize data
d1 <- lapply(res, function(x) {
  y<-x$inputs; 
  y$expr[, unlist(y$comparison)];
});
gs <- Reduce('intersect', lapply(d1, rownames)); 
d1 <- do.call('cbind', lapply(d1, function(x) x[gs, , drop=FALSE])); 
if (!prms$gene$processing$logged) d1 <- log2(d1+1); 
if (!prms$gene$processing$normalized) d1 <- NormLoess(d1+1); 
for (i in 1:2) {
  y <- res[[i]]$inputs$comparison;
  d1[, unlist(y)] <- d1[, unlist(y)] - rowMeans(d1[, y[[1]]]);
  d1[, unlist(y)] <- t(apply(d1[, unlist(y)], 1, function(x) x/sd(x))); 
}

# select seeds
x <- paov[order(paov[, 3]), ];
x <- x[x[,3]<=prms$geneset$cluster$panova, , drop=FALSE]; 
d0 <- d1[rownames(d1) %in% rownames(x), ]; 
d0 <- d0[1:min(nrow(d0), prms$geneset$cluster$top), , drop=FALSE];

# initiate clusters
hc<-hclust(as.dist(1-cor(t(d0)))); 
cl<-cutree(hc, k=prms$geneset$cluster$seed);
cl<-split(names(cl), cl); 

# merge similar clusters
flag<-TRUE;
while(flag) {
  cat('Number of clusters ', length(cl), '\n'); 
  ms<-sapply(cl, function(cl) colMeans(d0[cl, ])); 
  tr<-cutree(hclust(as.dist(1-cor(ms))), k=length(cl)-1); # find the 2 most similar clusters
  i<-tr[duplicated(tr)];
  c<-ms[, tr==i]; 
  r<-cor(c[, 1], c[, 2]); 
  p<-cor.test(c[, 1], c[, 2])$p.value[[1]]; 
  if (r>prms$geneset$cluster$merge$corr & p<prms$geneset$cluster$merge$p) {
    cl[tr==i][[1]]<-as.vector(unlist(cl[tr==i]));
    cl<-cl[names(cl)!=names(i)]; 
  } else flag<-FALSE;
}

# Sort clusters
m<-sapply(cl, function(cl) colMeans(d1[cl, ])); 
ind<-apply(m, 2, function(x) which(x==max(x)));  
cl<-cl[order(ind)]; 
names(cl)<-paste('Cluster', 1:length(cl), sep='_'); 

# re-cluster genes
reCl<-function(d, cl, r, dif, nmax) {
  d <- d[!is.na(rowMeans(d)), , drop=FALSE];
  cl <- lapply(cl, function(cl) cl[cl %in% rownames(d)]); 
  md<-sapply(cl, function(cl) apply(d[cl[cl %in% rownames(d)], , drop=FALSE], 2, median));
  corr<-cor(t(d), md); 
  c<-lapply(1:ncol(corr), function(i) {
    corr<-corr[rev(order(corr[, i])), ]; 
    mx<-apply(corr[, -i, drop=FALSE], 1, max);
    id<-rownames(corr)[corr[, i]>=r & (corr[, i]-mx)>dif]; 
    id[1:min(length(id), nmax)]; 
  });
  c;
}
for (i in 1:prms$geneset$cluster$recluster$time) 
  cl<-reCl(d0, cl, prms$geneset$cluster$recluster$corr, prms$geneset$cluster$recluster$diff, prms$geneset$cluster$recluster$nmax); 
cls<-reCl(d1, cl, prms$geneset$cluster$recluster$corr, prms$geneset$cluster$recluster$diff, prms$geneset$cluster$recluster$nmax); 
ms<-t(sapply(cls, function(x) colMeans(d1[x, ]))); 
rownames(ms)<-names(cls)<-paste('Cluster', 1:length(cls), sep='_'); 

fn.htmp<-sapply(names(cls), function(nm) {
  x<-d1[cls[[nm]], ];
  f<-paste(path.cl, '/Heatmap_', nm, '.pdf', sep=''); 
  sz<-CalculateColoredBlockSize(x);
  pdf(f, width = max(sz[2]/3, sz[1]), height = sz[2]);
  PlotColoredBlock(x, num.breaks = 31, key = 'Normalized expression', groups = grps); 
  dev.off();
  f;
});

saveRDS(list(cluster=cls, data=d1), file=paste(path.r, 'cluster.rds', sep='/')); 
```

The top `r nrow(d0)` genes with significant ANOVA p values (p <= 'r prms$geneset$cluster$panova') were used as seeds to perform a gene-gene clustering analysis and `r length(cls)` clusters were identified. ORA was performed on the clusters to identify their functional association (see table below); 

```{r clustering_ora, include=FALSE}
fn.tbl<-lapply(names(cls), function(nm) {
  cl<-cls[[nm]]; 
  s<-TestGSE(cl, gid, gset[[2]])[[1]]; 
  t<-gset[[1]][rownames(s), ];
  t$Name<-AddHref(t$Name, t$URL); 
  t<-cbind(t[, 1:3], FormatNumeric(s)); 
  f<-paste('CLUSTER/ORA_', nm, '.html', sep='');
  CreateDatatable(t, paste(path, f, sep='/'), rownames=FALSE, caption=paste('Over-represented gene set in', nm));
  c<-nrow(t); 
  names(c)<-f;
  t<-do.call('cbind', lapply(stat, function(s) s[cl, c(1, 2, 4, 5, 6)])); 
  t<-FormatNumeric(t);
  colnames(t)[c(3,4,5,8,9,10)]<-paste(colnames(t)[c(3,4,5,8,9,10)], rep(names(grp), each=3), sep='_'); 
  t<-data.frame(anno[rownames(t), ], t, stringsAsFactors = FALSE);
  CreateDatatable(t, paste(path.cl, '/', nm, '.html', sep=''), caption=nm);
  c;
}); 
mm<-round(sapply(grps, function(x) rowMeans(ms[, x, drop=FALSE])), 4);
n<-as.vector(unlist(fn.tbl)); 
sz<-sapply(cls, length);
lnk<-paste('[', n, '](', sapply(fn.tbl, names), ')', sep='');
cl.tbl<-data.frame(Size=sz, mm, Gene_set=lnk, stringsAsFactors = FALSE);
cl.tbl[, 1]<-paste('[', sz, '](CLUSTER/', names(cls), '.html)', sep=''); 
cl.tbl<-cbind(ID=rownames(cl.tbl), cl.tbl); 
cl.tbl[, 1]<-paste('[', cl.tbl[, 1], '](CLUSTER/Heatmap_', names(cls), '.pdf)', sep=''); 
colnames(cl.tbl)[3:6] <- paste(rep(comp, each=2), unlist(grp), sep='::');
```

<div style="color:darkblue; padding:0 2cm">
**Table 4.** This table lists the number of genes in each cluster (click the numbers to see gene lists), the average expression of all genes in a cluster of all sample groups, and then the gene sets over-represented in each cluster (click the numbers to see gene set lists). The gene expression levels were normalized so the mean of the control groups equals to 0 and the mean of the treatment groups is the number of standard deviations. </div>

<div align='center', style="padding:0 2cm">
`r kable(cl.tbl, row.names=FALSE, align='c')`
</div>

<div align='center'>
```{r clustering_heatmap, include=TRUE, fig.width=CalculateColoredBlockSize(ms)[1], fig.height=CalculateColoredBlockSize(ms)[2], out.width='640px'}
mx<-max(abs(ms), na.rm=TRUE); 
PlotColoredBlock(ms, min=-1*mx, max=mx, key='Average expression', groups=grps); 
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 8.**  This plot shows below the average expression levels of each cluster. Data was normalized before the analysis, so the mean of the control groups was zero and the standard deviation of all samples of each gene was 1.0. Values indicate number of standard deviation from mean of relative control group.
</div>

<div align='center'>
```{r clustering_mean, include=TRUE, fig.width=8, fig.height=6, out.width='800px'}
ms<-t(ms); 
m<-sapply(grps, function(g) colMeans(ms[g, , drop=FALSE]));
se<-sapply(grps, function(g) apply(ms[g, , drop=FALSE],  2, function(x) sd(x)/sqrt(length(x)))); 
PlotSeries(m, se, c('', names(grps))); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 9.** This plot summarizes the group means and standard errors of all clusters.
</div>

`r home.url`

***

# Appendix 

Check out the **[RoCA home page](http://zhezhangsh.github.io/RoCA)** for more information.  

## Reproduce this report

To reproduce this report: 

  1. Find the data analysis template you want to use and an example of its pairing YAML file  [here](https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples) and download the YAML example to your working directory

  2. To generate a new report using your own input data and parameter, edit the following items in the YAML file:

    - _output_        : where you want to put the output files
    - _home_          : the URL if you have a home page for your project
    - _analyst_       : your name
    - _description_   : background information about your project, analysis, etc.
    - _input_         : where are your input data, read instruction for preparing them
    - _parameter_     : parameters for this analysis; read instruction about how to prepare input data

  3. Run the code below within ***R Console*** or ***RStudio***, preferablly with a new R session:

```{r reproduce_report, eval=FALSE, echo=TRUE}
if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis
```

If there is no complaint, go to the _output_ folder and open the ***index.html*** file to view report. 

## Session information

```{r session_info, echo=FALSE}
sessionInfo(); 
```

`r home.url`

***
**END OF DOCUMENT**
