<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Zhe Zhang" />

<meta name="date" content="2021-08-16" />

<title>Differential gene expression using DESeq2</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="index_files/navigation-1.1/tabsets.js"></script>
<link href="index_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="index_files/highlightjs-9.12.0/highlight.js"></script>
<script src="index_files/kePrint-0.0.1/kePrint.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential gene expression using DESeq2</h1>
<h4 class="author"><em>Zhe Zhang</em></h4>
<h4 class="date"><em>2021-08-16</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p>This procedure uses the <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a> package to analyze RNA-seq read count data, or similar integer matrix, with a multi-factor design. Therefore, it allows for testing differential expression between two conditions while there are confounding variables. It takes an integer matrix of gene-level read counts, normalizes it across samples, compare their gene expression profiles, and run DESeq2 to test differential expression.</p>
</div>
<p>Â </p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>Transcriptome in immune cells of control-patient samples</p>
</div>
<div id="data" class="section level2">
<h2><span class="header-section-number">1.2</span> Data</h2>
<p>Rna-seq data was generated from of 3 types of immune cells of 3 controls and 3 patients. Raw data was processed to get gene-level read counts.</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.3</span> Analysis</h2>
<p>This is a demo.</p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="analysis-and-results" class="section level1">
<h1><span class="header-section-number">2</span> Analysis and results</h1>
<ul>
<li><strong>Total number of samples: 18</strong></li>
<li><strong>Total number of genes: 25207</strong></li>
<li><strong>Number of genes after filtering: 16175</strong></li>
<li><strong>Minimum number of total read: </strong></li>
<li><strong>Comparison:</strong>
<ul>
<li><strong>Condition: Cell</strong></li>
<li><strong>Group0: B</strong> (N0=6: B_Control_A, B_Control_B, B_Control_C, B_Patient_D, B_Patient_E, B_Patient_F)</li>
<li><strong>Group1: M</strong> (N1=6: M_Control_A, M_Control_B, M_Control_C, M_Patient_D, M_Patient_E, M_Patient_F)</li>
</ul></li>
</ul>
<div id="read-count-summary" class="section level2">
<h2><span class="header-section-number">2.1</span> Read count summary</h2>
<p>The read count matrix includes <strong>18</strong> samples and <strong>16175</strong> genes after removing genes with total read counts few than . The global average of read count per gene is <strong>547.8</strong> and the overall percentage of non-zero read counts is <strong>92.28%</strong>.</p>
<div style="color:darkblue">
<p><strong>Table 1.</strong> Summary statistics of read counts per sample: quantiles of read count per gene, average read count per gene, and percentage of non-zero read counts.</p>
</div>
<table class="table table-striped table-hover table-condensed table-responsive" style="font-size: 11px; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Cell
</th>
<th style="text-align:left;">
Donor
</th>
<th style="text-align:right;">
Min.
</th>
<th style="text-align:right;">
1st Qu.
</th>
<th style="text-align:right;">
Median
</th>
<th style="text-align:right;">
Mean
</th>
<th style="text-align:right;">
3rd Qu.
</th>
<th style="text-align:right;">
Max.
</th>
<th style="text-align:right;">
Total
</th>
<th style="text-align:right;">
Positive%
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
B_Control_A
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
630.07
</td>
<td style="text-align:right;">
184.0
</td>
<td style="text-align:right;">
4406443
</td>
<td style="text-align:right;">
10191351
</td>
<td style="text-align:right;">
95.96
</td>
</tr>
<tr>
<td style="text-align:left;">
B_Control_B
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
38
</td>
<td style="text-align:right;">
534.54
</td>
<td style="text-align:right;">
165.0
</td>
<td style="text-align:right;">
3709630
</td>
<td style="text-align:right;">
8646253
</td>
<td style="text-align:right;">
96.13
</td>
</tr>
<tr>
<td style="text-align:left;">
B_Control_C
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:right;">
474.61
</td>
<td style="text-align:right;">
185.0
</td>
<td style="text-align:right;">
2430985
</td>
<td style="text-align:right;">
7676738
</td>
<td style="text-align:right;">
92.84
</td>
</tr>
<tr>
<td style="text-align:left;">
B_Patient_D
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
D
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
528.77
</td>
<td style="text-align:right;">
127.0
</td>
<td style="text-align:right;">
4536479
</td>
<td style="text-align:right;">
8552878
</td>
<td style="text-align:right;">
93.36
</td>
</tr>
<tr>
<td style="text-align:left;">
B_Patient_E
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
E
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
452.63
</td>
<td style="text-align:right;">
149.0
</td>
<td style="text-align:right;">
3364661
</td>
<td style="text-align:right;">
7321246
</td>
<td style="text-align:right;">
92.77
</td>
</tr>
<tr>
<td style="text-align:left;">
B_Patient_F
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
557.36
</td>
<td style="text-align:right;">
120.0
</td>
<td style="text-align:right;">
4738669
</td>
<td style="text-align:right;">
9015353
</td>
<td style="text-align:right;">
91.44
</td>
</tr>
<tr>
<td style="text-align:left;">
M_Control_A
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
682.18
</td>
<td style="text-align:right;">
181.0
</td>
<td style="text-align:right;">
3946922
</td>
<td style="text-align:right;">
11034214
</td>
<td style="text-align:right;">
92.87
</td>
</tr>
<tr>
<td style="text-align:left;">
M_Control_B
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
764.93
</td>
<td style="text-align:right;">
178.5
</td>
<td style="text-align:right;">
4919120
</td>
<td style="text-align:right;">
12372760
</td>
<td style="text-align:right;">
92.06
</td>
</tr>
<tr>
<td style="text-align:left;">
M_Control_C
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:right;">
594.60
</td>
<td style="text-align:right;">
221.0
</td>
<td style="text-align:right;">
2947211
</td>
<td style="text-align:right;">
9617718
</td>
<td style="text-align:right;">
95.13
</td>
</tr>
<tr>
<td style="text-align:left;">
M_Patient_D
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
D
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
462.40
</td>
<td style="text-align:right;">
115.0
</td>
<td style="text-align:right;">
3147782
</td>
<td style="text-align:right;">
7479253
</td>
<td style="text-align:right;">
90.87
</td>
</tr>
<tr>
<td style="text-align:left;">
M_Patient_E
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
E
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
464.88
</td>
<td style="text-align:right;">
136.0
</td>
<td style="text-align:right;">
2583880
</td>
<td style="text-align:right;">
7519383
</td>
<td style="text-align:right;">
89.45
</td>
</tr>
<tr>
<td style="text-align:left;">
M_Patient_F
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
528.82
</td>
<td style="text-align:right;">
137.0
</td>
<td style="text-align:right;">
3261629
</td>
<td style="text-align:right;">
8553737
</td>
<td style="text-align:right;">
91.07
</td>
</tr>
</tbody>
</table>
<p />
</div>
<div id="normalization" class="section level2">
<h2><span class="header-section-number">2.2</span> Normalization</h2>
<div id="size-factor" class="section level3">
<h3><span class="header-section-number">2.2.1</span> Size factor</h3>
<p>DESeq2 uses the <strong>median ratio</strong> to estimate a size (normalization) effect for each sample in an RNA-seq data set. Size effects are positively correlated to total read counts of samples. Therefore, a simple normalization method is to divide all read counts of each sample by its size factor.</p>
<p>The size factors of all samples range between <strong>0.627</strong> and <strong>1.593</strong> (geometric mean = <strong>1.018</strong>). In general, we expect a positive correlation between total read count of a sample and its size factor, and their geometric mean is close to 1.0.</p>
<p><img src="figure/size_factor-1.png" width="800px" /></p>
<p><strong>Figure 1.</strong> Left: correlation between median read count per gene and size factor across all samples. Right: the distribution of size factors by groups. Average size effects of the 2 groups are <strong>1.17</strong> vs. <strong>1.12</strong> (p=0.7391).</p>
</div>
<div id="before-vs.after-normalization" class="section level3">
<h3><span class="header-section-number">2.2.2</span> Before vs.Â after normalization</h3>
<p>After dividing the original read counts of each sample by its size factor, normalized read counts are log2-transformed (<strong><em>log2(c+1)</em></strong>). Data before and after normalization is compared in terms of their correlation, distribution, and variance across samples.</p>
<p><img src="figure/scatter_normalization-1.png" width="960px" /></p>
<div style="color:darkblue">
<p><strong>Figure 2.</strong> Gene-level read count of two samples with the smallest (<strong><em>T_Control_C</em></strong>) and the largest (<strong><em>M_Control_C</em></strong>) size factors.</p>
</div>
<img src="figure/box_normalization-1.png" width="960px" />
<div style="color:darkblue">
<p><strong>Figure 3.</strong> Distribution of read counts before and after normalization with size factors. Read counts were log2-transformed.</p>
</div>
</div>
<div id="data-distribution" class="section level3">
<h3><span class="header-section-number">2.2.3</span> Data distribution</h3>
<p><img src="figure/data_distribution-1.png" width="640px" /></p>
<div style="color:darkblue">
<p><strong>Figure 4.</strong> Distribution of averaged gene-level read counts after normalization by size factors. The averages often have a bi-modal (two-peak) distribution. The narrow left peak corresponds to inactive genes with lower read counts and the wider right peak corresponds to active genes expressed at different levels.</p>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="sample-analysis" class="section level2">
<h2><span class="header-section-number">2.3</span> Sample analysis</h2>
<p>Analysis of samples by comparing their global gene expression profiles, potentially to identify outlier samples and confounding variables.</p>
<div id="hierarchical-clustering" class="section level3">
<h3><span class="header-section-number">2.3.1</span> Hierarchical clustering</h3>
<p><img src="figure/hier_clustering-1.png" width="640px" /></p>
<div style="color:darkblue">
<p><strong>Figure 5.</strong> Unsupervised hierarchical clustering of samples based on their pairwise correlation of all genes (normalzied and log2-transformed). The lowest common node of two samples on the clustering tree represents their similarity (lower = more similar). Colors in the heatmap correspond to the correlation coefficient of a pair of samples.</p>
</div>
</div>
<div id="principal-components-analysis" class="section level3">
<h3><span class="header-section-number">2.3.2</span> Principal components analysis</h3>
<p><img src="figure/pca-1.png" width="640px" /></p>
<div style="color:darkblue">
<p><strong>Figure 6.</strong> Principal Components Analysis (PCA) is also an unsupervised analysis that converts a large number of correlated variables (genes) into a smaller set of uncorrelated variables called principal components (PCs). Each principal component accounts for certain percentage of total variance in a data set and the PCs are ordered by their percentages. This figure plots the top two PCs. Generally speaking, samples closer to each other on the 2-D space have more similar profiles of gene expression.</p>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="differential-gene-expression" class="section level2">
<h2><span class="header-section-number">2.4</span> Differential gene expression</h2>
<p>The analysis of differential gene expression reports a statistical table with 6 columns:</p>
<ul>
<li><strong>Column 1/2 - Group means:</strong> Group averages of normalized read counts (<strong><em>B</em></strong> vs. <strong><em>M</em></strong>).</li>
<li><strong>Column 3 - Mean_Change:</strong> Difference of group means (<strong><em>Mean1 - Mean0</em></strong>).</li>
<li><strong>Column 4 - Log2FC:</strong> Log2 ratio of group means (<strong><em>Log2(Mean1/Mean0)</em></strong>); adjusted by shrinkage esimator.</li>
<li><strong>Column 5 - Pvalue:</strong> Statistical significance of group difference, calculated by a negative binomial (Wald) test.</li>
<li><strong>Column 6 - FDR:</strong> Adjusted p value known as false discovery rate, calculated by the Benjamini &amp; Hochberg method.</li>
</ul>
<div id="deg-summary" class="section level3">
<h3><span class="header-section-number">2.4.1</span> DEG summary</h3>
<p><img src="figure/deg_plot-1.png" width="800px" /></p>
<div style="color:darkblue">
<p><strong>Figure 7.</strong> Visualizations that summarize the differential expression of all 16175 genes.</p>
<ul>
<li><strong>M-A Plot</strong> This plot visualizes the relationship between average read counts on X-axis (<strong>A</strong>) and Log2 fold change on Y-axis (<strong>M</strong>). Each dot represents a gene. The red line is the LOWESS local smoothing fit, which indicates whether there is an overall pattern of skewed differential expression across the spectrum of gene expression levels.</li>
<li><strong>Volcano Plot</strong> This plot puts together two types of differential expression statistics: <strong>Log2 fold change</strong> (size effect) and <strong>P value</strong> (statistical significance). Log2 fold change indicates magnitude of differential expression and p value indicates the consistency of change across multiple samples. Selection of differentially expressed genes usually needs to take both into account. Each dot represents a gene and its size is proportional to its distance from [0, 1]. The horizontal line corresponds to p=0.01 and the vertical lines correspond to 2.0 fold increase/decrease.</li>
<li><strong>P Value Distribution</strong> This plot shows the counts of genes whose DESeq2 p values falling into each 1% interval. The first bar on the left then indicates the number of genes with p&lt;0.01. When the data is completely random with enough samples, the number of genes in each 1% interval will be approximately the same (uniform distribution). The genes with significant p values would be considered as false positives if this is the case. A distribution skewed to the left suggests there are indeed differentially expressed genes (true positives).</li>
<li><strong>False discovery rate</strong> This plots traces the number of selected genes with each FDR (false discovery rate) cutoff.</li>
</ul>
</div>
<div style="color:darkblue">
<p><strong>Table 2.</strong> Number of top DEGs selected via different FDR cutoffs. FDRs are calculated using the Benjamini &amp; Hochberg method (<em>Controlling the false discovery rate: a practical and powerful approach to multiple testing</em>. Journal of the Royal Statistical Society Series B 57, 289â300. 1995). A FDR of 0.15 to 0.25 (15% to 25% false positives) is acceptable for most follow-up analyses, while validation of differential expression via other technologies, such as qPCR, is highly recommended.</p>
</div>
<table class="table table-striped table-hover table-condensed table-responsive" style="font-size: 11px; width: auto !important; ">
<thead>
<tr>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:right;">
Higher_in_M
</th>
<th style="text-align:right;">
Lower_in_M
</th>
<th style="text-align:right;">
Total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
2896
</td>
<td style="text-align:right;">
2754
</td>
<td style="text-align:right;">
5650
</td>
</tr>
<tr>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
3234
</td>
<td style="text-align:right;">
3132
</td>
<td style="text-align:right;">
6366
</td>
</tr>
<tr>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
3685
</td>
<td style="text-align:right;">
3727
</td>
<td style="text-align:right;">
7412
</td>
</tr>
<tr>
<td style="text-align:right;">
0.10
</td>
<td style="text-align:right;">
4131
</td>
<td style="text-align:right;">
4297
</td>
<td style="text-align:right;">
8428
</td>
</tr>
<tr>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
4425
</td>
<td style="text-align:right;">
4725
</td>
<td style="text-align:right;">
9150
</td>
</tr>
<tr>
<td style="text-align:right;">
0.20
</td>
<td style="text-align:right;">
4708
</td>
<td style="text-align:right;">
5039
</td>
<td style="text-align:right;">
9747
</td>
</tr>
<tr>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
4937
</td>
<td style="text-align:right;">
5361
</td>
<td style="text-align:right;">
10298
</td>
</tr>
</tbody>
</table>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="top-genes" class="section level3">
<h3><span class="header-section-number">2.4.2</span> Top genes</h3>
<p><img src="figure/deg_no1-1.png" width="800px" /></p>
<div style="color:darkblue">
<p><strong>Figure 8.</strong> The top-ranked gene with the most up- (left) and down- (right) regulated expression in M.</p>
</div>
<p><img src="figure/deg_heatmap-1.png" width="640px" /></p>
<div style="color:darkblue">
<p><strong>Figure 9.</strong> Heatmap of the top 100 and 100 DEGs with the most up- (red) and down- (yellow) regulated expression in M. Each row corresponds a DEG and the colors represent its replative expression levels across samples. Samples are clustered by the genes and the columns are colored according to sample groups (blue=<strong><em>B</em></strong> and red=<strong><em>M</em></strong>).</p>
</div>
</div>
</div>
</div>
<div id="data-access" class="section level1">
<h1><span class="header-section-number">3</span> Data access</h1>
<p />
<div id="browse-online" class="section level2">
<h2><span class="header-section-number">3.1</span> Browse online</h2>
<p><strong>Differential gene expression:</strong></p>
<ul>
<li><a href="deg/all_genes.html">All genes</a>
<ul>
<li><a href="deg/Higher_in_M/index.html">100 DEGs with higher expression in M</a></li>
<li><a href="deg/Lower_in_M/index.html">100 DEGs with lower expression in M</a></li>
</ul></li>
</ul>
<p><strong>Principal components analysis</strong></p>
<ul>
<li><a href="table/pca.html">Principal components</a></li>
<li><a href="table/pca_loading.html">Gene loading</a></li>
</ul>
</div>
<div id="download" class="section level2">
<h2><span class="header-section-number">3.2</span> Download</h2>
<ul>
<li><a href="deg/deg.xlsx">DEG (Excel)</a></li>
<li><a href="deg/deg.zip">DEG (All files)</a></li>
<li><a href="table/pca.xlsx">PCA (Excel)</a></li>
<li><a href="R/R.zip">R objects</a></li>
<li><a href="result.zip">All files</a></li>
</ul>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
</div>
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">4</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="fold-change-vs.log2fold-change" class="section level2">
<h2><span class="header-section-number">4.1</span> Fold change vs.Â log2(fold change)</h2>
<p>The terms to represent differential expression can be used quite confusingly. In this report, <strong>fold change</strong> refers the ratio of two group means in their unlogged form. So a fold change of 2.0 means the average of the second group is increased to twice of the average of the first group; similarly, a fold change of 0.5 means the average is reduced to half. <strong>Log2(fold change)</strong> equals to the log2-transformation of the fold change. The table below gives a few examples of the conversion of these 2 variables. <strong>Log2(fold change)</strong> is more suitable for statistical analysis since it is symmetric around 0.</p>
<div style="color:darkblue">
<p><strong>Supplemental Table 1.</strong> Fold Change vs.Â Log(Fold Change) vs.Â Percentage Change</p>
</div>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
Fold change
</th>
<th style="text-align:right;">
Log2(fold change)
</th>
<th style="text-align:right;">
Percentage change (%)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
0.125
</td>
<td style="text-align:right;">
-3.000
</td>
<td style="text-align:right;">
-87.500
</td>
</tr>
<tr>
<td style="text-align:right;">
0.250
</td>
<td style="text-align:right;">
-2.000
</td>
<td style="text-align:right;">
-75.000
</td>
</tr>
<tr>
<td style="text-align:right;">
0.500
</td>
<td style="text-align:right;">
-1.000
</td>
<td style="text-align:right;">
-50.000
</td>
</tr>
<tr>
<td style="text-align:right;">
0.667
</td>
<td style="text-align:right;">
-0.585
</td>
<td style="text-align:right;">
-33.333
</td>
</tr>
<tr>
<td style="text-align:right;">
0.800
</td>
<td style="text-align:right;">
-0.322
</td>
<td style="text-align:right;">
-20.000
</td>
</tr>
<tr>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:right;">
1.250
</td>
<td style="text-align:right;">
0.322
</td>
<td style="text-align:right;">
25.000
</td>
</tr>
<tr>
<td style="text-align:right;">
1.500
</td>
<td style="text-align:right;">
0.585
</td>
<td style="text-align:right;">
50.000
</td>
</tr>
<tr>
<td style="text-align:right;">
2.000
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
100.000
</td>
</tr>
<tr>
<td style="text-align:right;">
4.000
</td>
<td style="text-align:right;">
2.000
</td>
<td style="text-align:right;">
300.000
</td>
</tr>
<tr>
<td style="text-align:right;">
8.000
</td>
<td style="text-align:right;">
3.000
</td>
<td style="text-align:right;">
700.000
</td>
</tr>
</tbody>
</table>
<p />
</div>
<div id="references" class="section level2">
<h2><span class="header-section-number">4.2</span> References</h2>
<ul>
<li><strong>R:</strong> R Development Core Team, 2011. <em>R: A Language and Environment for Statistical Computing.</em> ISBN 3-900051-07-0. <a href="http://www.R-project.org">Home page</a>.</li>
<li><strong>Bioconductor:</strong> Gentleman RC et al., 2004. <em>Bioconductor: open software development for computational biology and bioinformatics.</em> Genome Biology. <a href="http://www.bioconductor.org">Home page</a>.</li>
<li><strong>DESeq2:</strong> Love MI, Huber W, Anders S (2014). <em>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.</em> <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">Genome Biology, 15, 550</a>.</li>
<li><strong><a href="http://zhezhangsh.github.io/RoCA">RoCA</a></strong></li>
<li><strong><a href="awsomics.org">Awsomics</a></strong></li>
</ul>
</div>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">4.3</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p>
<ul>
<li><strong>output</strong> : where you want to put the output files</li>
<li><strong>home</strong> : the URL if you have a home page for your project</li>
<li><strong>analyst</strong> : your name</li>
<li><strong>description</strong> : background information about your project, analysis, etc.</li>
<li><strong>input</strong> : where are your input data, read instruction for preparing them</li>
<li><strong>parameter</strong> : parameters for this analysis; read instruction about how to prepare input data</li>
</ul></li>
<li><p>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</p></li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">4.4</span> Session information</h2>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] xlsx_0.6.1                  DEGandMore_0.0.0.9000      
##  [3] snow_0.4-3                  DESeq2_1.22.1              
##  [5] SummarizedExperiment_1.12.0 DelayedArray_0.8.0         
##  [7] BiocParallel_1.16.2         matrixStats_0.54.0         
##  [9] Biobase_2.42.0              GenomicRanges_1.34.0       
## [11] GenomeInfoDb_1.18.1         IRanges_2.16.0             
## [13] S4Vectors_0.20.1            BiocGenerics_0.28.0        
## [15] rchive_0.0.0.9000           colorspace_1.4-1           
## [17] gplots_3.0.1                MASS_7.3-51.1              
## [19] htmlwidgets_1.5.1           DT_0.15                    
## [21] kableExtra_0.9.0            awsomics_0.0.0.9000        
## [23] yaml_2.2.1                  rmarkdown_1.10             
## [25] knitr_1.20                  RoCA_0.0.0.9000            
## [27] RCurl_1.95-4.11             bitops_1.0-6               
## [29] devtools_2.3.1              usethis_1.6.1              
## 
## loaded via a namespace (and not attached):
##  [1] ellipsis_0.3.0         rprojroot_1.3-2        htmlTable_1.12        
##  [4] XVector_0.22.0         base64enc_0.1-3        fs_1.3.2              
##  [7] rstudioapi_0.11        remotes_2.2.0          bit64_0.9-7           
## [10] AnnotationDbi_1.44.0   fansi_0.4.1            xml2_1.2.0            
## [13] splines_3.5.1          geneplotter_1.60.0     pkgload_1.0.2         
## [16] jsonlite_1.6.1         Formula_1.2-3          rJava_0.9-11          
## [19] annotate_1.60.0        cluster_2.0.7-1        readr_1.3.1           
## [22] compiler_3.5.1         httr_1.4.2             backports_1.1.5       
## [25] assertthat_0.2.1       Matrix_1.2-15          cli_2.0.2             
## [28] acepack_1.4.1          htmltools_0.4.0        prettyunits_1.1.1     
## [31] tools_3.5.1            gtable_0.3.0           glue_1.3.2            
## [34] GenomeInfoDbData_1.2.0 dplyr_0.8.5            Rcpp_1.0.5            
## [37] vctrs_0.2.4            gdata_2.18.0           crosstalk_1.1.0.1     
## [40] stringr_1.3.1          xlsxjars_0.6.1         ps_1.3.2              
## [43] testthat_2.3.2         rvest_0.3.2            lifecycle_0.2.0       
## [46] gtools_3.8.1           XML_3.98-1.16          zlibbioc_1.28.0       
## [49] scales_1.1.1           hms_0.4.2              RColorBrewer_1.1-2    
## [52] memoise_1.1.0          gridExtra_2.3          ggplot2_3.3.2         
## [55] rpart_4.1-13           latticeExtra_0.6-28    stringi_1.2.4         
## [58] RSQLite_2.1.1          highr_0.7              genefilter_1.64.0     
## [61] desc_1.2.0             checkmate_1.8.5        caTools_1.17.1.1      
## [64] pkgbuild_1.1.0         rlang_0.4.5            pkgconfig_2.0.3       
## [67] evaluate_0.14          lattice_0.20-38        purrr_0.3.3           
## [70] bit_1.1-14             processx_3.4.2         tidyselect_1.0.0      
## [73] magrittr_1.5           R6_2.4.1               Hmisc_4.1-1           
## [76] DBI_1.0.0              pillar_1.4.6           foreign_0.8-71        
## [79] withr_2.2.0            survival_2.43-3        nnet_7.3-12           
## [82] tibble_2.1.3           crayon_1.3.4           KernSmooth_2.23-15    
## [85] locfit_1.5-9.1         grid_3.5.1             data.table_1.12.8     
## [88] blob_1.2.1             callr_3.4.3            digest_0.6.25         
## [91] xtable_1.8-3           munsell_0.5.0          viridisLite_0.3.0     
## [94] sessioninfo_1.1.1</code></pre>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
<p><em>END OF DOCUMENT</em></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
