<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Zhe Zhang" />

<meta name="date" content="2016-05-10" />

<title>Summarize &amp; filter sequence reads</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="index_files/highlight/default.css"
      type="text/css" />
<script src="index_files/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="index_files/navigation-1.0/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Summarize &amp; filter sequence reads</h1>
<h4 class="author"><em>Zhe Zhang</em></h4>
<h4 class="date"><em>2016-05-10</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p><strong>Introduction</strong> This analysis summarizes and filters sequencing reads that have been aligned to the reference based on SAM fields such as <strong><em>FLAG</em></strong>, <strong><em>MAPQ</em></strong>, and <strong><em>CIGAR</em></strong>. It uses the readGAlignments (single-ended reads) and readGAlignmentPairs (pair-ended reads) to import aligned reads from indexed BAM files. Summary statistics of imported reads are compared between samples for consistence and potential outliers. These statistics are</p>
<ul>
<li><strong><em>strand balance:</em></strong> reads mapped to forward vs. reverse strands</li>
<li><strong><em>mapping flag:</em></strong> a single bitwise value that summarizes the alignment result of each read</li>
<li><strong><em>mapping quality:</em></strong> p value of mapping certainty assigned by the aligment program</li>
<li><strong><em>base count:</em></strong> number of bases in each read with insertion, getting trimmed, etc. (CIGAR operations)</li>
<li><strong><em>matching length:</em></strong> total number of bases in each read matching to the reference</li>
</ul>
<p>Finally, the reads will be filtered based on given criteria covering 4 possible rules:</p>
<ul>
<li><strong><em>mapping region:</em></strong> reads must be mapped to given genomic region(s)</li>
<li><strong><em>mapping flag:</em></strong> the SAM field <em>FLAG</em> must have one of given values</li>
<li><strong><em>mapping quality:</em></strong> the SAM field <em>MAPQ</em> must have value greater than or equal to a given value</li>
<li><strong><em>matching length:</em></strong> the total number of bases matching the reference must be greater than or equal to a given value</li>
</ul>
</div>
<p> </p>
<div align="right">
<em><a href="http://grantome.com/grant/NIH/R01-AR058547-03">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>H3K4me3 of SLE in immune cells</p>
</div>
<div id="experiment" class="section level2">
<h2><span class="header-section-number">1.2</span> Experiment</h2>
<p>H3K4me3 was measured in B cell, T cell, and monocyte via ChIP-seq</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.3</span> Analysis</h2>
<p>Aligned single-ended ChIP-seq reads were loaded into R as GAlignments objects and filtered by their length, map quality, etc. Only a small portion of the total reads of each library were randomly selected as input for demo purpose Libraries were compared in this report using their summary statistics.</p>
<div align="right">
<em><a href="http://grantome.com/grant/NIH/R01-AR058547-03">Go to project home</a></em>
</div>
</div>
</div>
<div id="summary-statistics" class="section level1">
<h1><span class="header-section-number">2</span> Summary statistics</h1>
<div id="mapping-flag" class="section level2">
<h2><span class="header-section-number">2.1</span> Mapping <strong><em>FLAG</em></strong></h2>
<p>Each alignment software could assign a bitwise value as a combination of FLAGs to each read to represent the result of the alignment, as descrbed in <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM format manual</a>, Section 1.4. The bitwise value is 0 if the read is aligned “normally” and uniquely to the forward strand, and is 16 if the read is aligned “normally” and uniquely to the reverse strand. Any other values would suggest that there is something “wrong” with the alignment. The majority of the alignment will assign a strand to the read, but in some rare occasions, the strand is ambiguous and there is no strand information available for the read. Click <a href="table/flag_strand.html">here</a> to view the summary statistics about the FLAGs and alignment strands.</p>
<div align="center">
<img src="figure/strand-1.png" title="plot of chunk strand" alt="plot of chunk strand" width="600px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 1</strong> Numbers of reads aligned to the forward and reverse strands. Outliers were identified via fitting linear models (Sigma &gt; 3.0). The solid blue line corresponds to the same 50%-50% of the reads aligned to the 2 strands, and the dashed lines is the actual fitting lines of a linear model. The percents of reads aligned to the forward strand the are between 49.1171% and 50.8829% (mean = 50%). For paired end reads, only the first read of each pair was used.
</div>
<div align="right">
<em><a href="http://grantome.com/grant/NIH/R01-AR058547-03">Go to project home</a></em>
</div>
</div>
<div id="mapping-quality-mapq" class="section level2">
<h2><span class="header-section-number">2.2</span> Mapping quality <strong><em>MAPQ</em></strong></h2>
<p>Each alignment software could use its own algorithm to assign an alignment score to each align reads. The scores are usually non-negative integers, with 0 corresponding to the lowest mapping quality.</p>
<ul>
<li><a href="table/mapq_count.html">Number of reads with each MAPQ score</a></li>
<li><a href="table/mapq_percent.html">Percent of reads with each MAPQ score</a></li>
</ul>
<div align="center">
<img src="figure/mapq_heatmap-1.png" title="plot of chunk mapq_heatmap" alt="plot of chunk mapq_heatmap" width="700px" />
</div>
<div style="color:darkblue; padding:0 1cm">
<strong>Figure 2</strong> Each cell indicates the relative frequency of reads with a given mapping score, comparing to all the other samples.
</div>
<div align="center">
<img src="figure/mapq_hist-1.png" title="plot of chunk mapq_hist" alt="plot of chunk mapq_hist" width="720px" />
</div>
<div style="color:darkblue; padding:0 1cm">
<strong>Figure 3</strong> Each plot shows the distribution of read frequencies with the lowest, highest, and the most common mapping quality scores.
</div>
<div align="right">
<em><a href="http://grantome.com/grant/NIH/R01-AR058547-03">Go to project home</a></em>
</div>
</div>
<div id="cigar-operations" class="section level2">
<h2><span class="header-section-number">2.3</span> <strong><em>CIGAR</em></strong> operations</h2>
<p>CIGAR string indicates the internal structure of each alignment such as the base where insertion or deletion starts. By breaking CIGAR string of each read, bases belonging to each CIGAR operation was counted:</p>
<ul>
<li><a href="table/cigar_count.html">Number of bases with each CIGAR operation</a></li>
<li><a href="table/cigar_percent.html">Percent of bases with CIGAR operation</a></li>
</ul>
<div align="center">
<img src="figure/cigar_heatmap-1.png" title="plot of chunk cigar_heatmap" alt="plot of chunk cigar_heatmap" width="200px" />
</div>
<div style="color:darkblue; padding:0 4cm">
<strong>Figure 4</strong> Each cell indicates the relative frequency of reads with a given character in their CIGAR strings, comparing to all the other samples.
</div>
<div align="right">
<em><a href="http://grantome.com/grant/NIH/R01-AR058547-03">Go to project home</a></em>
</div>
</div>
<div id="matching-length" class="section level2">
<h2><span class="header-section-number">2.4</span> Matching length</h2>
<p>The matching length (total number of matched bases) of each read can be retrieved from the <strong><em>M</em></strong> operation in its CIGAR string. Matching length of all reads in each library was summarized to get the <a href="table/matching_length_percent.html">cummulative percents</a> of reads with matching length no greater than a given number of bases.</p>
<div align="center">
<img src="figure/match_percent-1.png" title="plot of chunk match_percent" alt="plot of chunk match_percent" width="720px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 5</strong> Cummulative percent of sequence reads with matching length no greater than a given number. Blue line is the average of all libraries whereas the dashed lines correspond to the minimum and maximum of all libraries. This plot roughly indicates the proportion of reads that will be filtered out with each cutoff value of matching length.
</div>
<div align="center">
<img src="figure/match_length-1.png" title="plot of chunk match_length" alt="plot of chunk match_length" width="600px" />
</div>
<div style="color:darkblue; padding:0 4cm">
<strong>Figure 6</strong> Numbers of total reads vs. the reads with longest matching length in each library. Outliers were identified via fitting linear models (Sigma &gt; 3.0). The blue lines is the fitting to the linear model using all libraries.
</div>
<div align="right">
<em><a href="http://grantome.com/grant/NIH/R01-AR058547-03">Go to project home</a></em>
</div>
</div>
</div>
<div id="read-filtering" class="section level1">
<h1><span class="header-section-number">3</span> Read filtering</h1>
<div id="step-by-step-filtering" class="section level2">
<h2><span class="header-section-number">3.1</span> Step-by-step filtering</h2>
<p>Sequence reads were filtered by the following steps to keep those</p>
<ul>
<li>mapped to these <a href="/table/region.html">genomic regions</a></li>
<li>with the <strong><em>FLAG</em></strong> field of SAM format having one of these values: <em>0; 16</em>.</li>
<li>with the <strong><em>MAPQ</em></strong> field of SAM format having values greater than or equal to <strong><em>20, 20</em></strong>.</li>
<li>with at least <strong><em>20, 20</em></strong> mapped bases based on the <strong><em>CIGAR</em></strong> field of SAM format.</li>
</ul>
<p>The details about SAM format can be found <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">here</a>.</p>
<p>The number and percent of remaining reads after each filter step are listed <a href="table/remained_number.html">here</a> and <a href="table/remained_percent.html">here</a>.</p>
<div align="center">
<img src="figure/read_percent-1.png" title="plot of chunk read_percent" alt="plot of chunk read_percent" width="600px" />
</div>
<div style="color:darkblue; padding:0 3cm">
<strong>Figure 7.</strong> Each box represents the distribution of remaining read percents of all samples.
</div>
<div align="right">
<em><a href="http://grantome.com/grant/NIH/R01-AR058547-03">Go to project home</a></em>
</div>
</div>
<div id="final-filtering-result" class="section level2">
<h2><span class="header-section-number">3.2</span> Final filtering result</h2>
<p>The final remaining percents of original reads after all filtering steps were compared between samples. The numbers of reads before and after filtering were fit to a linear regression based on the assumption that all samples had the same remaining percent. Using this model, the sigma value of each sample was calculated as the difference between predicted and actual numbers of remaining reads. Samples with sigmas greater than 3.0 could be considered as outliers with extraordinarily low or high remaining percent. Click <a href="table/read_count.html">here</a> to see the result table.</p>
<div align="center">
<img src="figure/before_vs_after-1.png" title="plot of chunk before_vs_after" alt="plot of chunk before_vs_after" width="600px" />
</div>
<div style="color:darkblue; padding:0 3cm">
<strong>Figure 8</strong> Numbers of reads before and after filtering.
</div>
<div align="right">
<em><a href="http://grantome.com/grant/NIH/R01-AR058547-03">Go to project home</a></em>
</div>
<hr />
<p><strong>END_OF_DOCUMENT</strong></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
