<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jim Zhang" />

<meta name="date" content="2016-06-07" />

<title>Summary of STAR alignment</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="index_files/highlight/default.css"
      type="text/css" />
<script src="index_files/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="index_files/navigation-1.0/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Summary of STAR alignment</h1>
<h4 class="author"><em>Jim Zhang</em></h4>
<h4 class="date"><em>2016-06-07</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p><strong>Introduction</strong> This report is a summary of outputs from the STAR program that aligned the sequence reads of a set of RNA-seq libraries. It uses information provided by the <strong><em>Log.final.out</em></strong> file generated by STAR from each RNA-seq library. Its primary goal is to evaluate the consistency of several summary statistics, such as alignment rate and mismatch frequency, between multiple RNA-seq libraries.</p>
</div>
<p> </p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>RNA-seq data in 3 immune cells of 4 donors</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.2</span> Analysis</h2>
<p>This is a demo.</p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="star-options" class="section level1">
<h1><span class="header-section-number">2</span> STAR options</h1>
<p>STAR was run using the following options:</p>
<ul>
<li><strong>program</strong>: STAR-STAR_2.4.1c/bin/Linux_x86_64_static/STAR</li>
<li><strong>output</strong>: star/pass_2</li>
<li><strong>genome</strong>: refs/hg38/star</li>
<li><strong>transcriptome</strong>: refs/hg38/genes.gtf</li>
<li><strong>star</strong>:
<ul>
<li><strong>runThreadN</strong>: 8</li>
<li><strong>readFilesCommand</strong>: zcat</li>
<li><strong>outSAMtype</strong>: BAM</li>
<li><strong>chimSegmentMin</strong>: 32</li>
<li><strong>outFilterType</strong>: BySJout</li>
<li><strong>outFilterMultimapNmax</strong>: 20</li>
<li><strong>alignSJoverhangMin</strong>: 8</li>
<li><strong>alignSJDBoverhangMin</strong>: 1</li>
<li><strong>outFilterMismatchNmax</strong>: 999</li>
<li><strong>outFilterMismatchNoverLmax</strong>: 0.04</li>
<li><strong>alignIntronMin</strong>: 20</li>
<li><strong>alignIntronMax</strong>: 1000000</li>
<li><strong>alignMatesGapMax</strong>: 1000000</li>
</ul></li>
</ul>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="summary-statistics" class="section level1">
<h1><span class="header-section-number">3</span> Summary statistics</h1>
<div style="color:darkblue; padding:0 2cm">
<strong>Table 1.</strong> Summary of the summary statistics of all libraries, including the total number of sequence reads, percent of uniquely mapped reads, etc. Click here to see the summary statistics of individual libraries.
</div>
<div align="center" ,="" style="padding:0 2cm">
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Min.</th>
<th align="right">1st Qu.</th>
<th align="right">Median</th>
<th align="right">Mean</th>
<th align="right">3rd Qu.</th>
<th align="right">Max.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Total input, million reads</td>
<td align="right">30.74</td>
<td align="right">32.7000</td>
<td align="right">34.480</td>
<td align="right">36.3400</td>
<td align="right">40.3900</td>
<td align="right">44.52</td>
</tr>
<tr class="even">
<td align="left">Alignment rate (%), unique mapping</td>
<td align="right">80.45</td>
<td align="right">82.9800</td>
<td align="right">83.600</td>
<td align="right">84.1800</td>
<td align="right">85.3400</td>
<td align="right">88.79</td>
</tr>
<tr class="odd">
<td align="left">Alignment rate (%), unique + multiple</td>
<td align="right">92.64</td>
<td align="right">93.3500</td>
<td align="right">93.640</td>
<td align="right">93.6200</td>
<td align="right">93.8700</td>
<td align="right">94.52</td>
</tr>
<tr class="even">
<td align="left">Mismatch rate (%)</td>
<td align="right">0.27</td>
<td align="right">0.2800</td>
<td align="right">0.295</td>
<td align="right">0.2992</td>
<td align="right">0.3200</td>
<td align="right">0.33</td>
</tr>
<tr class="odd">
<td align="left">Deletion rate (%)</td>
<td align="right">0.02</td>
<td align="right">0.0200</td>
<td align="right">0.020</td>
<td align="right">0.0200</td>
<td align="right">0.0200</td>
<td align="right">0.02</td>
</tr>
<tr class="even">
<td align="left">Insertion rate (%)</td>
<td align="right">0.01</td>
<td align="right">0.0100</td>
<td align="right">0.010</td>
<td align="right">0.0100</td>
<td align="right">0.0100</td>
<td align="right">0.01</td>
</tr>
<tr class="odd">
<td align="left">Too many loci (%)</td>
<td align="right">0.05</td>
<td align="right">0.0675</td>
<td align="right">0.110</td>
<td align="right">0.1033</td>
<td align="right">0.1325</td>
<td align="right">0.15</td>
</tr>
<tr class="even">
<td align="left">Too many mismatch (%)</td>
<td align="right">0.09</td>
<td align="right">0.1000</td>
<td align="right">0.110</td>
<td align="right">0.1125</td>
<td align="right">0.1200</td>
<td align="right">0.14</td>
</tr>
<tr class="odd">
<td align="left">Too short (%)</td>
<td align="right">5.23</td>
<td align="right">5.7850</td>
<td align="right">6.055</td>
<td align="right">6.0860</td>
<td align="right">6.4050</td>
<td align="right">7.10</td>
</tr>
</tbody>
</table>
</div>
<div id="alignment-rate" class="section level2">
<h2><span class="header-section-number">3.1</span> Alignment rate</h2>
<p>In most RNA-seq data sets, the percentage of total input reads that can be aligned to reference genome/transcriptome could range between 50% and 90%. Alignment rate is an important quality index of RNA-seq library and high throughput sequencing. However, it also highly depends on the experimental material and protocol, so it is hard to have a predefined cutoff of “high” alignment rate for all data sets. On the other hand, the consistence of alignment rates between samples of the same data set is at least equally important. Inconsistency of alignment rates is usually the consequence of systematic bias during the whole experimental procedure. It adds unwanted between-sample variance into data and might have profound impact on statistic analysis, such as differential gene expression. Therefore, the focus of this analysis is whether there are libraries having much lower alignment rates than the others.</p>
<p>The rate of unique vs. multiple alignment is a similar index of data quality. High percent of multiple alignment might indicate low complexity of sequence reads, higher sequencing error rate, and other issues. This analysis also evaluates the consistency of unique vs. multiple alignment between samples.</p>
<div align="center">
<img src="figure/alignment_rate-1.png" title="plot of chunk alignment_rate" alt="plot of chunk alignment_rate" width="800px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 1.</strong> The global alignment rate (left) and the rate of unique vs. multiple alignment (right). Each spot represents a RNA-seq library and is colored based on number of sigma. For each library, a linear model is built with all the other libraries and the value of sigma (variance of random error) is obtained from the model. The number of sigma is then calculated by dividing the observed-predicted difference of that library with the sigma value.
</div>
</div>
<div id="non-canonical-splice-sites" class="section level2">
<h2><span class="header-section-number">3.2</span> Non-canonical splice sites</h2>
<p>An important aspect of processing RNA-seq data is to alignment sequence reads to splicing sites, called gap alignment. Most commonly, STAR performs gap alignment first by using the known splicing sites based on the reference transcriptome and then by detecting novel splicing sites based on the reference genome. Most splicing sites have canonical donor/acceptor bases, such as GT/AG. While non-canonical splicing sites have been observed, they are relatively rare and often suggestive of false positives.</p>
<div align="center">
<img src="figure/splice_site-1.png" title="plot of chunk splice_site" alt="plot of chunk splice_site" width="400px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 2.</strong> The total number of reads gap-aligned reads and the number of gap-aligned with non-canonical splicing sites are fitted to linear models as in <strong>Figure 1</strong>. On average of all samples in this data set, 1.128% of all gap-aligned reads have non-canonical splicing sites.
</div>
</div>
<div id="mismatches-and-indels" class="section level2">
<h2><span class="header-section-number">3.3</span> Mismatches and INDELs</h2>
<p>STAR alignment reports the frequency of mismatch, deletion, and insertion bases. The consistency of these statistics should also be evaluated.</p>
<div align="center">
<img src="figure/mismatch_indel-1.png" title="plot of chunk mismatch_indel" alt="plot of chunk mismatch_indel" width="600px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 3.</strong> Distribution of insertion/deletion/mismatch frequency in all samples.
</div>
</div>
<div id="unmapped-reads" class="section level2">
<h2><span class="header-section-number">3.4</span> Unmapped reads</h2>
<p>STAR alignment also reports the percent of unmapped reads due to different reasons, including too many mismatches, too short, and other. Additionally, it reports to percent of reads that were mapped, but to too many loci. Again, the focus here is the consistency of these percents between samples.</p>
<div align="center">
<img src="figure/unmapped-1.png" title="plot of chunk unmapped" alt="plot of chunk unmapped" width="800px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 4.</strong> Distribution of the frequency of poorly aligned reads due to different reasons. The frequency is relative to all mapped reads in the first plot, and relative to all unmapped reads in the others.
</div>
</div>
<div id="alerts" class="section level2">
<h2><span class="header-section-number">3.5</span> Alerts</h2>
<p>Listed below are samples with potentially quality problem, based on consistency of summary statistics between samples:</p>
<ul>
<li>Low alignment rate: M_2</li>
<li>High multiple alignment rate: none</li>
<li>High rate of noncanonical splice: T_4</li>
<li>High insertion rate: none</li>
<li>High deletion rate: none</li>
<li>High mismatch rate: none</li>
<li>High rate of <em>too many loci</em>: none</li>
<li>High rate of <em>too many mismatch</em>: none</li>
<li>High rate of <em>too short</em>: none</li>
<li>High rate of <em>other</em>: none</li>
</ul>
</div>
<div id="more-details" class="section level2">
<h2><span class="header-section-number">3.6</span> More details</h2>
<p>Click links to view full tables of summary statistics of all samples:</p>
<ul>
<li><a href="table/run_info.html">STAR run information</a>
<ul>
<li>Speed: mapping speed, million of reads per hour</li>
</ul></li>
<li>Mapped reads: <a href="table/count_mapped.html">count</a>, <a href="table/percent_mapped.html">percent</a>
<ul>
<li>Unique: reads mapped to one and only one genomic loci</li>
<li>Splice: reads mapped to splice junctions</li>
<li>Multiple: reads mapped to multiple genomic loci</li>
</ul></li>
<li><a href="table/count_splice.html">Splice sites</a></li>
<li><a href="table/mismatch_indel.html">Mismatch and INDEL</a></li>
<li><a href="table/percent_unmapped.html">Unmapped</a></li>
</ul>
<hr />
</div>
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">4</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">4.1</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p>
<ul>
<li><em>output</em> : where you want to put the output files</li>
<li><em>home</em> : the URL if you have a home page for your project</li>
<li><em>analyst</em> : your name</li>
<li><em>description</em> : background information about your project, analysis, etc.</li>
<li><em>input</em> : where are your input data, read instruction for preparing them</li>
<li><em>parameter</em> : parameters for this analysis; read instruction about how to prepare input data</li>
</ul></li>
<li><p>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</p></li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">4.2</span> Session information</h2>
<pre><code>## R version 3.2.2 (2015-08-14)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.10.5 (Yosemite)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] htmlwidgets_0.5      DT_0.1               GtUtility_0.0.0.9000
##  [4] gplots_3.0.1         awsomics_0.0.0.9000  yaml_2.1.13         
##  [7] rmarkdown_0.9.6      knitr_1.12.3         RoCA_0.0.0.9000     
## [10] RCurl_1.95-4.8       bitops_1.0-6         devtools_1.11.1     
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.4          AnnotationDbi_1.32.3 magrittr_1.5        
##  [4] IRanges_2.4.8        BiocGenerics_0.16.1  highr_0.5.1         
##  [7] stringr_1.0.0        GenomeInfoDb_1.6.3   caTools_1.17.1      
## [10] tools_3.2.2          parallel_3.2.2       Biobase_2.28.0      
## [13] KernSmooth_2.23-15   DBI_0.3.1            withr_1.0.1         
## [16] htmltools_0.3.5      gtools_3.5.0         digest_0.6.9        
## [19] formatR_1.3          S4Vectors_0.8.11     RSQLite_1.0.0       
## [22] memoise_1.0.0        evaluate_0.9         gdata_2.17.0        
## [25] stringi_1.0-1        stats4_3.2.2         jsonlite_0.9.20</code></pre>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
<p><em>END OF DOCUMENT</em></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
