<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Zhe Zhang" />

<meta name="date" content="2018-04-25" />

<title>Summarize two pairwise comparisons of differential expression</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="index_files/navigation-1.1/tabsets.js"></script>
<link href="index_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="index_files/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Summarize two pairwise comparisons of differential expression</h1>
<h4 class="author"><em>Zhe Zhang</em></h4>
<h4 class="date"><em>2018-04-25</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p><strong>Introduction</strong>: This analysis is based on the outputs of pairwise comparisons of differential gene expression generated by <a href="https://raw.githubusercontent.com/zhezhangsh/RoCA/master/template/de/pairwise_comparison/pairwise_comparison.Rmd">this</a> template. It uses results from 2 pairwise comparisons of 2 sample groups vs. their corresponding control groups and compares how these 2 sample groups are different from each other in terms of their sample-control differences (<em>delta-delta</em>). An example of such analysis is the different responses of 2 cell types to the treatment of the same drug. This analysis is focused on the overlapping of differentially expression at both gene and gene set levels.</p>
</div>
<p> </p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>Transcriptome in immune cells of control-patient samples</p>
</div>
<div id="data" class="section level2">
<h2><span class="header-section-number">1.2</span> Data</h2>
<p>Rna-seq data was generated from of 2 types of immune cells of 3 controls and 3 patients. Raw data was processed to get gene-level read counts. Pairwise comparisons were performed between controls and patients in each immune cell.</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.3</span> Analysis</h2>
<p>This is a demo.</p>
</div>
<div id="pairwise-comparisons" class="section level2">
<h2><span class="header-section-number">1.4</span> Pairwise comparisons</h2>
<p>This report compares the results of the following pairwise comparisons.</p>
<div style="color:darkblue; padding:0 1cm">
<p><strong>Table 1.</strong> Information about both comparisons, including comparison name, group names, group size, total number of genes, etc. Click links below to view details about individual comparisons.</p>
<ul>
<li><a href="table/sample.html">Samples</a></li>
<li><a href="table/gene_table.html">Gene-level results</a></li>
</ul>
</div>
<div style="padding:0 1cm">
<table>
<thead>
<tr class="header">
<th align="center">Name</th>
<th align="center">Group1</th>
<th align="center">Group2</th>
<th align="center">Num1</th>
<th align="center">Num2</th>
<th align="center">Num_Gene</th>
<th align="center">Test</th>
<th align="center">Paired</th>
<th align="center">DEG_Higher</th>
<th align="center">DEG_Lower</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">b_cell</td>
<td align="center">Control</td>
<td align="center">SLE</td>
<td align="center">3</td>
<td align="center">3</td>
<td align="center">23272</td>
<td align="center">EdgeR</td>
<td align="center">FALSE</td>
<td align="center">984</td>
<td align="center">1853</td>
</tr>
<tr class="even">
<td align="center">t_cell</td>
<td align="center">Control</td>
<td align="center">SLE</td>
<td align="center">3</td>
<td align="center">3</td>
<td align="center">23272</td>
<td align="center">EdgeR</td>
<td align="center">FALSE</td>
<td align="center">693</td>
<td align="center">671</td>
</tr>
</tbody>
</table>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="gene-level-comparison" class="section level1">
<h1><span class="header-section-number">2</span> Gene-level comparison</h1>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="global-delta-delta-correlation" class="section level2">
<h2><span class="header-section-number">2.1</span> Global delta-delta correlation</h2>
<p>Both comparisons reported the log ratio of 2 group means for each gene. The global agreement of log ratios of all genes indicates how much the results of these 2 comparisons are similar to or different from each other. Full table of gene-level statistics side-by-side is <a href="table/gene_table.html">here</a>.</p>
<div align="center">
<p><img src="figure/log_ratio-1.png" width="1800" /></p>
</div>
<div style="color:darkblue; padding:0 2cm">
<p><strong>Figure 1.</strong> This plot shows the global correlation (correlation coefficient = 0.24) between the 2 pairwise comparisons: <strong>b_cell</strong> and <strong>t_cell</strong>. Genes having p values less than 0.01 from both comparisons are highlighted.</p>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="differentially-expressed-genes-degs" class="section level2">
<h2><span class="header-section-number">2.2</span> Differentially expressed genes (DEGs)</h2>
<p>Both comparisons identified DEGs between 2 compared groups. Overlapped DEGs identified by both comparisons are worthy of a closer look.</p>
<div style="color:darkblue; padding:0 3cm">
<p><strong>Table 2.</strong> Number of DEGs reported by both comparisons. The last 2 columns list the number of DEGs identified by both comparisons using their own DEG selection criteria.</p>
</div>
<div style="padding:0 3cm">
<table>
<thead>
<tr class="header">
<th></th>
<th align="center">Total_gene</th>
<th align="center">P &lt; 0.01</th>
<th align="center">b_cell::G2 &gt; G1</th>
<th align="center">t_cell::G2 &lt; G1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>b_cell</td>
<td align="center">23272</td>
<td align="center">3235</td>
<td align="center">984</td>
<td align="center">1853</td>
</tr>
<tr class="even">
<td>t_cell</td>
<td align="center">23272</td>
<td align="center">2130</td>
<td align="center">693</td>
<td align="center">671</td>
</tr>
</tbody>
</table>
</div>
<div align="center">
<p><img src="figure/deg_overlap-1.png" width="800px" /></p>
</div>
<div style="color:darkblue; padding:0 1cm">
<p><strong>Figure 2.</strong> Overlapping of DEGs. All combinations of differential expression towards opposite directions are plotted and Fisher’s exact test is performed to evaluate the significance of overlapping or lack of overlapping. Click links below to view overlapping DEGs:</p>
<ul>
<li><a href="DEG/gene_up-up.html">b_cell, Higher_in_SLE and t_cell, Higher_in_SLE</a></li>
<li><a href="DEG/gene_up-down.html">b_cell, Higher_in_SLE and t_cell, Lower_in_SLE</a></li>
<li><a href="DEG/gene_down-up.html">b_cell, Lower_in_SLE and t_cell, Higher_in_SLE</a></li>
<li><a href="DEG/gene_down-down.html">b_cell, Lower_in_SLE and t_cell, Lower_in_SLE</a></li>
</ul>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="anova" class="section level2">
<h2><span class="header-section-number">2.3</span> ANOVA</h2>
<p>2-way ANOVA analysis is performed to identify genes responding to <strong><em>SLE</em></strong> differently in different <strong><em>Cell</em></strong>. The analysis reported 2 or 3 p values, corresponding to the effect of <strong><em>SLE</em></strong>, <strong><em>Cell</em></strong>, as well as their interaction; if required. The analysis identified 3139 significant genes with interaction p values less than 0.01. The ANOVA results are summarized in a table <a href="table/anova_table.html">here</a>.</p>
<div align="center">
<p><img src="figure/aov_top-1.png" width="2400" /></p>
</div>
<div style="color:darkblue; padding:0 1cm">
<p><strong>Figure 3.</strong> Examples: the top 4 genes having the most significant interactive p value, among the genes with significant differential expression in at least one of the two pairwise comparisons.</p>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="gene-set-over-representation-analysis-ora" class="section level2">
<h2><span class="header-section-number">2.4</span> Gene set over-representation analysis (ORA)</h2>
<p>Each 2-group comparison performs gene set over-representation analysis (ORA) that identifies gene sets over-represented with differentially expressed genes. The results of ORA of both 2-group comparisons are summarized and compared here. The ORA of each gene set reports an odds ratio and p value. These statistics from both comparisons were combined and listed side-by-side, as well as the difference of their odds ratios and ratio of their p values (p set to 0.5 when not available), in this table <a href="table/ora_table.html">here</a></p>
<div style="color:darkblue; padding:0 3cm">
<p><strong>Table 3.</strong> Gene sets were broken down into subgroups by their sources. Click on the numbers of over-represented gene sets to see a full list.</p>
</div>
<div style="padding:0 3cm">
<table>
<thead>
<tr class="header">
<th></th>
<th align="center">b_cell, Higher_in_Control</th>
<th align="center">b_cell, Higher_in_SLE</th>
<th align="center">t_cell, Higher_in_Control</th>
<th align="center">t_cell, Higher_in_SLE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BioSystems</td>
<td align="center"><a href="ORA/BioSystems_b_cell,%20Higher_in_Control.html">438</a></td>
<td align="center"><a href="ORA/BioSystems_b_cell,%20Higher_in_SLE.html">3212</a></td>
<td align="center"><a href="ORA/BioSystems_t_cell,%20Higher_in_Control.html">921</a></td>
<td align="center"><a href="ORA/BioSystems_t_cell,%20Higher_in_SLE.html">479</a></td>
</tr>
<tr class="even">
<td>KEGG</td>
<td align="center"><a href="ORA/KEGG_b_cell,%20Higher_in_Control.html">40</a></td>
<td align="center"><a href="ORA/KEGG_b_cell,%20Higher_in_SLE.html">319</a></td>
<td align="center"><a href="ORA/KEGG_t_cell,%20Higher_in_Control.html">49</a></td>
<td align="center"><a href="ORA/KEGG_t_cell,%20Higher_in_SLE.html">119</a></td>
</tr>
<tr class="odd">
<td>MSigDb</td>
<td align="center"><a href="ORA/MSigDb_b_cell,%20Higher_in_Control.html">857</a></td>
<td align="center"><a href="ORA/MSigDb_b_cell,%20Higher_in_SLE.html">4125</a></td>
<td align="center"><a href="ORA/MSigDb_t_cell,%20Higher_in_Control.html">1565</a></td>
<td align="center"><a href="ORA/MSigDb_t_cell,%20Higher_in_SLE.html">686</a></td>
</tr>
<tr class="even">
<td>OMIM</td>
<td align="center"><a href="ORA/OMIM_b_cell,%20Higher_in_Control.html">0</a></td>
<td align="center"><a href="ORA/OMIM_b_cell,%20Higher_in_SLE.html">1</a></td>
<td align="center"><a href="ORA/OMIM_t_cell,%20Higher_in_Control.html">0</a></td>
<td align="center"><a href="ORA/OMIM_t_cell,%20Higher_in_SLE.html">0</a></td>
</tr>
<tr class="odd">
<td>PubTator</td>
<td align="center"><a href="ORA/PubTator_b_cell,%20Higher_in_Control.html">123</a></td>
<td align="center"><a href="ORA/PubTator_b_cell,%20Higher_in_SLE.html">7634</a></td>
<td align="center"><a href="ORA/PubTator_t_cell,%20Higher_in_Control.html">632</a></td>
<td align="center"><a href="ORA/PubTator_t_cell,%20Higher_in_SLE.html">892</a></td>
</tr>
</tbody>
</table>
</div>
<div align="center">
<p><img src="figure/ora_overlap-1.png" width="800px" /></p>
</div>
<div style="color:darkblue; padding:0 1cm">
<p><strong>Figure 5.</strong> The overlapping of over-represented gene sets from both comparisons. Click links below to view tables of overlapping significant gene sets:</p>
<ul>
<li><a href="ORA/ora_up-up.html">b_cell, Higher_in_SLE and t_cell, Higher_in_SLE</a></li>
<li><a href="ORA/ora_up-down.html">b_cell, Higher_in_SLE and t_cell, Lower_in_SLE</a></li>
<li><a href="ORA/ora_down-up.html">b_cell, Lower_in_SLE and t_cell, Higher_in_SLE</a></li>
<li><a href="ORA/ora_down-down.html">b_cell, Lower_in_SLE and t_cell, Lower_in_SLE</a></li>
</ul>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level2">
<h2><span class="header-section-number">2.5</span> Gene set enrichment analysis (GSEA)</h2>
<p>Each 2-group comparison performs gene set enrichment analysis (GSEA) on genes ranked by their differential expression. The results of GSEA of both 2-group comparisons are summarized and compared here. The GSEA of each gene set reports an enrichment score and p value. These statistics from both comparisons were combined and listed side-by-side in this table <a href="table/gsea_table.html">here</a></p>
<div style="color:darkblue; padding:0 3cm">
<p><strong>Table 4.</strong> Gene sets were broken down into subgroups by collections. Click on the numbers of enriched gene sets to see a full list.</p>
</div>
<div style="padding:0 3cm">
<table>
<thead>
<tr class="header">
<th></th>
<th align="center">b_cell, Higher_in_Control</th>
<th align="center">b_cell, Higher_in_SLE</th>
<th align="center">t_cell, Higher_in_Control</th>
<th align="center">t_cell, Higher_in_SLE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>C0_Hallmark</td>
<td align="center"><a href="GSEA/C0_Hallmark_b_cell,%20Higher_in_Control.html">2</a></td>
<td align="center"><a href="GSEA/C0_Hallmark_b_cell,%20Higher_in_SLE.html">37</a></td>
<td align="center"><a href="GSEA/C0_Hallmark_t_cell,%20Higher_in_Control.html">12</a></td>
<td align="center"><a href="GSEA/C0_Hallmark_t_cell,%20Higher_in_SLE.html">3</a></td>
</tr>
<tr class="even">
<td>C1_Positional</td>
<td align="center"><a href="GSEA/C1_Positional_b_cell,%20Higher_in_Control.html">13</a></td>
<td align="center"><a href="GSEA/C1_Positional_b_cell,%20Higher_in_SLE.html">26</a></td>
<td align="center"><a href="GSEA/C1_Positional_t_cell,%20Higher_in_Control.html">28</a></td>
<td align="center"><a href="GSEA/C1_Positional_t_cell,%20Higher_in_SLE.html">10</a></td>
</tr>
<tr class="odd">
<td>C2_BioCarta_Pathways</td>
<td align="center"><a href="GSEA/C2_BioCarta_Pathways_b_cell,%20Higher_in_Control.html">1</a></td>
<td align="center"><a href="GSEA/C2_BioCarta_Pathways_b_cell,%20Higher_in_SLE.html">68</a></td>
<td align="center"><a href="GSEA/C2_BioCarta_Pathways_t_cell,%20Higher_in_Control.html">14</a></td>
<td align="center"><a href="GSEA/C2_BioCarta_Pathways_t_cell,%20Higher_in_SLE.html">2</a></td>
</tr>
<tr class="even">
<td>C2_Chemical_and_genetic_perturbations</td>
<td align="center"><a href="GSEA/C2_Chemical_and_genetic_perturbations_b_cell,%20Higher_in_Control.html">36</a></td>
<td align="center"><a href="GSEA/C2_Chemical_and_genetic_perturbations_b_cell,%20Higher_in_SLE.html">1356</a></td>
<td align="center"><a href="GSEA/C2_Chemical_and_genetic_perturbations_t_cell,%20Higher_in_Control.html">411</a></td>
<td align="center"><a href="GSEA/C2_Chemical_and_genetic_perturbations_t_cell,%20Higher_in_SLE.html">122</a></td>
</tr>
<tr class="odd">
<td>C3_MicroRNA_targets</td>
<td align="center"><a href="GSEA/C3_MicroRNA_targets_b_cell,%20Higher_in_Control.html">0</a></td>
<td align="center"><a href="GSEA/C3_MicroRNA_targets_b_cell,%20Higher_in_SLE.html">51</a></td>
<td align="center"><a href="GSEA/C3_MicroRNA_targets_t_cell,%20Higher_in_Control.html">3</a></td>
<td align="center"><a href="GSEA/C3_MicroRNA_targets_t_cell,%20Higher_in_SLE.html">5</a></td>
</tr>
<tr class="even">
<td>C3_TF_targets</td>
<td align="center"><a href="GSEA/C3_TF_targets_b_cell,%20Higher_in_Control.html">4</a></td>
<td align="center"><a href="GSEA/C3_TF_targets_b_cell,%20Higher_in_SLE.html">284</a></td>
<td align="center"><a href="GSEA/C3_TF_targets_t_cell,%20Higher_in_Control.html">12</a></td>
<td align="center"><a href="GSEA/C3_TF_targets_t_cell,%20Higher_in_SLE.html">89</a></td>
</tr>
<tr class="odd">
<td>C4_Cancer_gene_neighborhoods</td>
<td align="center"><a href="GSEA/C4_Cancer_gene_neighborhoods_b_cell,%20Higher_in_Control.html">42</a></td>
<td align="center"><a href="GSEA/C4_Cancer_gene_neighborhoods_b_cell,%20Higher_in_SLE.html">86</a></td>
<td align="center"><a href="GSEA/C4_Cancer_gene_neighborhoods_t_cell,%20Higher_in_Control.html">170</a></td>
<td align="center"><a href="GSEA/C4_Cancer_gene_neighborhoods_t_cell,%20Higher_in_SLE.html">17</a></td>
</tr>
<tr class="even">
<td>C4_Cancer_modules</td>
<td align="center"><a href="GSEA/C4_Cancer_modules_b_cell,%20Higher_in_Control.html">10</a></td>
<td align="center"><a href="GSEA/C4_Cancer_modules_b_cell,%20Higher_in_SLE.html">176</a></td>
<td align="center"><a href="GSEA/C4_Cancer_modules_t_cell,%20Higher_in_Control.html">78</a></td>
<td align="center"><a href="GSEA/C4_Cancer_modules_t_cell,%20Higher_in_SLE.html">18</a></td>
</tr>
<tr class="odd">
<td>C6_Oncogenic_signatures</td>
<td align="center"><a href="GSEA/C6_Oncogenic_signatures_b_cell,%20Higher_in_Control.html">2</a></td>
<td align="center"><a href="GSEA/C6_Oncogenic_signatures_b_cell,%20Higher_in_SLE.html">116</a></td>
<td align="center"><a href="GSEA/C6_Oncogenic_signatures_t_cell,%20Higher_in_Control.html">9</a></td>
<td align="center"><a href="GSEA/C6_Oncogenic_signatures_t_cell,%20Higher_in_SLE.html">17</a></td>
</tr>
<tr class="even">
<td>C7_Immunologic_signatures</td>
<td align="center"><a href="GSEA/C7_Immunologic_signatures_b_cell,%20Higher_in_Control.html">58</a></td>
<td align="center"><a href="GSEA/C7_Immunologic_signatures_b_cell,%20Higher_in_SLE.html">922</a></td>
<td align="center"><a href="GSEA/C7_Immunologic_signatures_t_cell,%20Higher_in_Control.html">432</a></td>
<td align="center"><a href="GSEA/C7_Immunologic_signatures_t_cell,%20Higher_in_SLE.html">52</a></td>
</tr>
<tr class="odd">
<td>GO_BP</td>
<td align="center"><a href="GSEA/GO_BP_b_cell,%20Higher_in_Control.html">145</a></td>
<td align="center"><a href="GSEA/GO_BP_b_cell,%20Higher_in_SLE.html">2065</a></td>
<td align="center"><a href="GSEA/GO_BP_t_cell,%20Higher_in_Control.html">437</a></td>
<td align="center"><a href="GSEA/GO_BP_t_cell,%20Higher_in_SLE.html">239</a></td>
</tr>
<tr class="even">
<td>GO_CC</td>
<td align="center"><a href="GSEA/GO_CC_b_cell,%20Higher_in_Control.html">67</a></td>
<td align="center"><a href="GSEA/GO_CC_b_cell,%20Higher_in_SLE.html">159</a></td>
<td align="center"><a href="GSEA/GO_CC_t_cell,%20Higher_in_Control.html">120</a></td>
<td align="center"><a href="GSEA/GO_CC_t_cell,%20Higher_in_SLE.html">34</a></td>
</tr>
<tr class="odd">
<td>GO_MF</td>
<td align="center"><a href="GSEA/GO_MF_b_cell,%20Higher_in_Control.html">44</a></td>
<td align="center"><a href="GSEA/GO_MF_b_cell,%20Higher_in_SLE.html">359</a></td>
<td align="center"><a href="GSEA/GO_MF_t_cell,%20Higher_in_Control.html">87</a></td>
<td align="center"><a href="GSEA/GO_MF_t_cell,%20Higher_in_SLE.html">71</a></td>
</tr>
<tr class="even">
<td>KEGG_compound</td>
<td align="center"><a href="GSEA/KEGG_compound_b_cell,%20Higher_in_Control.html">4</a></td>
<td align="center"><a href="GSEA/KEGG_compound_b_cell,%20Higher_in_SLE.html">126</a></td>
<td align="center"><a href="GSEA/KEGG_compound_t_cell,%20Higher_in_Control.html">41</a></td>
<td align="center"><a href="GSEA/KEGG_compound_t_cell,%20Higher_in_SLE.html">47</a></td>
</tr>
<tr class="odd">
<td>KEGG_enzyme</td>
<td align="center"><a href="GSEA/KEGG_enzyme_b_cell,%20Higher_in_Control.html">1</a></td>
<td align="center"><a href="GSEA/KEGG_enzyme_b_cell,%20Higher_in_SLE.html">1</a></td>
<td align="center"><a href="GSEA/KEGG_enzyme_t_cell,%20Higher_in_Control.html">2</a></td>
<td align="center"><a href="GSEA/KEGG_enzyme_t_cell,%20Higher_in_SLE.html">3</a></td>
</tr>
<tr class="even">
<td>KEGG_module</td>
<td align="center"><a href="GSEA/KEGG_module_b_cell,%20Higher_in_Control.html">11</a></td>
<td align="center"><a href="GSEA/KEGG_module_b_cell,%20Higher_in_SLE.html">13</a></td>
<td align="center"><a href="GSEA/KEGG_module_t_cell,%20Higher_in_Control.html">24</a></td>
<td align="center"><a href="GSEA/KEGG_module_t_cell,%20Higher_in_SLE.html">3</a></td>
</tr>
<tr class="odd">
<td>KEGG_pathway</td>
<td align="center"><a href="GSEA/KEGG_pathway_b_cell,%20Higher_in_Control.html">9</a></td>
<td align="center"><a href="GSEA/KEGG_pathway_b_cell,%20Higher_in_SLE.html">161</a></td>
<td align="center"><a href="GSEA/KEGG_pathway_t_cell,%20Higher_in_Control.html">27</a></td>
<td align="center"><a href="GSEA/KEGG_pathway_t_cell,%20Higher_in_SLE.html">25</a></td>
</tr>
<tr class="even">
<td>KEGG_reaction</td>
<td align="center"><a href="GSEA/KEGG_reaction_b_cell,%20Higher_in_Control.html">2</a></td>
<td align="center"><a href="GSEA/KEGG_reaction_b_cell,%20Higher_in_SLE.html">35</a></td>
<td align="center"><a href="GSEA/KEGG_reaction_t_cell,%20Higher_in_Control.html">24</a></td>
<td align="center"><a href="GSEA/KEGG_reaction_t_cell,%20Higher_in_SLE.html">23</a></td>
</tr>
<tr class="odd">
<td>OMIM_gene</td>
<td align="center"><a href="GSEA/OMIM_gene_b_cell,%20Higher_in_Control.html">1</a></td>
<td align="center"><a href="GSEA/OMIM_gene_b_cell,%20Higher_in_SLE.html">2</a></td>
<td align="center"><a href="GSEA/OMIM_gene_t_cell,%20Higher_in_Control.html">2</a></td>
<td align="center"><a href="GSEA/OMIM_gene_t_cell,%20Higher_in_SLE.html">2</a></td>
</tr>
<tr class="even">
<td>REACTOME</td>
<td align="center"><a href="GSEA/REACTOME_b_cell,%20Higher_in_Control.html">92</a></td>
<td align="center"><a href="GSEA/REACTOME_b_cell,%20Higher_in_SLE.html">283</a></td>
<td align="center"><a href="GSEA/REACTOME_t_cell,%20Higher_in_Control.html">230</a></td>
<td align="center"><a href="GSEA/REACTOME_t_cell,%20Higher_in_SLE.html">63</a></td>
</tr>
<tr class="odd">
<td>WikiPathways</td>
<td align="center"><a href="GSEA/WikiPathways_b_cell,%20Higher_in_Control.html">2</a></td>
<td align="center"><a href="GSEA/WikiPathways_b_cell,%20Higher_in_SLE.html">91</a></td>
<td align="center"><a href="GSEA/WikiPathways_t_cell,%20Higher_in_Control.html">13</a></td>
<td align="center"><a href="GSEA/WikiPathways_t_cell,%20Higher_in_SLE.html">5</a></td>
</tr>
</tbody>
</table>
</div>
<div align="center">
<p><img src="figure/nes-1.png" width="640px" /></p>
</div>
<div style="color:darkblue; padding:0 2cm">
<p><strong>Figure 6.</strong> Nominal enrichment scores from both comparisons. Each dot represents a gene set. Gene sets with p values less than 0.01 from both comparisons are highlighted.</p>
</div>
<div align="center">
<p><img src="figure/gsea_overlap-1.png" width="800px" /></p>
</div>
<div style="color:darkblue; padding:0 1cm">
<p><strong>Figure 7.</strong> The overlapping of over-represented gene sets from both comparisons. Click links to view tables of overlapping significant gene sets from GSEA:</p>
<ul>
<li><a href="GSEA/gsea_up-up.html">b_cell, Higher_in_SLE and t_cell, Higher_in_SLE</a></li>
<li><a href="GSEA/gsea_up-down.html">b_cell, Higher_in_SLE and t_cell, Lower_in_SLE</a></li>
<li><a href="GSEA/gsea_down-up.html">b_cell, Lower_in_SLE and t_cell, Higher_in_SLE</a></li>
<li><a href="GSEA/gsea_down-down.html">b_cell, Lower_in_SLE and t_cell, Lower_in_SLE</a></li>
</ul>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="gene-clustering" class="section level2">
<h2><span class="header-section-number">2.6</span> Gene clustering</h2>
<p>The top 1000 genes with significant ANOVA p values (p &lt;= 0.01) were used as seeds to perform a gene-gene clustering analysis and 6 clusters were identified. ORA was performed on the clusters to identify their functional association (see table below);</p>
<div style="color:darkblue; padding:0 2cm">
<strong>Table 5.</strong> This table lists the number of genes in each cluster (click the numbers to see gene lists), the average expression of all genes in a cluster of all sample groups, and then the gene sets over-represented in each cluster (click the numbers to see gene set lists). The gene expression levels were normalized so the mean of the control groups equals to 0 and the mean of the treatment groups is the number of standard deviations.
</div>
<div style="padding:0 2cm">
<table>
<thead>
<tr class="header">
<th align="center">ID</th>
<th align="center">Size</th>
<th align="center">b_cell, Control</th>
<th align="center">b_cell, SLE</th>
<th align="center">t_cell, Control</th>
<th align="center">t_cell, SLE</th>
<th align="center">Gene_set</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><a href="CLUSTER/Heatmap_Cluster_1.pdf">Cluster_1</a></td>
<td align="center"><a href="CLUSTER/Cluster_1.html">671</a></td>
<td align="center">0</td>
<td align="center">1.3368</td>
<td align="center">0</td>
<td align="center">-1.3933</td>
<td align="center"><a href="CLUSTER/ORA_Cluster_1.html">974</a></td>
</tr>
<tr class="even">
<td align="center"><a href="CLUSTER/Heatmap_Cluster_2.pdf">Cluster_2</a></td>
<td align="center"><a href="CLUSTER/Cluster_2.html">118</a></td>
<td align="center">0</td>
<td align="center">1.3693</td>
<td align="center">0</td>
<td align="center">0.0033</td>
<td align="center"><a href="CLUSTER/ORA_Cluster_2.html">2487</a></td>
</tr>
<tr class="odd">
<td align="center"><a href="CLUSTER/Heatmap_Cluster_3.pdf">Cluster_3</a></td>
<td align="center"><a href="CLUSTER/Cluster_3.html">1000</a></td>
<td align="center">0</td>
<td align="center">1.6649</td>
<td align="center">0</td>
<td align="center">1.5938</td>
<td align="center"><a href="CLUSTER/ORA_Cluster_3.html">2514</a></td>
</tr>
<tr class="even">
<td align="center"><a href="CLUSTER/Heatmap_Cluster_4.pdf">Cluster_4</a></td>
<td align="center"><a href="CLUSTER/Cluster_4.html">535</a></td>
<td align="center">0</td>
<td align="center">-1.5190</td>
<td align="center">0</td>
<td align="center">-1.5478</td>
<td align="center"><a href="CLUSTER/ORA_Cluster_4.html">2878</a></td>
</tr>
<tr class="odd">
<td align="center"><a href="CLUSTER/Heatmap_Cluster_5.pdf">Cluster_5</a></td>
<td align="center"><a href="CLUSTER/Cluster_5.html">446</a></td>
<td align="center">0</td>
<td align="center">-1.3751</td>
<td align="center">0</td>
<td align="center">1.3500</td>
<td align="center"><a href="CLUSTER/ORA_Cluster_5.html">3785</a></td>
</tr>
<tr class="even">
<td align="center"><a href="CLUSTER/Heatmap_Cluster_6.pdf">Cluster_6</a></td>
<td align="center"><a href="CLUSTER/Cluster_6.html">603</a></td>
<td align="center">0</td>
<td align="center">-1.5017</td>
<td align="center">0</td>
<td align="center">-0.7758</td>
<td align="center"><a href="CLUSTER/ORA_Cluster_6.html">10905</a></td>
</tr>
</tbody>
</table>
</div>
<div align="center">
<p><img src="figure/clustering_heatmap-1.png" width="640px" /></p>
</div>
<div style="color:darkblue; padding:0 2cm">
<p><strong>Figure 8.</strong> This plot shows below the average expression levels of each cluster. Data was normalized before the analysis, so the mean of the control groups was zero and the standard deviation of all samples of each gene was 1.0. Values indicate number of standard deviation from mean of relative control group.</p>
</div>
<div align="center">
<p><img src="figure/clustering_mean-1.png" width="800px" /></p>
</div>
<div style="color:darkblue; padding:0 1cm">
<p><strong>Figure 9.</strong> This plot summarizes the group means and standard errors of all clusters.</p>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
</div>
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">3</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">3.1</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p></li>
</ol>
<pre><code>- _output_        : where you want to put the output files
- _home_          : the URL if you have a home page for your project
- _analyst_       : your name
- _description_   : background information about your project, analysis, etc.
- _input_         : where are your input data, read instruction for preparing them
- _parameter_     : parameters for this analysis; read instruction about how to prepare input data</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">3.2</span> Session information</h2>
<pre><code>## R version 3.3.3 (2017-03-06)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X Yosemite 10.10.5
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] DEGandMore_0.0.0.9000 snow_0.4-2            rchive_0.0.0.9000    
##  [4] gplots_3.0.1          MASS_7.3-45           htmlwidgets_0.9      
##  [7] DT_0.2                awsomics_0.0.0.9000   yaml_2.1.16          
## [10] rmarkdown_1.9.8       knitr_1.18            RoCA_0.0.0.9000      
## [13] RCurl_1.95-4.9        bitops_1.0-6          devtools_1.13.4      
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.14       magrittr_1.5       highr_0.6         
##  [4] stringr_1.2.0      caTools_1.17.1     tools_3.3.3       
##  [7] parallel_3.3.3     KernSmooth_2.23-15 withr_2.1.1       
## [10] htmltools_0.3.6    gtools_3.5.0       rprojroot_1.3-2   
## [13] digest_0.6.13      memoise_1.1.0      evaluate_0.10.1   
## [16] gdata_2.18.0       stringi_1.1.6      backports_1.1.2   
## [19] jsonlite_1.5</code></pre>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
<p><strong>END OF DOCUMENT</strong></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
