<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Zhe Zhang" />

<meta name="date" content="2016-06-30" />

<title>Meta-analysis of differential gene expression</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="index_files/highlight/default.css"
      type="text/css" />
<script src="index_files/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="index_files/navigation-1.0/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Meta-analysis of differential gene expression</h1>
<h4 class="author"><em>Zhe Zhang</em></h4>
<h4 class="date"><em>2016-06-30</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p><strong>Introduction</strong>: This analysis is the summary of a set of pairwise comparisons. It performs meta-analysis on the results of differential gene expression generated by <a href="https://raw.githubusercontent.com/zhezhangsh/RoCA/master/template/de/pairwise_comparison/pairwise_comparison.Rmd">this</a> template. Both gene and gene set-level results of individual comparisons are summarized through visualization and <a href="https://en.wikipedia.org/wiki/Fisher%27s_method">Fisher’s method</a> of p value meta-analysis.</p>
</div>
<p> </p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>Transcriptome in immune cells of control-patient samples</p>
</div>
<div id="data" class="section level2">
<h2><span class="header-section-number">1.2</span> Data</h2>
<p>Rna-seq data was generated from of 3 types of immune cells of 3 controls and 3 patients. Raw data was processed to get gene-level read counts. Pairwise comparisons were performed between controls and patients in each immune cell.</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.3</span> Analysis</h2>
<p>This is a demo.</p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="individual-comparisons" class="section level2">
<h2><span class="header-section-number">1.4</span> Individual comparisons</h2>
<p>This report performs meta-analysis on the results of the following pairwise comparisons.</p>
<div style="color:darkblue; padding:0 1cm">
<strong>Table 1.</strong> Information about individual comparisons, including comparison name, group names, group size, total number of genes, etc.
</div>
<div align="center" ,="" style="padding:0 1cm">
<table>
<thead>
<tr class="header">
<th align="center">Name</th>
<th align="center">Group1</th>
<th align="center">Group2</th>
<th align="center">Num1</th>
<th align="center">Num2</th>
<th align="center">Num_Gene</th>
<th align="center">Test</th>
<th align="center">Paired</th>
<th align="center">DEG_Higher</th>
<th align="center">DEG_Lower</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">B_Cell</td>
<td align="center">Control</td>
<td align="center">SLE</td>
<td align="center">3</td>
<td align="center">3</td>
<td align="center">23272</td>
<td align="center">EdgeR</td>
<td align="center">FALSE</td>
<td align="center">984</td>
<td align="center">1853</td>
</tr>
<tr class="even">
<td align="center">T_Cell</td>
<td align="center">Control</td>
<td align="center">SLE</td>
<td align="center">3</td>
<td align="center">3</td>
<td align="center">23272</td>
<td align="center">EdgeR</td>
<td align="center">FALSE</td>
<td align="center">693</td>
<td align="center">671</td>
</tr>
<tr class="odd">
<td align="center">Monocyte</td>
<td align="center">Control</td>
<td align="center">SLE</td>
<td align="center">3</td>
<td align="center">3</td>
<td align="center">23272</td>
<td align="center">EdgeR</td>
<td align="center">FALSE</td>
<td align="center">501</td>
<td align="center">465</td>
</tr>
</tbody>
</table>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="gene-level-analysis" class="section level1">
<h1><span class="header-section-number">2</span> Gene-level analysis</h1>
<p>Gene-level statistics, log2 of fold change and p value, are retrieved from all individual comparisons. Genes presented in any of the comparisons are included. <a href="https://en.wikipedia.org/wiki/Fisher%27s_method">Fisher’s method</a> of meta-analysis is performed on the p values of individual comparisons to get a combined p value for each gene. Other summary statistics of each gene include its maximum p value from all comparisons, and the average/minimum of <em>|log2 of fold change|</em> from all comparisons. The joint of p value and fold change, <em>sign(log2FC)<em>sqrt(abs(log10(p)</em>logFC))</em>, is used as a representative statistic of the differential expression of each gene in each comparisons.</p>
<div align="center">
<img src="figure/gene_cluster-1.png" title="plot of chunk gene_cluster" alt="plot of chunk gene_cluster" width="640px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 1.</strong> Clustering of all the pairwise comparisons. The joint of p value and fold change, <em>sign(log2FC)<em>sqrt(abs(log10(p)</em>logFC))</em>, is used as a representative statistic of the differential expression of each gene in each comparisons. The representative statistics of all genes are then used for the clustering.<br />
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="top-genes" class="section level2">
<h2><span class="header-section-number">2.1</span> Top genes</h2>
<p>Selection of the top genes with the most significant meta-analysis results.</p>
<div align="center">
<img src="figure/gene_top_mean-1.png" title="plot of chunk gene_top_mean" alt="plot of chunk gene_top_mean" width="350px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 2A.</strong> The <a href="table/gene_top_mean.html">top genes</a> with the most overall differential expression across all comparisons, based on their representative statistics.
</div>
<p> </p>
<div align="center">
<img src="figure/gene_top_diff-1.png" title="plot of chunk gene_top_diff" alt="plot of chunk gene_top_diff" width="350px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 2B.</strong> The <a href="table/gene_top_diff.html">top genes</a> with the most different results of differential expression between the comparisons, based on their representative statistics.
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="gene-set-level-analysis" class="section level1">
<h1><span class="header-section-number">3</span> Gene set-level analysis</h1>
<p>Statisitcs of gene set enrichment analysis, p value and normalized enriched score (NES), are retrieved from individual comparisons. Gene sets presented in any of the comparisons are included. <a href="https://en.wikipedia.org/wiki/Fisher%27s_method">Fisher’s method</a> of meta-analysis is performed on the GSEA p values of individual comparisons to get a combined p value for each gene set. For the meta-analysis, p values of 0 from the original GSEA are rounded to 0.001. Other summary statistics of each gene set include its maximum p value from all comparisons, and the average/minimum of <em>|NES|</em> from all comparisons.</p>
<div align="center">
<img src="figure/gsea_cluster-1.png" title="plot of chunk gsea_cluster" alt="plot of chunk gsea_cluster" width="640px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 3.</strong> Clustering of all the pairwise comparisons, using NES of all gene sets.
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="top-gene-sets" class="section level2">
<h2><span class="header-section-number">3.1</span> Top gene sets</h2>
<p>Selection of the top gene sets with the most significant meta-analysis results.</p>
<div align="center">
<img src="figure/gsea_top_nes-1.png" title="plot of chunk gsea_top_nes" alt="plot of chunk gsea_top_nes" width="600px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 4A.</strong> The <a href="table/gsea_top_nes.html">top gene sets</a> with the highest average <em>|NES|</em> of all comparisons.
</div>
<p> </p>
<div align="center">
<img src="figure/gsea_top_diff-1.png" title="plot of chunk gsea_top_diff" alt="plot of chunk gsea_top_diff" width="600px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 4B.</strong> The <a href="table/gsea_top_diff.html">top gene sets</a> with the most different NES values among all comparisons.
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="gene-set-collections" class="section level2">
<h2><span class="header-section-number">3.2</span> Gene set collections</h2>
<p>Gene sets are split into different collections. Click on the collection names to view a plot of NES values of top gene sets and the numbers to see a full table of all gene sets:</p>
<ul>
<li><a href="figure/collection_C0_Hallmark.png">C0_Hallmark</a> : <a href="table/collection_C0_Hallmark.html">50 gene sets</a></li>
<li><a href="figure/collection_C1_Positional.png">C1_Positional</a> : <a href="table/collection_C1_Positional.html">312 gene sets</a></li>
<li><a href="figure/collection_C2_BioCarta_Pathways.png">C2_BioCarta_Pathways</a> : <a href="table/collection_C2_BioCarta_Pathways.html">217 gene sets</a></li>
<li><a href="figure/collection_C2_Chemical_and_genetic_perturbations.png">C2_Chemical_and_genetic_perturbations</a> : <a href="table/collection_C2_Chemical_and_genetic_perturbations.html">3252 gene sets</a></li>
<li><a href="figure/collection_C3_MicroRNA_targets.png">C3_MicroRNA_targets</a> : <a href="table/collection_C3_MicroRNA_targets.html">213 gene sets</a></li>
<li><a href="figure/collection_C3_TF_targets.png">C3_TF_targets</a> : <a href="table/collection_C3_TF_targets.html">573 gene sets</a></li>
<li><a href="figure/collection_C4_Cancer_gene_neighborhoods.png">C4_Cancer_gene_neighborhoods</a> : <a href="table/collection_C4_Cancer_gene_neighborhoods.html">427 gene sets</a></li>
<li><a href="figure/collection_C4_Cancer_modules.png">C4_Cancer_modules</a> : <a href="table/collection_C4_Cancer_modules.html">415 gene sets</a></li>
<li><a href="figure/collection_C6_Oncogenic_signatures.png">C6_Oncogenic_signatures</a> : <a href="table/collection_C6_Oncogenic_signatures.html">189 gene sets</a></li>
<li><a href="figure/collection_C7_Immunologic_signatures.png">C7_Immunologic_signatures</a> : <a href="table/collection_C7_Immunologic_signatures.html">1910 gene sets</a></li>
<li><a href="figure/collection_GO_BP.png">GO_BP</a> : <a href="table/collection_GO_BP.html">7131 gene sets</a></li>
<li><a href="figure/collection_GO_CC.png">GO_CC</a> : <a href="table/collection_GO_CC.html">875 gene sets</a></li>
<li><a href="figure/collection_GO_MF.png">GO_MF</a> : <a href="table/collection_GO_MF.html">1541 gene sets</a></li>
<li><a href="figure/collection_KEGG_compound.png">KEGG_compound</a> : <a href="table/collection_KEGG_compound.html">526 gene sets</a></li>
<li><a href="figure/collection_KEGG_enzyme.png">KEGG_enzyme</a> : <a href="table/collection_KEGG_enzyme.html">33 gene sets</a></li>
<li><a href="figure/collection_KEGG_module.png">KEGG_module</a> : <a href="table/collection_KEGG_module.html">128 gene sets</a></li>
<li><a href="figure/collection_KEGG_pathway.png">KEGG_pathway</a> : <a href="table/collection_KEGG_pathway.html">285 gene sets</a></li>
<li><a href="figure/collection_KEGG_reaction.png">KEGG_reaction</a> : <a href="table/collection_KEGG_reaction.html">193 gene sets</a></li>
<li><a href="figure/collection_OMIM_gene.png">OMIM_gene</a> : <a href="table/collection_OMIM_gene.html">101 gene sets</a></li>
<li><a href="figure/collection_REACTOME.png">REACTOME</a> : <a href="table/collection_REACTOME.html">1372 gene sets</a></li>
<li><a href="figure/collection_WikiPathways.png">WikiPathways</a> : <a href="table/collection_WikiPathways.html">216 gene sets</a></li>
</ul>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
</div>
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">4</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">4.1</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p>
<ul>
<li><em>output</em> : where you want to put the output files</li>
<li><em>home</em> : the URL if you have a home page for your project</li>
<li><em>analyst</em> : your name</li>
<li><em>description</em> : background information about your project, analysis, etc.</li>
<li><em>input</em> : where are your input data, read instruction for preparing them</li>
<li><em>parameter</em> : parameters for this analysis; read instruction about how to prepare input data</li>
</ul></li>
<li><p>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</p></li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">4.2</span> Session information</h2>
<pre><code>## R version 3.2.2 (2015-08-14)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.10.5 (Yosemite)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] rchive_0.0.0.9000     DEGandMore_0.0.0.9000 snow_0.4-1           
##  [4] gplots_3.0.1          MASS_7.3-45           htmlwidgets_0.6      
##  [7] DT_0.1                awsomics_0.0.0.9000   yaml_2.1.13          
## [10] rmarkdown_0.9.6       knitr_1.13            RoCA_0.0.0.9000      
## [13] RCurl_1.95-4.8        bitops_1.0-6          devtools_1.12.0      
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.5        magrittr_1.5       highr_0.6         
##  [4] stringr_1.0.0      caTools_1.17.1     tools_3.2.2       
##  [7] KernSmooth_2.23-15 withr_1.0.2        htmltools_0.3.5   
## [10] gtools_3.5.0       digest_0.6.9       formatR_1.4       
## [13] memoise_1.0.0      evaluate_0.9       gdata_2.17.0      
## [16] stringi_1.1.1      jsonlite_0.9.22</code></pre>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
<p><strong>END OF DOCUMENT</strong></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
