---
title: "Meta-analysis of differential gene expression"
author: "Zhe Zhang"
date: "2016-06-30"
output:
  html_document:
    number_sections: yes
    self_contained: no
    toc: yes
    toc_float:
      collapsed: no
---

<div style="border:black 1px solid; padding: 0.5cm 0.5cm">

**Introduction**: This analysis is the summary of a set of pairwise comparisons. It performs meta-analysis on the results of differential gene expression generated by [this](https://raw.githubusercontent.com/zhezhangsh/RoCA/master/template/de/pairwise_comparison/pairwise_comparison.Rmd) template. Both gene and gene set-level results of individual comparisons are summarized through visualization and [Fisher's method](https://en.wikipedia.org/wiki/Fisher%27s_method) of p value meta-analysis. 

</div>

&nbsp;


cp <- ImportTable(DownloadFile(yml$input$comparison, path.input), rownames = FALSE, colnames = FALSE); 

NM <- comp[[1]]; 
N  <- length(NM);

c <- as.vector(cp[, 2]); 
names(c) <- as.vector(cp[, 1]); 
res <- lapply(names(c), function(nm) {
  f <- c[[nm]]; 
  
  if (dir.exists(f)) {
    res <- ImportR(paste(f, 'R', 'result.rds', sep='/')); 
  } else {
    fn <- DownloadFile(f, path.input); 
    res <- ImportR(fn); 
  };
  
  saveRDS(res, paste(path.input, paste('result_', nm, '.rds', sep=''), sep='/')); 

  res; 
});

expr <- lapply(res, function(res) res$inputs$expr);
anno <- lapply(res, function(res) res$inputs$anno);
comp <- lapply(res, function(res) res$inputs$comparison);
de   <- lapply(res, function(res) res$de); 
ora  <- lapply(res, function(res) res$ora); 
gsea <- lapply(res, function(res) res$gsea); 
stat <- lapply(de, function(de) de$results$stat); 
deg  <- lapply(de, function(de) de$DEG); 
names(res) <- names(expr) <- names(anno) <- names(comp) <- names(de) <- NM;
names(ora) <- names(gsea) <- names(stat) <- names(deg) <- NM; 
```

<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div> 

# Description
  
## Project


Transcriptome in immune cells of control-patient samples


## Data


Rna-seq data was generated from of 3 types of immune cells of 3 controls and 3 patients. Raw data was processed to get gene-level read counts. Pairwise comparisons were performed between controls and patients in each immune cell.


## Analysis


This is a demo.



<div align='right'>_[Go to project home](http://zhezhangsh.github.io/RoCA)_</div> 

## Individual comparisons

This report performs meta-analysis on the results of the following pairwise comparisons.  































