<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jim Zhang" />

<meta name="date" content="2019-09-09" />

<title>Frequency of a DNA motif in a set of sequences</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="index_files/navigation-1.1/tabsets.js"></script>
<link href="index_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="index_files/highlightjs-9.12.0/highlight.js"></script>
<script src="index_files/kePrint-0.0.1/kePrint.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Frequency of a DNA motif in a set of sequences</h1>
<h4 class="author"><em>Jim Zhang</em></h4>
<h4 class="date"><em>2019-09-09</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p>This analysis reports of the frequency and location of a DNA motif in a given set of sequences, and compares the observed frequency to expected frequency.</p>
<ul>
<li>The motif is searched in each sequence to report its occurances location in the sequence.</li>
<li>For each sequence, the observed frequency is compared to expected frequency, which is calculated based on overall base frequency of the sequence.</li>
<li>Over- or under-representation of the motif in each sequence is calculated as the ratio of its observed frequency over expected frequency, and tested by Poisson test.</li>
<li>Overall over- or under-representation of the motif in all sequences is summarized.</li>
<li>Overall location of motif within all sequences is summarized.</li>
</ul>
</div>
<p> </p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>RoCA for motif frequency in a set of DNA sequences</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.2</span> Analysis</h2>
<p>Search for GATA1 motif (GATA) around TSS of a few random genes.</p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="results" class="section level1">
<h1><span class="header-section-number">2</span> Results</h1>
<div id="dna-sequences" class="section level2">
<h2><span class="header-section-number">2.1</span> DNA sequences</h2>
<p><strong>Table 1.</strong> Length and GC content of <strong>81</strong> DNA sequences.</p>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:center;">
Min.
</th>
<th style="text-align:center;">
1st Qu.
</th>
<th style="text-align:center;">
Median
</th>
<th style="text-align:center;">
Mean
</th>
<th style="text-align:center;">
3rd Qu.
</th>
<th style="text-align:center;">
Max.
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Length (bp)
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
</tr>
<tr>
<td style="text-align:left;">
GC (%)
</td>
<td style="text-align:center;">
25.87
</td>
<td style="text-align:center;">
46.27
</td>
<td style="text-align:center;">
56.22
</td>
<td style="text-align:center;">
56.88
</td>
<td style="text-align:center;">
69.65
</td>
<td style="text-align:center;">
86.57
</td>
</tr>
</tbody>
</table>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
<div id="motif-frequency-in-sequences" class="section level2">
<h2><span class="header-section-number">2.2</span> Motif frequency in sequences</h2>
<p><img src="index_files/figure-html/plot_freq-1.png" width="640px" /></p>
<p><strong>Figure 1.</strong> Expected vs. observed frequency of motif <strong>AAAG</strong> in sequences. Each diamond represents one sequence and its expected frequency of the motif is calculated according to its base frequency. Both of the original and reverse-complemented sequence of the motif were counted Difference between expected and observed frequencies was tested by Poisson test and the sequences with p value less than 0.05 are highlighted.</p>
<blockquote>
<p><a href="table/stat_sequence.html">Details about individual sequences</a></p>
</blockquote>
</div>
<div id="motif-location-in-sequences" class="section level2">
<h2><span class="header-section-number">2.3</span> Motif location in sequences</h2>
<p><img src="index_files/figure-html/plot_position-1.png" width="640px" /></p>
<p><strong>Figure 2.</strong> Distribution of matches to motif within the sequences, all of them combined. Sequence length was adjusted so the positions are relative.</p>
</div>
<div id="summary" class="section level2">
<h2><span class="header-section-number">2.4</span> Summary</h2>
<ul>
<li>Motif to search: <strong>AAAG</strong></li>
<li>Search reverse-complement motif: <strong>TRUE</strong></li>
<li>Total number of sequences: <strong>81</strong></li>
<li>Total number of bases: <strong>16,281</strong></li>
<li>Total number of matches (<em>expected</em>): <strong>110.65</strong></li>
<li>Total number of matches (<em>observed</em>): <strong>181</strong></li>
<li>Observed/expected: <strong>1.64</strong> (p=9.43e-10, Exact Poisson test)</li>
<li>Number of over-representing sequences: <strong>54</strong></li>
<li>Number of over-representing and significant sequences (<em>p&lt;=0.05</em>): <strong>8</strong></li>
<li>Matches near the center (<em>0.25 to 0.75</em>): <strong>95</strong> (p=9.43e-10, uniform test)</li>
</ul>
<hr />
</div>
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">3</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">3.1</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p>
<ul>
<li><em>output</em> : where you want to put the output files</li>
<li><em>home</em> : the URL if you have a home page for your project</li>
<li><em>analyst</em> : your name</li>
<li><em>description</em> : background information about your project, analysis, etc.</li>
<li><em>input</em> : where are your input data, read instruction for preparing them</li>
<li><em>parameter</em> : parameters for this analysis; read instruction about how to prepare input data</li>
</ul></li>
<li><p>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</p></li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">3.2</span> Session information</h2>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      stats4    parallel  stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] gplots_3.0.1                          
##  [2] arrangements_1.1.5                    
##  [3] combinat_0.0-8                        
##  [4] seqinr_3.4-5                          
##  [5] e1071_1.7-2                           
##  [6] BSgenome.Hsapiens.NCBI.GRCh38_1.3.1000
##  [7] Agri_0.0.0.9000                       
##  [8] rGADEM_2.30.0                         
##  [9] seqLogo_1.48.0                        
## [10] BSgenome_1.50.0                       
## [11] rtracklayer_1.42.1                    
## [12] Biostrings_2.50.1                     
## [13] XVector_0.22.0                        
## [14] GenomicFeatures_1.34.1                
## [15] AnnotationDbi_1.44.0                  
## [16] Biobase_2.42.0                        
## [17] GenomicRanges_1.34.0                  
## [18] GenomeInfoDb_1.18.1                   
## [19] IRanges_2.16.0                        
## [20] S4Vectors_0.20.1                      
## [21] BiocGenerics_0.28.0                   
## [22] htmlwidgets_1.3                       
## [23] DT_0.5                                
## [24] kableExtra_0.9.0                      
## [25] awsomics_0.0.0.9000                   
## [26] yaml_2.2.0                            
## [27] rmarkdown_1.10                        
## [28] knitr_1.20                            
## [29] RoCA_0.0.0.9000                       
## [30] RCurl_1.95-4.11                       
## [31] bitops_1.0-6                          
## [32] usethis_1.4.0                         
## [33] devtools_2.0.1                        
## 
## loaded via a namespace (and not attached):
##  [1] matrixStats_0.54.0          fs_1.2.6                   
##  [3] bit64_0.9-7                 progress_1.2.0             
##  [5] httr_1.3.1                  rprojroot_1.3-2            
##  [7] tools_3.5.1                 backports_1.1.2            
##  [9] R6_2.3.0                    KernSmooth_2.23-15         
## [11] DBI_1.0.0                   colorspace_1.3-2           
## [13] ade4_1.7-13                 withr_2.1.2                
## [15] prettyunits_1.0.2           processx_3.2.0             
## [17] bit_1.1-14                  compiler_3.5.1             
## [19] cli_1.0.1                   rvest_0.3.2                
## [21] xml2_1.2.0                  desc_1.2.0                 
## [23] DelayedArray_0.8.0          caTools_1.17.1.1           
## [25] scales_1.0.0                readr_1.3.1                
## [27] callr_3.0.0                 stringr_1.3.1              
## [29] digest_0.6.18               Rsamtools_1.34.0           
## [31] base64enc_0.1-3             pkgconfig_2.0.2            
## [33] htmltools_0.3.6             sessioninfo_1.1.1          
## [35] highr_0.7                   rlang_0.4.0                
## [37] rstudioapi_0.8              RSQLite_2.1.1              
## [39] shiny_1.2.0                 jsonlite_1.5               
## [41] gtools_3.8.1                crosstalk_1.0.0            
## [43] BiocParallel_1.16.2         magrittr_1.5               
## [45] GenomeInfoDbData_1.2.0      Matrix_1.2-15              
## [47] Rcpp_1.0.0                  munsell_0.5.0              
## [49] stringi_1.2.4               MASS_7.3-51.1              
## [51] SummarizedExperiment_1.12.0 zlibbioc_1.28.0            
## [53] pkgbuild_1.0.2              blob_1.1.1                 
## [55] gdata_2.18.0                promises_1.0.1             
## [57] crayon_1.3.4                lattice_0.20-38            
## [59] hms_0.4.2                   ps_1.2.1                   
## [61] pillar_1.3.0                biomaRt_2.38.0             
## [63] pkgload_1.0.2               XML_3.98-1.16              
## [65] glue_1.3.0                  evaluate_0.12              
## [67] remotes_2.0.2               httpuv_1.4.5               
## [69] testthat_2.2.1              assertthat_0.2.0           
## [71] mime_0.6                    xtable_1.8-3               
## [73] later_0.7.5                 class_7.3-14               
## [75] viridisLite_0.3.0           tibble_1.4.2               
## [77] GenomicAlignments_1.18.0    memoise_1.1.0              
## [79] gmp_0.5-13.5</code></pre>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<hr />
<p><em>END OF DOCUMENT</em></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
