<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jim Zhang" />

<meta name="date" content="2016-06-07" />

<title>Summary of FastQC reports</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="index_files/highlight/default.css"
      type="text/css" />
<script src="index_files/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="index_files/navigation-1.0/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Summary of FastQC reports</h1>
<h4 class="author"><em>Jim Zhang</em></h4>
<h4 class="date"><em>2016-06-07</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p><strong>Introduction</strong> This analysis compares a set of high-throughput sequencing libraries based on their <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> reports. Please refer to the original FastQC documentation for details about FastQC statistics. This analysis only requires the <em>fastqc.zip</em> file generated by each FastQC run and don’t use the original .fastq or .bam file.</p>
<p><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> aims to provide a simple way to do some quality control checks on raw sequence data coming from high throughput sequencing pipelines. It provides a modular set of analyses which you can use to give a quick impression of whether your data has any problems of which you should be aware before doing any further analysis. The main functions of FastQC are</p>
<ul>
<li>Import of data from BAM, SAM or FastQ files (any variant)</li>
<li>Providing a quick overview to tell you in which areas there may be problems</li>
<li>Summary graphs and tables to quickly assess your data</li>
<li>Export of results to an HTML based permanent report</li>
<li>Offline operation to allow automated generation of reports without running the interactive application</li>
</ul>
</div>
<p> </p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
</div>
<div id="description-1" class="section level1">
<h1><span class="header-section-number">2</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">2.1</span> Project</h2>
<p>Compare the transcriptomes of early embryos of lean and obese mice.</p>
</div>
<div id="experiment" class="section level2">
<h2><span class="header-section-number">2.2</span> Experiment</h2>
<p>Prepare RNA-seq libraries using Clontech’s Low Input SMARTer® Stranded RNA-Seq Kit and did a paired-end 2x75bp NextSeq run.</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">2.3</span> Analysis</h2>
<p>Diagnosis of RNA-seq libaries with low alignment rate. The sequencing reads of this data set have low alignment rate (TopHat~=50% STAR~=65%). One pair of .fastq files, <strong>CtrlM201A2_S16_L001</strong>, with ~12 million reads were selected for diagnosis purpose. Each of the original .fastq files were split into 2 files with just aligned and unaligned reads according to STAR alignment result. This analysis compares the FastQC reports of the original, aligned, and unaligned fastq files.</p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="summary-statistics" class="section level1">
<h1><span class="header-section-number">3</span> Summary statistics</h1>
<p>Summary statistics were compared across libraries. Inconsistence of a statistics might indicate quality issues of some libraries.</p>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<div id="quick-evaluation" class="section level2">
<h2><span class="header-section-number">3.1</span> Quick evaluation</h2>
<p><strong>FastQC</strong> makes a quick evaluation on each summary statistic whether it is entirely normal (<strong>green</strong>), slightly abnormal (<strong>orange</strong>) or abnormal (<strong>red</strong>). The figure below compiles the evaluation results of all summary statistics from all libraries.</p>
<ul>
<li><strong><a href="table/quick_evaluation.html">Quick evaluation results of all libraries</a></strong></li>
</ul>
<div align="center">
<img src="figure/quick_eval_heatmap-1.png" title="plot of chunk quick_eval_heatmap" alt="plot of chunk quick_eval_heatmap" width="720px" />
</div>
<div style="color:darkblue; padding:0 2.5cm">
<strong>Figure 1.</strong> The summary statistics are color-coded (green=normal, orange=slightly abnormal, and red=abnormal). Please refer to FastQC manual for intepretation of each statistic.
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="total-reads" class="section level2">
<h2><span class="header-section-number">3.2</span> Total reads</h2>
<div align="center">
<img src="figure/total_reads-1.png" title="plot of chunk total_reads" alt="plot of chunk total_reads" width="600px" />
</div>
<div style="color:darkblue; padding:0 3cm">
<strong>Figure 2.</strong> The distribution of library size (number of total reads). One pair of paired reads is counted as one read.
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="paired-read-correlation" class="section level2">
<h2><span class="header-section-number">3.3</span> Paired read correlation</h2>
<p>Compare the consistency of several key summary statistics between paired sequence reads. Strong agreement between pairs is expected.</p>
<ul>
<li><strong><a href="table/summary_stats.html">Summary statistics to be compared between paired reads</a></strong></li>
</ul>
<div align="center">
<img src="figure/between_pair-1.png" title="plot of chunk between_pair" alt="plot of chunk between_pair" width="640px" />
</div>
<div style="color:darkblue; padding:0 2cm">
<strong>Figure 3.</strong> These plots compare the summary statistics of paired reads (R1 = read1 and R2 = read2). Each dot represents a library. No plots will be shown if the sequencing reads are single-ended.
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="per-read-statistics" class="section level2">
<h2><span class="header-section-number">3.4</span> Per read statistics</h2>
<div id="sequencing-quality" class="section level3">
<h3><span class="header-section-number">3.4.1</span> Sequencing quality</h3>
<div align="center">
<img src="figure/average_quality-1.png" title="plot of chunk average_quality" alt="plot of chunk average_quality" width="600px" />
</div>
<div style="color:darkblue; padding:0 3cm">
<strong>Figure 4.</strong> Each base in a sequencing read got a quality score. This plot shows the average quality score of all bases and all reads in each library. The scores from Illumina sequencers usually range from 0 (p=1) to 40 (p=1E-4).
</div>
</div>
<div id="gc-percent" class="section level3">
<h3><span class="header-section-number">3.4.2</span> GC percent</h3>
<div align="center">
<img src="figure/gc_percent-1.png" title="plot of chunk gc_percent" alt="plot of chunk gc_percent" width="600px" />
</div>
<div style="color:darkblue; padding:0 3cm">
<strong>Figure 5.</strong> The average GC percent of all reads in each library.
</div>
</div>
<div id="n-percent" class="section level3">
<h3><span class="header-section-number">3.4.3</span> N percent</h3>
<div align="center">
<img src="figure/n_percent-1.png" title="plot of chunk n_percent" alt="plot of chunk n_percent" width="600px" />
</div>
<div style="color:darkblue; padding:0 3cm">
<strong>Figure 6.</strong> The average N frequency of all reads in each library.
</div>
</div>
<div id="duplication-level" class="section level3">
<h3><span class="header-section-number">3.4.4</span> Duplication level</h3>
<p>Duplicated reads have the exact same sequences. Duplicated reads are often treated as one read in variant calling, so higher percentage of duplicated reads reduces the total number of usable reads. However, the duplication of paired end reads is evaluated separately by FastQC, so the actual level of duplication might be lower than the number in FastQC report.</p>
<div align="center">
<img src="figure/dup_percent-1.png" title="plot of chunk dup_percent" alt="plot of chunk dup_percent" width="600px" />
</div>
<div style="color:darkblue; padding:0 3cm">
<strong>Figure 7.</strong> The average frequency of duplicated reads in each library.
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="per-base-statistics" class="section level2">
<h2><span class="header-section-number">3.5</span> Per base statistics</h2>
<div id="sequence-quality" class="section level3">
<h3><span class="header-section-number">3.5.1</span> Sequence quality</h3>
<p>Sequence quality of single bases is often position-dependent, with later positions usually having lower quality scores.</p>
<ul>
<li><strong><a href="table/perbase_quality.html">Position-specific quality scores of all libraries</a></strong></li>
</ul>
<div align="center">
<img src="figure/perbase_quality_heatmap-1.png" title="plot of chunk perbase_quality_heatmap" alt="plot of chunk perbase_quality_heatmap" width="720px" />
</div>
<div style="color:darkblue; padding:0 2.5cm">
<strong>Figure 8.</strong> The average of position-specific quality scores in each library (red = higher).
</div>
</div>
<div id="gc-content" class="section level3">
<h3><span class="header-section-number">3.5.2</span> GC content</h3>
<p>Sequence GC content of single bases is often position-dependent too.</p>
<ul>
<li><strong><a href="table/perbase_content.html">Position-specific GC content of all libraries</a></strong></li>
</ul>
<div align="center">
<img src="figure/perbase_content_heatmap-1.png" title="plot of chunk perbase_content_heatmap" alt="plot of chunk perbase_content_heatmap" width="720px" />
</div>
<div style="color:darkblue; padding:0 2.5cm">
<strong>Figure 9.</strong> The average of position-specific GC content in each library (red = higher).
</div>
</div>
<div id="n-content" class="section level3">
<h3><span class="header-section-number">3.5.3</span> N content</h3>
<p>The occurance of Ns is also position-specific.</p>
<ul>
<li><strong><a href="table/perbase_N_content.html">Position-specific occurance of Ns in all libraries</a></strong></li>
</ul>
<div align="center">
<img src="figure/perbase_N_content_heatmap-1.png" title="plot of chunk perbase_N_content_heatmap" alt="plot of chunk perbase_N_content_heatmap" width="720px" />
</div>
<div style="color:darkblue; padding:0 2.5cm">
<strong>Figure 10.</strong> The average of position-specific GC content in each library (red = higher).
</div>
</div>
<div id="adaptor-content" class="section level3">
<h3><span class="header-section-number">3.5.4</span> Adaptor content</h3>
<p>Among commonly known adaptor sequences, <strong><em>Illumina Universal Adapter</em></strong>, had the highest occurances in the sequencing reads. So, only the occurance of this adaptor was summarized.</p>
<ul>
<li><strong><a href="table/adaptor_content.html">Occurance of adaptor sequence at different positions in reads</a></strong></li>
</ul>
<div align="center">
<img src="figure/adaptor_content_heatmap-1.png" title="plot of chunk adaptor_content_heatmap" alt="plot of chunk adaptor_content_heatmap" width="720px" />
</div>
<div style="color:darkblue; padding:0 2.5cm">
<strong>Figure 11.</strong> The position-specific percent of the occurance of adaptor sequence, <strong><em>Illumina Universal Adapter</em></strong> (red = higher).
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="k-mer-enrichment" class="section level2">
<h2><span class="header-section-number">3.6</span> K-mer enrichment</h2>
<p><strong>FastQC</strong> identifies top Kmers that were enriched in each library with their total occurance (count), significance (p value), enrichment (observed/expected), and the base position with maximum enrichment. Kmers of all libraries were summarized below. Use <strong>Table 2</strong> for details about Kmers in individual libraries.</p>
<ul>
<li><a href="table/kmer_count.html">Kmer count</a></li>
<li><a href="table/kmer_enrichment.html">Kmer max enrichment</a></li>
<li><a href="table/kmer_position.html">Kmer max positioin</a></li>
</ul>
<div style="color:darkblue; padding:0 1cm">
<strong>Table 1</strong> Top 21 Kmers that were enriched in multiple libraries.
</div>
<div align="center" ,="" style="padding:0 1cm">
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Library</th>
<th align="right">Count</th>
<th align="right">Obs/Exp</th>
<th align="right">Position_Mean</th>
<th align="right">Position_Min</th>
<th align="right">Position_Max</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TATCGCG</td>
<td align="right">3</td>
<td align="right">148.3333</td>
<td align="right">32.49667</td>
<td align="right">18.33333</td>
<td align="right">1</td>
<td align="right">53</td>
</tr>
<tr class="even">
<td align="left">AACGCTC</td>
<td align="right">2</td>
<td align="right">1737.5000</td>
<td align="right">33.31000</td>
<td align="right">3.00000</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">AAGTCGG</td>
<td align="right">2</td>
<td align="right">1712.5000</td>
<td align="right">45.03500</td>
<td align="right">1.00000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">ACCCTTC</td>
<td align="right">2</td>
<td align="right">6367.5000</td>
<td align="right">55.94500</td>
<td align="right">3.00000</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">ACGTACG</td>
<td align="right">2</td>
<td align="right">452.5000</td>
<td align="right">36.99000</td>
<td align="right">67.00000</td>
<td align="right">67</td>
<td align="right">67</td>
</tr>
<tr class="even">
<td align="left">AGTCGGC</td>
<td align="right">2</td>
<td align="right">1820.0000</td>
<td align="right">48.19500</td>
<td align="right">2.00000</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">CGACCAA</td>
<td align="right">2</td>
<td align="right">1257.5000</td>
<td align="right">35.43500</td>
<td align="right">23.00000</td>
<td align="right">23</td>
<td align="right">23</td>
</tr>
<tr class="even">
<td align="left">CGTTGCG</td>
<td align="right">2</td>
<td align="right">905.0000</td>
<td align="right">47.15000</td>
<td align="right">1.00000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">GACCCTT</td>
<td align="right">2</td>
<td align="right">3845.0000</td>
<td align="right">48.18500</td>
<td align="right">2.00000</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">GCACGAC</td>
<td align="right">2</td>
<td align="right">2130.0000</td>
<td align="right">45.53000</td>
<td align="right">3.00000</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">GGCACGA</td>
<td align="right">2</td>
<td align="right">1725.0000</td>
<td align="right">37.84000</td>
<td align="right">2.00000</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">GTACGCG</td>
<td align="right">2</td>
<td align="right">525.0000</td>
<td align="right">47.52500</td>
<td align="right">69.00000</td>
<td align="right">69</td>
<td align="right">69</td>
</tr>
<tr class="odd">
<td align="left">GTTGCGG</td>
<td align="right">2</td>
<td align="right">1200.0000</td>
<td align="right">44.15500</td>
<td align="right">2.00000</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">TAACCCT</td>
<td align="right">2</td>
<td align="right">1327.5000</td>
<td align="right">47.45500</td>
<td align="right">1.00000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">TACCTCG</td>
<td align="right">2</td>
<td align="right">1195.0000</td>
<td align="right">47.11500</td>
<td align="right">4.00000</td>
<td align="right">4</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">TAGCGCG</td>
<td align="right">2</td>
<td align="right">387.5000</td>
<td align="right">35.94000</td>
<td align="right">1.00000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">TCACGAC</td>
<td align="right">2</td>
<td align="right">1040.0000</td>
<td align="right">36.54500</td>
<td align="right">3.00000</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">TCGGCGC</td>
<td align="right">2</td>
<td align="right">20690.0000</td>
<td align="right">64.98000</td>
<td align="right">69.00000</td>
<td align="right">69</td>
<td align="right">69</td>
</tr>
<tr class="odd">
<td align="left">TGCGACG</td>
<td align="right">2</td>
<td align="right">617.5000</td>
<td align="right">34.27000</td>
<td align="right">2.00000</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">TTGCGAC</td>
<td align="right">2</td>
<td align="right">680.0000</td>
<td align="right">36.32500</td>
<td align="right">1.00000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">TTTCTCG</td>
<td align="right">2</td>
<td align="right">2527.5000</td>
<td align="right">47.87000</td>
<td align="right">4.00000</td>
<td align="right">4</td>
<td align="right">4</td>
</tr>
</tbody>
</table>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="fastqc-reports" class="section level1">
<h1><span class="header-section-number">4</span> FastQC reports</h1>
<div id="per-read" class="section level2">
<h2><span class="header-section-number">4.1</span> Per read</h2>
<div style="color:darkblue; padding:0 1cm">
<strong>Table 2</strong> Click links to view full reports and individual plots of per read statistics.
</div>
<div align="center" ,="" style="padding:0 1cm">
<table>
<thead>
<tr class="header">
<th align="left">Library</th>
<th align="left">Quality</th>
<th align="left">GC_percent</th>
<th align="left">Length</th>
<th align="left">Duplication</th>
<th align="left">Overrepresented</th>
<th align="left">Kmer</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/fastqc_report.html">Original_CtrlM201A2_S16_L001_R1</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_sequence_quality.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_sequence_quality_scores.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_sequence_gc_content.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_sequence_gc_content.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/sequence_length_distribution.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/sequence_length_distribution.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/duplication_levels.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/sequence_duplication_levels.html">table</a></td>
<td align="left">image;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/overrepresented_sequences.html">table</a></td>
<td align="left">image;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/kmer_content.html">table</a></td>
</tr>
<tr class="even">
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/fastqc_report.html">Original_CtrlM201A2_S16_L001_R2</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_sequence_quality.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_sequence_quality_scores.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_sequence_gc_content.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_sequence_gc_content.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/sequence_length_distribution.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/sequence_length_distribution.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/duplication_levels.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/sequence_duplication_levels.html">table</a></td>
<td align="left">image;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/overrepresented_sequences.html">table</a></td>
<td align="left">image;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/kmer_content.html">table</a></td>
</tr>
<tr class="odd">
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/fastqc_report.html">Aligned_CtrlM201A2_S16_L001_R1</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_sequence_quality.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_sequence_quality_scores.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_sequence_gc_content.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_sequence_gc_content.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/sequence_length_distribution.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/sequence_length_distribution.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/duplication_levels.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/sequence_duplication_levels.html">table</a></td>
<td align="left">image;table</td>
<td align="left">image;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/kmer_content.html">table</a></td>
</tr>
<tr class="even">
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/fastqc_report.html">Aligned_CtrlM201A2_S16_L001_R2</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_sequence_quality.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_sequence_quality_scores.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_sequence_gc_content.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_sequence_gc_content.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/sequence_length_distribution.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/sequence_length_distribution.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/duplication_levels.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/sequence_duplication_levels.html">table</a></td>
<td align="left">image;table</td>
<td align="left">image;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/kmer_content.html">table</a></td>
</tr>
<tr class="odd">
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/fastqc_report.html">Unaligned_CtrlM201A2_S16_L001_R1</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_sequence_quality.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_sequence_quality_scores.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_sequence_gc_content.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_sequence_gc_content.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/sequence_length_distribution.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/sequence_length_distribution.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/duplication_levels.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/sequence_duplication_levels.html">table</a></td>
<td align="left">image;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/overrepresented_sequences.html">table</a></td>
<td align="left">image;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/kmer_content.html">table</a></td>
</tr>
<tr class="even">
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/fastqc_report.html">Unaligned_CtrlM201A2_S16_L001_R2</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_sequence_quality.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_sequence_quality_scores.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_sequence_gc_content.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_sequence_gc_content.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/sequence_length_distribution.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/sequence_length_distribution.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/duplication_levels.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/sequence_duplication_levels.html">table</a></td>
<td align="left">image;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/overrepresented_sequences.html">table</a></td>
<td align="left">image;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/kmer_content.html">table</a></td>
</tr>
</tbody>
</table>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
</div>
<div id="per-base" class="section level2">
<h2><span class="header-section-number">4.2</span> Per base</h2>
<div style="color:darkblue; padding:0 1cm">
<strong>Table 3</strong> Click links to view full reports and individual plots of per base statistics.
</div>
<div align="center" ,="" style="padding:0 1cm">
<table>
<thead>
<tr class="header">
<th align="left">Library</th>
<th align="left">Base_quality</th>
<th align="left">Tile_quality</th>
<th align="left">GC_content</th>
<th align="left">N_content</th>
<th align="left">Kmer_profile</th>
<th align="left">Adapter_content</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/fastqc_report.html">Original_CtrlM201A2_S16_L001_R1</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_base_quality.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_base_sequence_quality.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_tile_quality.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_tile_sequence_quality.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_base_sequence_content.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_base_sequence_content.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_base_n_content.png">image</a>;table</td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/kmer_profiles.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/kmer_content.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Images/adapter_content.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R1_001_fastqc/Tables/adapter_content.html">table</a></td>
</tr>
<tr class="even">
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/fastqc_report.html">Original_CtrlM201A2_S16_L001_R2</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_base_quality.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_base_sequence_quality.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_tile_quality.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_tile_sequence_quality.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_base_sequence_content.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_base_sequence_content.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_base_n_content.png">image</a>;table</td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/kmer_profiles.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/kmer_content.html">table</a></td>
<td align="left"><a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Images/adapter_content.png">image</a>;<a href="input/CtrlM201A2_S16_L001_R2_001_fastqc/Tables/adapter_content.html">table</a></td>
</tr>
<tr class="odd">
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/fastqc_report.html">Aligned_CtrlM201A2_S16_L001_R1</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_base_quality.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_base_sequence_quality.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_tile_quality.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_tile_sequence_quality.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_base_sequence_content.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_base_sequence_content.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_base_n_content.png">image</a>;table</td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/kmer_profiles.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/kmer_content.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/adapter_content.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/adapter_content.html">table</a></td>
</tr>
<tr class="even">
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/fastqc_report.html">Aligned_CtrlM201A2_S16_L001_R2</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_base_quality.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_base_sequence_quality.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_tile_quality.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_tile_sequence_quality.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_base_sequence_content.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_base_sequence_content.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_base_n_content.png">image</a>;table</td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/kmer_profiles.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/kmer_content.html">table</a></td>
<td align="left"><a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/adapter_content.png">image</a>;<a href="input/Aligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/adapter_content.html">table</a></td>
</tr>
<tr class="odd">
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/fastqc_report.html">Unaligned_CtrlM201A2_S16_L001_R1</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_base_quality.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_base_sequence_quality.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_tile_quality.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_tile_sequence_quality.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_base_sequence_content.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/per_base_sequence_content.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/per_base_n_content.png">image</a>;table</td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/kmer_profiles.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/kmer_content.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Images/adapter_content.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R1_001_fastqc/Tables/adapter_content.html">table</a></td>
</tr>
<tr class="even">
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/fastqc_report.html">Unaligned_CtrlM201A2_S16_L001_R2</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_base_quality.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_base_sequence_quality.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_tile_quality.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_tile_sequence_quality.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_base_sequence_content.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/per_base_sequence_content.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/per_base_n_content.png">image</a>;table</td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/kmer_profiles.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/kmer_content.html">table</a></td>
<td align="left"><a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Images/adapter_content.png">image</a>;<a href="input/Unaligned_CtrlM201A2_S16_L001_R2_001_fastqc/Tables/adapter_content.html">table</a></td>
</tr>
</tbody>
</table>
</div>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
</div>
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">5</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">5.1</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p>
<ul>
<li><em>output</em> : where you want to put the output files</li>
<li><em>home</em> : the URL if you have a home page for your project</li>
<li><em>analyst</em> : your name</li>
<li><em>description</em> : background information about your project, analysis, etc.</li>
<li><em>input</em> : where are your input data, read instruction for preparing them</li>
<li><em>parameter</em> : parameters for this analysis; read instruction about how to prepare input data</li>
</ul></li>
<li><p>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</p></li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">5.2</span> Session information</h2>
<pre><code>## R version 3.2.2 (2015-08-14)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.10.5 (Yosemite)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] htmlwidgets_0.5      DT_0.1               GtUtility_0.0.0.9000
##  [4] gplots_3.0.1         awsomics_0.0.0.9000  yaml_2.1.13         
##  [7] rmarkdown_0.9.6      knitr_1.12.3         RoCA_0.0.0.9000     
## [10] RCurl_1.95-4.8       bitops_1.0-6         devtools_1.11.1     
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.4          AnnotationDbi_1.32.3 magrittr_1.5        
##  [4] IRanges_2.4.8        BiocGenerics_0.16.1  highr_0.5.1         
##  [7] stringr_1.0.0        GenomeInfoDb_1.6.3   caTools_1.17.1      
## [10] tools_3.2.2          parallel_3.2.2       Biobase_2.28.0      
## [13] KernSmooth_2.23-15   DBI_0.3.1            withr_1.0.1         
## [16] htmltools_0.3.5      gtools_3.5.0         digest_0.6.9        
## [19] formatR_1.3          S4Vectors_0.8.11     RSQLite_1.0.0       
## [22] memoise_1.0.0        evaluate_0.9         gdata_2.17.0        
## [25] stringi_1.0-1        stats4_3.2.2         jsonlite_0.9.20</code></pre>
<div align="right">
<em><a href="http://zhezhangsh.github.io/RoCA">Go to project home</a></em>
</div>
<hr />
<p><em>END OF DOCUMENT</em></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
