---
title: "Compare 2 RNA-seq samples with no or few replicates"
author: "Jim Zhang"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: true
---

**Introduction**: Analysis of differential expression using RNA-seq sample groups with no or very few replicates. The analysis uses gene-level read counts obtained from RNA-seq data and [edgeR](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2796818) method to compare groups/samples.
***

```{r global_setup, include=FALSE}
knitr::opts_chunk$set(dpi=300, dev=c('png', 'pdf'), fig.pos="H", fig.width=8, fig.height=6, echo=FALSE, warning=FALSE, message=FALSE);

require(yaml);
require(knitr);
require(RankProd);
require(DEGandMore);
require(awsomics);
require(rchive);

fn.yaml<-"~/R/source/RoCA/template/rnaseq/compare_small/compare_small.yml";
yml<-yaml.load_file(fn.yaml);

cnts<-readRDS(yml$input$data); 
anno<-readRDS(yml$input$annotation); 
gset<-readRDS(yml$input$geneset); 
path<-yml$output; 
  
if (!file.exists(path)) dir.create(path, recursive=TRUE); 

path.stat<-paste(path, 'edgeR', sep='/');
if (!file.exists(path.stat)) dir.create(path.stat, recursive=TRUE); 

path.ora<-paste(path, 'ora', sep='/');
if (!file.exists(path.ora)) dir.create(path.ora, recursive=TRUE); 

path.fig<-paste(path, 'figure', sep='/');
if (!file.exists(path.fig)) dir.create(path.fig, recursive=TRUE); 

path.r<-paste(path, 'R', sep='/');
if (!file.exists(path.r)) dir.create(path.r, recursive=TRUE); 

path.tbl<-paste(path, 'table', sep='/');
if (!file.exists(path.tbl)) dir.create(path.tbl, recursive=TRUE); 

if (is.null(yml$home)) home.url<-'' else 
  home.url<-paste("<div align='right'>**_[Go back to project home](", yml$home, ")_**</div>", sep='');

anno<-anno[rownames(anno) %in% rownames(cnts), ];
cnts<-cnts[rownames(anno), ];

anno0<-anno;
anno0$Symbol<-CleanHtmlTags(anno0$Symbol, FALSE);

write.csv(anno0, paste(path.tbl, 'anno.csv', sep='/')); 
write.csv(cnts, paste(path.tbl, 'count.csv', sep='/')); 
saveRDS(anno, paste(path.r, 'anno.rds', sep='/')); 
saveRDS(cnts, paste(path.r, 'count.rds', sep='/')); 

# Comparisons
comp<-yml$parameter$comparison;
grps<-sapply(comp, names); 
```

`r home.url`

# Description

```{r description, eval=TRUE, include=FALSE}

lns<-lapply(names(yml$description), function(nm) {
  c(paste('##', nm), '\n', yml$description[[nm]], '\n'); 
});
lns<-paste(do.call('c', lns), collapse='\n'); 
```

`r lns`

`r home.url`

# Gene-level analysis

```{r edgeR, include=FALSE}
stat<-lapply(comp, function(c) DeEdgeR(cnts, c)[[1]]); 
stat<-lapply(stat, function(stat) stat[order(stat[, 'Pvalue']), c(1, 2, 7, 4, 5, 6)]); 

# Write outputs
fns<-sapply(names(stat), function(nm) {
  s<-stat[[nm]]; 
  f<-paste('edgeR', nm, sep='_'); 
  saveRDS(s, paste(path.r, '/', f, '.rds', sep=''));
  write.csv(cbind(s, anno0[rownames(s), ]), paste(path.tbl, '/', f, '.csv', sep=''));
  CreateDatatable(cbind(FormatNumeric(s), anno[rownames(s), ]), paste(path.stat, '/', f, '.html', sep=''), caption=paste('edgeR:', nm));
  pdf(paste(path.fig, '/scatter_', f, '.pdf', sep=''), w=6, h=6);
  PlotLogScatter(s[, 1], s[, 2], xlab=colnames(s)[1], ylab=colnames(s)[2]); 
  abline(0, 1, col='gold', lwd=2, lty=2); 
  dev.off();
  c(paste('edgeR/', f, '.html', sep=''), paste('figure/scatter_', f, '.pdf', sep=''));  
}); 

deg<-lapply(names(stat), function(nm) {
  nms<-names(comp[[nm]]); 
  s<-stat[[nm]]; 
  s<-FormatNumeric(s); 
  up<-s[s[, 4]>0 & s[, 5]<=yml$parameter$p & s[, 3]>=yml$parameter$logCPM, , drop=FALSE]; 
  dn<-s[s[, 4]<0 & s[, 5]<=yml$parameter$p & s[, 3]>=yml$parameter$logCPM, , drop=FALSE]; 
  up<-up[rev(order(up[, 4])), , drop=FALSE]; 
  dn<-dn[order(dn[, 4]), , drop=FALSE]; 
  up<-up[1:min(yml$parameter$top, nrow(up)), , drop=FALSE]; 
  dn<-dn[1:min(yml$parameter$top, nrow(dn)), , drop=FALSE]; 

  CreateDatatable(cbind(dn, anno[rownames(dn), ]), paste(path.stat, '/', nm, '_Down.html', sep=''), caption=paste(nms, collapse=' > ')); 
  CreateDatatable(cbind(up, anno[rownames(up), ]), paste(path.stat, '/', nm, '_Up.html', sep=''), caption=paste(nms, collapse=' < ')); 
  
  deg<-list(dn, up); 
  names(deg)<-paste('edgeR/', nm, '_', c('Down', 'Up'), '.html', sep=''); 
  deg; 
}); 
names(deg)<-names(comp); 
n.deg<-sapply(deg, function(x) sapply(x, nrow)); 
saveRDS(deg, file=paste(path.r, 'deg.rds', sep='/')); 

tbl<-data.frame(Comparison=paste('[', names(comp), '](', fns[2, ], ')', sep=''), A=grps[1, ], B=grps[2, ], stringsAsFactors = FALSE);
tbl$'A > B'= paste('[', n.deg[1, ], '](', sapply(deg, function(x) names(x)[1]), ')', sep=''); 
tbl$'A < B'= paste('[', n.deg[2, ], '](', sapply(deg, function(x) names(x)[2]), ')', sep=''); 
tbl$'All_Genes'= paste('[', nrow(anno), '](', fns[1, ], ')', sep=''); 
```

EdgeR was used to compare the groups/samples. When there are no replicates, the edgeR p value and FDR were estimated based on a conservative global dispersion. However, they are theoretically not reliable statistics. Therefore, the selection of differentially expressed genes was based on the combination of a modest p value (_Pvalue_ <= `r yml$parameter$p`),  the magnitude of sample/group difference (_LogFC_ >= top `r yml$parameter$top`), as well as gene expression level (_logCPM_ >= `r yml$parameter$logCPM`). 

**Table 1** Differentially expressed genes (DEGs). The top `r yml$parameter$top` genes with the largest fold changes, logCPM greater than `r yml$parameter$logCPM`, and edgeR p values less than `r yml$parameter$p` were selected from each comparison. 

`r kable(tbl, row.names=FALSE, align=rep('c', ncol(tbl)))`

`r home.url`

# Gene set-level analysis

Over-representation analysis (ORA) of differentially expressed genes. Pre-defined gene sets were split by their sources. 

```{r ora, include=FALSE}
gs.src<-sort(unique(gset[[1]][, 'Source']));
tbls<-sapply(names(deg), function(nm) {
  cat(nm, '\n');
  g<-lapply(deg[[nm]], rownames);
  names(g)<-paste('Higher_in_', names(comp[[nm]]), sep=''); 
  out<-lapply(names(g), function(nm1) {
    stat<-TestGSE(g[[nm1]], rownames(anno), gset[[2]])[[1]]; 
    ora<-WrapGSE(stat, gset[[1]], paste(path.ora, paste(nm, nm1, sep='_'), sep='/'), FALSE); 
    saveRDS(ora, file=paste(path.r, paste('ORA_', nm, '_', nm1, '.rds', sep=''), sep='/'));
    sapply(names(ora$formatted), function(nm2) {
      t<-ora$formatted[[nm2]];
      t$Name<-CleanHtmlTags(t$Name, FALSE); 
      write.csv(t, paste(path.tbl, paste('ORA_', nm2, '_', nm, '_', nm1, '.csv', sep=''), sep='/')); 
    })->x; 
    cnt<-sapply(ora[[2]], nrow); 
    lnk<-paste('[', cnt, '](', sub(paste(path, '/', sep=''), '', ora[[3]]), ')', sep=''); 
    names(lnk)<-names(cnt);
    lnk<-lnk[gs.src];
    lnk[is.na(lnk)]<-'0';
    names(lnk)<-gs.src;
    lnk;
  }); 
  out; 
}); 

tbl<-lapply(colnames(tbls), function(nm) {
  t<-tbls[, nm]; 
  t<-rbind(t[[1]], t[[2]]); 
  t<-cbind(Genes=paste('Higher_in_', grps[, nm], sep=''), t); 
  t<-cbind(Comparison=rep(nm, 2), t); 
  t; 
}); 
tbl<-do.call('rbind', tbl); 
```

**Table 2** Gene set over-represented in the top differentially expressed genes.

`r kable(tbl, row.names=FALSE, align=c('l', 'l', rep('c', ncol(tbl)-2)))`

`r home.url`

***
**END_OF_DOCUMENT**

