---
title: "Developer gives a name to the analysis to be implemented"
author: "`r if (exists('yml')) yml$analyst else 'Jim Zhang'`"
date: "`r Sys.Date()`"
output:
  html_document:
    number_sections: yes
    self_contained: no
    toc: yes
    toc_float:
      collapsed: no
---

<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
**Introduction** 

Developer of this R Markdown template should describe here the goal, scope, input data and major steps of this analysis. 

This skeleton helps developers to implement their own reproducible analysis using the RoCA framework. Visit the [RoCA site](http://zhezhangsh.github.io/RoCA) for more information about this project.

Definitation of terms: 

  - ***RoCA framework***       : RoCA is a paradigm of reproducible data analysis for bioinformaticians to share their custom analysis procedure. It creates dynamic data analysis reports based on R Markdown templates. Visit [RoCA home](http://zhezhangsh.github.io/RoCA) for more information.
  - ***R Markdown template***  : An .Rmd document that includes both R code and documentation of a data analysis procedure. It generates dynamic report according to user-specific input data and parameters for each analysis.
  - ***YAML key-value pairs*** : The pairs are specifications of each analysis performed using the R Markdown template. They are formatted and saved together as a [YAML file](http://www.yaml.org). It is highly recommended to pair each R Markdown template with a YAML file having the same file name. 
  - ***Knitr report***         : Data analysis report in HTML, PDF, or Word file, generated by the [knitr tool](http://yihui.name/knitr) for each analysis, based on the R Markdown template and the YAML specifications.
  - ***Developer***            : A developer implements a data analysis procedure into a R Markdown template, so the analysis can be repeated by any data analysts using their own data and parameters. The developer should also provide introduction and examples about how to set up an analysis using the template.
  - ***Analyst***              : Given an available R Markdown template generated by a developer, an analyst only needs to work with the pairing YAML file to set up a specific analysis. The analyst should follow instructions to prepare input data and choose parameters for each analysis. 
  
</div>

&nbsp;

```{r global_setup, eval=TRUE, include=FALSE}
# <!-- 
# The path and name of the pairing YAML file
# List the R packages required by this analysis
# List the subfolder names in the output folder
# -->

name.yaml <- 'skeleton.yaml'; 
name.packages <- c('rmarkdown', 'knitr', 'yaml', 'DT', 'htmlwidgets', 'GenomicRanges', 'RoCA', 'awsomics'); 
name.subfolders <- c('path.input'='input', 'path.r'='R', 'path.fig'='figure', 'path.tbl'='table'); # <full.path.to.subfolder>=<subfolder.name>

# <!-- 
# Keep the remaining part of this code chunk unchanged for all RoCA templates, k.
# -->

## Default knitr parameters
knitr::opts_chunk$set(dpi=300, dev=c('png', 'pdf'), echo=FALSE, warning=FALSE, message=FALSE);
if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

## Load required R packages
loaded <- LoadPackage(name.packages);
if (length(loaded[!loaded])) stop('Error: failed to load package(s) ', paste(names(loaded[!loaded]), collapse=', '));

## By default, before knitting this R Markdown file, the YAML file pairing it has been loaded.
## But, the developer can also provide the option to load the YAML file on the fly.
## So, the template can be run using the "Knit HTML" button in RStudio
if (!exists('yml'))                   # if the 'yml' variable doesn't exist yet, create it by loading the YAML file
	if (file.exists(name.yaml))         # assume the pairing YAML file exists in the current folder with the same name
		yml<-yaml.load_file(name.yaml);   # rename the YAML file to fit this template

prms <- yml$parameter;

## Generate directory and sub-directories where the output files will be
f <- GenerateFolder(yml$output, name.subfolders);
path <- yml$output;
for (i in 1:length(name.subfolders)) assign(names(name.subfolders)[i], f[name.subfolders[i]]); 

## URL to project home
## Use this line to add a link to project home in the report: `r home.url`
home.url <- Link2Home(yml$home);
```

`r home.url` <!-- plug this line anywhere you want to add a link to the project home page in the report -->

<!--
This whole section is dedicated to the <description> block in the YAML file.
Each key-value pair in that block will be written out as a subsection. 
Keep it unchanged for all RoCA templates
-->

# Description

`r WriteDescription(yml$description)`

`r home.url`

```{r load_data, include=FALSE}
# <!--
# Load data files specified by the <input> block in the YAML file.
# RoCA provides a few helper function to import data in common structures and file types.
# Use the DownloadFile function to download file first if remote files are acceptable.
# Read <https://github.com/zhezhangsh/RoCA/wiki/Format-input-data> for more information. 
# -->

# Import input data 
a.vector <- ImportVector(DownloadFile(yml$input$vector, path.input));         # Import a vector-like data object saved in acceptable file types
a.list <- ImportList(DownloadFile(yml$input$list, path.input));               # Import a list-like data object saved in acceptable file types
a.table <- ImportTable(DownloadFile(yml$input$table, path.input));            # Import a table-like data object saved in acceptable file types
a.robject <- ImportR(DownloadFile(yml$input$rdata, path.input));              # Import any R object saved in an .rds, .rdata or .rda file
a.rds <- readRDS(DownloadFile(yml$input$rds, path.input));                    # Import any R object saved in an .rds file
a.rdata <- eval(parse(text=load(DownloadFile(yml$input$rdata, path.input)))); # Import any R object saved in an .rdata or .rda file
a.matrix <- read.table(DownloadFile(yml$input$table, path.input), sep='\t');  # Import using basic R functions (read.csv, readLine, etc.)  
# a.text <- YourParser(DownloadFile(yml$input$log, path.input));              # Import a text file using your own parser

# <!--
# Optionally, save imported data object to output folder
# -->

# save imported data to be included in the output folder
saveRDS(a.vector, paste(path.input, 'a_vector.rds', sep='/')); 
```

```{r prepare_data, include=FALSE}
# <!--
# Use this chunk to process imported data to prepare for analysis
# Save processed data that might be used in the future to the output folder or subfolders
# Use the CreateDatatable function to write a table in Java DataTables format into an HTML file
# -->

# Process data to prepare for analysis
grp <- unique(a.vector); 
chr <- BiocGenerics::as.data.frame(a.robject); 
tbl <- a.table[, a.vector==grp[1], drop=FALSE]; 

# Save processed data that might be re-used in the future 
saveRDS(grp, paste(path.r, 'group_name.rds', sep='/')); 
write.csv(tbl, paste(path.tbl, 'subset.csv', sep='/')); 
fn.html <- TrimPath(CreateDatatable(chr, paste(path.tbl, 'chromosome.html', sep='/')));  # <!-- TrimPath function removes the path -->
```

# The first result section <!-- Change section title -->

<!-- replace this paragraph in your template -->
[R Markdown](http://rMarkdown.rstudio.com) generates dynamic documents using R code for data analysis and [Markdown](https://daringfireball.net/projects/markdown) markup language for text. For each analysis step in an R Markdown file, the deveoper should describe the analysis in terms of its goal, rational, method, etc.

<!-- this is a silent code chunk that does some analysis but will not be written into the report -->
```{r analysis_step_1, include=FALSE} 
# <!-- Write some code to do some analysis -->
```

## The first subsection <!-- Change subsection title -->

<!-- replace this paragraph in your template -->
There are two ways to show a table: by using the kable function to include a small table in the final report, or by adding a [link](table/`r fn.html`) that points to an HTML table in Jave DataTables format. By default, all HTML tables will be saved in the _table/_ subfolder.

<!-- this is a code chunk that prepares for a summary table to be included in the report -->
```{r table_1, include=FALSE}
# <!-- create a random, meaningless small table -->
tbl<-t(sapply(1:4, function(i) summary(rnorm(100)))); 
rownames(tbl)<-paste('Draw', 1:nrow(tbl), sep='_'); 
```

<div style="color:darkblue; padding:0 3cm">
**Table 1** Write the table title and caption here.
</div>

<div align='center', style="padding:0 3cm"> <!-- Horizontally align the table at the center of the page -->
<!-- Use the kable function to directly put a small table in the report -->
`r kable(tbl, row.names=TRUE, align=rep('c', ncol(tbl)))`
</div>

## The second subsection <!-- Change subsection title -->

<!-- replace this paragraph in your template -->
By default, all figures will be saved in the _figure/_ subfolder as both .pdf and .png files.

<!-- this is a code chunk that will generate a plot to be written into the report -->
<div align='center'> <!-- Horizontally align the figure at the center of the page -->
```{r figure_1, include=TRUE, fig.width=6, fig.height=4, out.width='600px'}
# <!-- a random, meaningless plot -->
plot(density(rnorm(100)), lwd=2, col='gold', main='', xlab='', ylab=''); 
```
</div>

<div style="color:darkblue; padding:0 3cm">
**Figure 1.** Write the figure caption here. 
</div>

`r home.url`

# The second result section <!-- Change section title -->

<!-- replace this paragraph in your template -->
The develop will decide how to split a full data analysis procedure into multiple steps and present the results in different sections and subsection. Be careful with the dependency between steps. 

<!-- this is a silent code chunk that does some analysis but will not be written into the report -->
```{r analysis_step_2, include=FALSE} 
# <!-- Write more code to do more analysis -->
```

# More sections ...

`r home.url`

***

<!-- Keep this section unchanged for all RoCA templates -->
# Appendix 

Check out the **[RoCA home page](http://zhezhangsh.github.io/RoCA)** for more information.  

## Reproduce this report

To reproduce this report: 

  1. Find the data analysis template you want to use and an example of its pairing YAML file  [here](https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples) and download the YAML example to your working directory

  2. To generate a new report using your own input data and parameter, edit the following items in the YAML file:

    - _output_        : where you want to put the output files
    - _home_          : the URL if you have a home page for your project
    - _analyst_       : your name
    - _description_   : background information about your project, analysis, etc.
    - _input_         : where are your input data, read instruction for preparing them
    - _parameter_     : parameters for this analysis; read instruction about how to prepare input data

  3. Run the code below within ***R Console*** or ***RStudio***, preferablly with a new R session:

```{r reproduce_report, eval=FALSE, echo=TRUE}
if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis
```

If there is no complaint, go to the _output_ folder and open the ***index.html*** file to view report. 

## Session information

```{r session_info, echo=FALSE}
sessionInfo(); 
```

`r home.url`

***
_END OF DOCUMENT_
