---
title: "Developer gives a name to the analysis to be implemented"
author: "`r if (exists('yml')) yml$analyst else 'Jim Zhang'`"
date: "`r Sys.Date()`"
output:
  html_document:
    number_sections: yes
    self_contained: no
    toc: yes
    toc_float:
      collapsed: no
---

<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
**Introduction** 

Developer of this R markdown template should describe here the goal, scope, input data and major steps of this analysis. 

This skeleton helps developers to implement their own reproducible analysis using the RoCA framework. Visit the [RoCA site](http://zhezhangsh.github.io/RoCA) for more information about this project.

Definitation of terms: 

  - ***RoCA framework***       : RoCA is a paradigm of reproducible data analysis for bioinformaticians to share their custom analysis procedure. It creates dynamic data analysis reports based on R markdown templates. Visit [RoCA home](http://zhezhangsh.github.io/RoCA) for more information.
  - ***R markdown template***  : An .Rmd document that includes both R code and documentation of a data analysis procedure. It generates dynamic report according to user-specific input data and parameters for each analysis.
  - ***YAML key-value pairs*** : The pairs are specifications of each analysis performed using the R Markdown template. They are formatted and saved together as a [YAML file](http://www.yaml.org). It is highly recommended to pair each R markdown template with a YAML file having the same file name. 
  - ***Knitr report***         : Data analysis report in HTML, PDF, or Word file, generated by the [knitr tool](http://yihui.name/knitr) for each analysis, based on the R markdown template and the YAML specifications.
  - ***Developer***            : A developer implements a data analysis procedure into a R markdown template, so the analysis can be repeated by any data analysts using their own data and parameters. The developer should also provide introduction and examples about how to set up an analysis using the template.
  - ***Analyst***              : Given an available R markdown template generated by a developer, an analyst only needs to work with the pairing YAML file to set up a specific analysis. The analyst should follow instructions to prepare input data and choose parameters for each analysis. 
  
</div>

&nbsp;

```{r global_setup, eval=TRUE, include=FALSE}
# <!-- 
# The path and name of the pairing YAML file
# List the R packages required by this analysis
# List the subfolder names in the output folder
# -->

name.yaml <- 'skeleton.yaml'; 
name.packages <- c('rmarkdown', 'knitr', 'yaml', 'DT', 'htmlwidgets', 'RoCA', 'Biobase'); 
name.subfolders <- c('path.input'='input', 'path.r'='R', 'path.fig'='figure', 'path.tbl'='table'); # <full.path.to.subfolder>=<subfolder.name>

# <!-- 
# For all RoCA templates, keep the remaining code of this chunk unchanged.
# -->

## Default knitr parameters
knitr::opts_chunk$set(dpi=300, dev=c('png', 'pdf'), echo=FALSE, warning=FALSE, message=FALSE);
if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

## Load required R packages
loaded <- LoadPackage(name.packages);
if (length(loaded[!loaded])) stop('Error: failed to load package(s) ', paste(names(loaded[!loaded]), collapse=', '));

## By default, before knitting this R Markdown file, the YAML file pairing it has been loaded.
## But, the developer can also provide the option to load the YAML file on the fly.
## So, the template can be run using the "Knit HTML" button in RStudio
if (!exists('yml'))                   # if the 'yml' variable doesn't exist yet, create it by loading the YAML file
	if (file.exists(name.yaml))         # assume the pairing YAML file exists in the current folder with the same name
		yml<-yaml.load_file(name.yaml);   # rename the YAML file to fit this template

prms <- yml$parameter;

## Generate directory and sub-directories where the output files will be
f <- GenerateFolder(yml$output, name.subfolders);
path <- yml$output;
for (i in 1:length(name.subfolders)) assign(names(name.subfolders)[i], f[name.subfolders[i]]); 

## URL to project home
## Use this line to add a link to project home in the report: `r home.url`
home.url <- Link2Home(yml$home);
```

`r home.url` <!-- plug this line anywhere you want to add a link to the project home page in the report -->

<!--
This whole section is dedicated to the <description> block in the YAML file.
Each key-value pair in that block will be written out as a subsection. 
Keep it unchanged for all RoCA reports.
-->

# Description

`r WriteDescription(yml$description)`

`r home.url`

```{r load_data, include=FALSE}
# <!--
# Load data files specified by the <input> block in the YAML file.
# RoCA provides a few helper function to import data in common structures and file types.
# Use the DownloadFile function to download file first if remote files are acceptable.
# Read RoCA instruction () for more information. 
# -->

a.rds <- readRDS(DownloadFile(yml$input$data1, path.input));                    # Import any types of data saved in an .rds file
a.rdata <- eval(parse(text=load(DownloadFile(yml$input$data2, path.input))));   # Import any types of data saved in an .rdata or .rda file

a.vector <- readRDS(DownloadFile(yml$input$rds, path.input));   
#### Import input data; use the DownloadFile function to download a file first if it's remote
## Import from different file types
# data1<-readRDS(DownloadFile(yml$input$data1, path.input));                 # import from .rds file
# data2<-eval(parse(text=load(DownloadFile(yml$input$data2, path.input))));  # import from .rdata or .rds file
# data3<-read.csv(DownloadFile(yml$input$data3, path.input));                # import from .csv file
# data4<-read.table(DownloadFile(yml$input$data4, path.input), sep='\t');    # import from tab-delimited .txt file
## Or, import from files in supported formats: https://github.com/zhezhangsh/RoCA/blob/master/document/instruction/data_format.md
# data5<-ImportTable(DownloadFile(yml$input$data5, path.input));             # import table-like data in recognizable file types
# data6<-ImportList(DownloadFile(yml$input$data6, path.input));              # import list-like data in recognizable file types
# data7<-ImportVector(DownloadFile(yml$input$data7, path.input));            # import vector-like data in recognizable file types
## Or, for large remote files (.fastq, .bam, .vcf, etc.), establish a connection
```

```{r R_code, include=FALSE}
## One or more R code chunks
```

# Section Title 

Mesh up Markdown text with R code

## Subsection Title

<div align='center'>
```{r plot, include=TRUE, fig.width=6, fig.height=4, out.width='480px'}
# Plot example
par(mar=c(5,2,2,2));
v<-rnorm(100); 
plot(density(v), lwd=2, col='blue', main='', xlab=''); 
```
</div>

<div style="color:darkblue; padding:0 3cm">
**Figure 1.** Figure example: density of `r length(v)` random values drawn from a normal distribution, using the **_rnorm_** function. 
</div>

## More Subsection Title

```{r table}
tbl<-t(sapply(1:4, function(i) summary(rnorm(length(v))))); 
rownames(tbl)<-paste('Draw', 1:nrow(tbl), sep='_'); 
```

<div style="color:darkblue; padding:0 3cm">
**Table 1** Example table: summary of `r nrow(tbl)` random value vectors drawn from a normal distribution, using the **_rnorm_** function
</div>

<div align='center', style="padding:0 3cm">
`r kable(tbl, row.names=TRUE, align=rep('c', ncol(tbl)))`
</div>

# More Section Title

`r home.url`

***
_END OF DOCUMENT_
