<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jim Zhang" />

<meta name="date" content="2019-09-05" />

<title>Frequency of oligomers in DNA sequences</title>

<script src="index_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="index_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="index_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="index_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="index_files/navigation-1.1/tabsets.js"></script>
<link href="index_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="index_files/highlightjs-9.12.0/highlight.js"></script>
<script src="index_files/kePrint-0.0.1/kePrint.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Frequency of oligomers in DNA sequences</h1>
<h4 class="author"><em>Jim Zhang</em></h4>
<h4 class="date"><em>2019-09-05</em></h4>

</div>


<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
<p>This analysis reports of the frequency of oligomers in DNA sequences and compare the observed frequency to expected frequency.</p>
<ul>
<li>Oligomers of the same length are analyzed together.</li>
<li>The length of the oligomers can range from 2 to 8, depending on the user-specified values.</li>
<li>The expected frequency will be calculated according to the overall base frequency in DNA sequences (A, C, G, or T only).</li>
<li>Over- or under-representation of oligomers is calculated as the ratio of their observed frequency over expected frequency, and tested by Fisher’s exact test.</li>
<li>For each oligomer, skewness of its occurances at relative positions of DNA sequences is tested by Kolmogorov-Smirnov test, and its enrichment with the center or both ends of the sequences is tested by Fisher’s exact test.</li>
</ul>
</div>
<p> </p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<div id="description" class="section level1">
<h1><span class="header-section-number">1</span> Description</h1>
<div id="project" class="section level2">
<h2><span class="header-section-number">1.1</span> Project</h2>
<p>rGADAM wrapper</p>
</div>
<div id="analysis" class="section level2">
<h2><span class="header-section-number">1.2</span> Analysis</h2>
<p>Search for DNA motifs around TSS of a few random genes.</p>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
</div>
<div id="results" class="section level1">
<h1><span class="header-section-number">2</span> Results</h1>
<div id="dna-sequences" class="section level2">
<h2><span class="header-section-number">2.1</span> DNA sequences</h2>
<p><strong>Table 1.</strong> Length and GC content of <strong>81</strong> DNA sequences.</p>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:center;">
Min.
</th>
<th style="text-align:center;">
1st Qu.
</th>
<th style="text-align:center;">
Median
</th>
<th style="text-align:center;">
Mean
</th>
<th style="text-align:center;">
3rd Qu.
</th>
<th style="text-align:center;">
Max.
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Length (bp)
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
<td style="text-align:center;">
201.00
</td>
</tr>
<tr>
<td style="text-align:left;">
GC (%)
</td>
<td style="text-align:center;">
25.87
</td>
<td style="text-align:center;">
46.27
</td>
<td style="text-align:center;">
56.22
</td>
<td style="text-align:center;">
56.88
</td>
<td style="text-align:center;">
69.65
</td>
<td style="text-align:center;">
86.57
</td>
</tr>
</tbody>
</table>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
</div>
<div id="oligo-frequency" class="section level2">
<h2><span class="header-section-number">2.2</span> Oligo frequency</h2>
<p><strong>Table 2.</strong> Summary of oligomer combinations.</p>
<ul>
<li><strong>Length</strong>: length of the oligomers; click link to view details of all combinations</li>
<li><strong>N_Theoretical</strong>: total number of combinations in theory</li>
<li><strong>N_Observed</strong>: number of combinations observed in sequences</li>
<li><strong>N_Significant</strong>: number of combinations signifantly over- or under-represented in sequences according to proportional test</li>
<li><strong>Correlation</strong>: correlation between the expected and observed frequency of all combinations</li>
<li><strong>P_Normality</strong>: p value of the normality test on the log-ratio of observed over expected frequencies</li>
<li><strong>Top 2</strong>: top 2 most over-represented combination(s) with p value of proportional test less than 0.01</li>
</ul>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Length
</th>
<th style="text-align:right;">
N_Theoretical
</th>
<th style="text-align:right;">
N_Observed
</th>
<th style="text-align:right;">
N_Significant
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:right;">
P_Normality
</th>
<th style="text-align:left;">
Top 2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="table/stat_2mer.html">2mer</a>
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
0.74
</td>
<td style="text-align:right;">
3.5e-01
</td>
<td style="text-align:left;">
AA;TT
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="table/stat_3mer.html">3mer</a>
</td>
<td style="text-align:right;">
64
</td>
<td style="text-align:right;">
64
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0.66
</td>
<td style="text-align:right;">
1.1e-02
</td>
<td style="text-align:left;">
AAA;TTT
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="table/stat_4mer.html">4mer</a>
</td>
<td style="text-align:right;">
256
</td>
<td style="text-align:right;">
256
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:right;">
0.60
</td>
<td style="text-align:right;">
2.0e-06
</td>
<td style="text-align:left;">
TTTT;AAAA
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="table/stat_5mer.html">5mer</a>
</td>
<td style="text-align:right;">
1024
</td>
<td style="text-align:right;">
1021
</td>
<td style="text-align:right;">
212
</td>
<td style="text-align:right;">
0.54
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
ATTTT;AAGAA
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="table/stat_6mer.html">6mer</a>
</td>
<td style="text-align:right;">
4096
</td>
<td style="text-align:right;">
3694
</td>
<td style="text-align:right;">
239
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
AAAATA;TTTATT
</td>
</tr>
</tbody>
</table>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<hr />
</div>
</div>
<div id="appendix" class="section level1">
<h1><span class="header-section-number">3</span> Appendix</h1>
<p>Check out the <strong><a href="http://zhezhangsh.github.io/RoCA">RoCA home page</a></strong> for more information.</p>
<div id="reproduce-this-report" class="section level2">
<h2><span class="header-section-number">3.1</span> Reproduce this report</h2>
<p>To reproduce this report:</p>
<ol style="list-style-type: decimal">
<li><p>Find the data analysis template you want to use and an example of its pairing YAML file <a href="https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples">here</a> and download the YAML example to your working directory</p></li>
<li><p>To generate a new report using your own input data and parameter, edit the following items in the YAML file:</p>
<ul>
<li><em>output</em> : where you want to put the output files</li>
<li><em>home</em> : the URL if you have a home page for your project</li>
<li><em>analyst</em> : your name</li>
<li><em>description</em> : background information about your project, analysis, etc.</li>
<li><em>input</em> : where are your input data, read instruction for preparing them</li>
<li><em>parameter</em> : parameters for this analysis; read instruction about how to prepare input data</li>
</ul></li>
<li><p>Run the code below within <strong><em>R Console</em></strong> or <strong><em>RStudio</em></strong>, preferablly with a new R session:</p></li>
</ol>
<pre class="r"><code>if (!require(devtools)) { install.packages(&#39;devtools&#39;); require(devtools); }
if (!require(RCurl)) { install.packages(&#39;RCurl&#39;); require(RCurl); }
if (!require(RoCA)) { install_github(&#39;zhezhangsh/RoCAR&#39;); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited</code></pre>
<p>If there is no complaint, go to the <em>output</em> folder and open the <strong><em>index.html</em></strong> file to view report.</p>
</div>
<div id="session-information" class="section level2">
<h2><span class="header-section-number">3.2</span> Session information</h2>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      stats4    parallel  stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] arrangements_1.1.5                    
##  [2] combinat_0.0-8                        
##  [3] seqinr_3.4-5                          
##  [4] e1071_1.7-2                           
##  [5] BSgenome.Hsapiens.NCBI.GRCh38_1.3.1000
##  [6] Agri_0.0.0.9000                       
##  [7] rGADEM_2.30.0                         
##  [8] seqLogo_1.48.0                        
##  [9] BSgenome_1.50.0                       
## [10] rtracklayer_1.42.1                    
## [11] Biostrings_2.50.1                     
## [12] XVector_0.22.0                        
## [13] GenomicFeatures_1.34.1                
## [14] AnnotationDbi_1.44.0                  
## [15] Biobase_2.42.0                        
## [16] GenomicRanges_1.34.0                  
## [17] GenomeInfoDb_1.18.1                   
## [18] IRanges_2.16.0                        
## [19] S4Vectors_0.20.1                      
## [20] BiocGenerics_0.28.0                   
## [21] htmlwidgets_1.3                       
## [22] DT_0.5                                
## [23] kableExtra_0.9.0                      
## [24] awsomics_0.0.0.9000                   
## [25] yaml_2.2.0                            
## [26] rmarkdown_1.10                        
## [27] knitr_1.20                            
## [28] RoCA_0.0.0.9000                       
## [29] RCurl_1.95-4.11                       
## [30] bitops_1.0-6                          
## [31] usethis_1.4.0                         
## [32] devtools_2.0.1                        
## 
## loaded via a namespace (and not attached):
##  [1] matrixStats_0.54.0          fs_1.2.6                   
##  [3] bit64_0.9-7                 progress_1.2.0             
##  [5] httr_1.3.1                  rprojroot_1.3-2            
##  [7] tools_3.5.1                 backports_1.1.2            
##  [9] R6_2.3.0                    DBI_1.0.0                  
## [11] colorspace_1.3-2            ade4_1.7-13                
## [13] withr_2.1.2                 prettyunits_1.0.2          
## [15] processx_3.2.0              bit_1.1-14                 
## [17] compiler_3.5.1              cli_1.0.1                  
## [19] rvest_0.3.2                 xml2_1.2.0                 
## [21] desc_1.2.0                  DelayedArray_0.8.0         
## [23] scales_1.0.0                readr_1.3.1                
## [25] callr_3.0.0                 stringr_1.3.1              
## [27] digest_0.6.18               Rsamtools_1.34.0           
## [29] base64enc_0.1-3             pkgconfig_2.0.2            
## [31] htmltools_0.3.6             sessioninfo_1.1.1          
## [33] highr_0.7                   rlang_0.4.0                
## [35] rstudioapi_0.8              RSQLite_2.1.1              
## [37] shiny_1.2.0                 jsonlite_1.5               
## [39] crosstalk_1.0.0             BiocParallel_1.16.2        
## [41] magrittr_1.5                GenomeInfoDbData_1.2.0     
## [43] Matrix_1.2-15               Rcpp_1.0.0                 
## [45] munsell_0.5.0               stringi_1.2.4              
## [47] MASS_7.3-51.1               SummarizedExperiment_1.12.0
## [49] zlibbioc_1.28.0             pkgbuild_1.0.2             
## [51] blob_1.1.1                  promises_1.0.1             
## [53] crayon_1.3.4                lattice_0.20-38            
## [55] hms_0.4.2                   ps_1.2.1                   
## [57] pillar_1.3.0                biomaRt_2.38.0             
## [59] pkgload_1.0.2               XML_3.98-1.16              
## [61] glue_1.3.0                  evaluate_0.12              
## [63] remotes_2.0.2               httpuv_1.4.5               
## [65] testthat_2.2.1              assertthat_0.2.0           
## [67] mime_0.6                    xtable_1.8-3               
## [69] later_0.7.5                 class_7.3-14               
## [71] viridisLite_0.3.0           tibble_1.4.2               
## [73] GenomicAlignments_1.18.0    memoise_1.1.0              
## [75] gmp_0.5-13.5</code></pre>
<div align="right">
<em><a href="https://github.com/zhezhangsh/RoCA">Go to project home</a></em>
</div>
<hr />
<p><em>END OF DOCUMENT</em></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
