---
title: "Summarize three pairwise comparisons of differential expression"
author: "`r if (exists('yml')) yml$analyst else 'Jim Zhang'`"
date: "`r Sys.Date()`"
output:
  html_document:
    number_sections: yes
    self_contained: no
    toc: yes
    toc_float:
      collapsed: no
---

<div style="border:black 1px solid; padding: 0.5cm 0.5cm">
**Introduction**: This analysis is based on the outputs of pairwise comparisons of differential gene expression generated by [this](https://raw.githubusercontent.com/zhezhangsh/RoCA/master/template/de/pairwise_comparison/pairwise_comparison.Rmd) template. It uses results from 3 pairwise comparisons of 3 sample groups vs. their corresponding control groups and compares how these 3 sample groups are different from each other in terms of their sample-control differences (_delta-delta_). An example of such analysis is the different responses of 3 cell types to the treatment of the same drug. This analysis is focused on the overlapping of differentially expression at both gene and gene set levels.

</div>

&nbsp;

```{r global_setup, include=FALSE}
name.yaml <- 'summarize_three.yaml'; 
name.packages <- c('rmarkdown', 'knitr', 'yaml', 'DT', 'htmlwidgets', 'MASS', 'gplots', 'scatterplot3d',
                   'VennDiagram', 'RoCA', 'rchive', 'DEGandMore'); 
name.subfolders <- c('path.input'='input', 'path.r'='R', 'path.fig'='figure', 'path.tbl'='table');

knitr::opts_chunk$set(eval=TRUE, dpi=300, fig.pos="H", fig.width=8, fig.height=6, echo=FALSE, warning=FALSE, message=FALSE, fig.path='FIGURE/');

## Default knitr parameters
knitr::opts_chunk$set(dpi=300, dev=c('png', 'pdf'), echo=FALSE, warning=FALSE, message=FALSE);

if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

## Load required R packages
loaded <- LoadPackage(name.packages);
if (length(loaded[!loaded])) stop('Error: failed to load package(s) ', paste(names(loaded[!loaded]), collapse=', '));

## By default, before knitting this R Markdown file, the YAML file pairing it has been loaded.
## But, the developer can also provide the option to load the YAML file on the fly.
## So, the template can be run using the "Knit HTML" button in RStudio
if (!exists('yml'))                   # if the 'yml' variable doesn't exist yet, create it by loading the YAML file
  if (file.exists(name.yaml))         # assume the pairing YAML file exists in the current folder with the same name
    yml<-yaml.load_file(name.yaml);   # rename the YAML file to fit this template

prms <- yml$parameter;

## Generate directory and sub-directories where the output files will be
f <- GenerateFolder(yml$output, name.subfolders);
path <- yml$output;
for (i in 1:length(name.subfolders)) assign(names(name.subfolders)[i], f[name.subfolders[i]]); 

## URL to project home
## Use this line to add a link to project home in the report: `r home.url`
home.url <- Link2Home(yml$home);
```

```{r load_data, include=FALSE}
# Loading results from pairwsie DEG analysis
yml$input$comparison <- lapply(yml$input$comparison, function(c) {
  if (is.list(c)) c else {
    if (is.character(c) & dir.exists(c[1])) 
      list(result=paste(c, 'R/result.rds', sep='/'), index=paste(c, 'index.html', sep='/')) else
        stop('Error: comparison results not found\n'); 
  }
})

res <- lapply(yml$input$comparison, function(f) ImportR(DownloadFile(f$result, path.input))); 
gset <- ImportR(DownloadFile(yml$input$geneset, path.input)); 
smpl <- ImportR(DownloadFile(yml$input$sample, path.input));
comp <- names(yml$input$comparison); 

grp <- lapply(res, function(res) names(res$inputs$comparison));
smp <- lapply(res, function(x) x$inputs$comparison);
names(res) <- names(grp) <- names(smp) <- names(yml$input$comparison);

sapply(names(res), function(nm) 
  saveRDS(res[[nm]], paste(path.input, paste('result_', nm, '.rds', sep=''), sep='/'))) -> x;

grps<-do.call('c', lapply(res, function(res) res$inputs$comparison)); 
smpl <- smpl[unlist(smp, use.names=FALSE), ]; 

s0 <- setdiff(unlist(smp), rownames(smpl));
if (length(s0) > 0) stop('Error: missing metadata of sample(s): ', paste(s0, collapse='; '), '\n');

de<-lapply(res, function(res) res$de); 
ora<-lapply(res, function(res) res$ora); 
gsea<-lapply(res, function(res) res$gsea); 

ind1 <- c(1,1,2);
ind2 <- c(2,3,3);

# Combined matrix
expr.all <- lapply(res, function(res) {
  input<-res$inputs;
  s<-c(input$comparison[[1]], input$comparison[[2]]); 
  sapply(s, function(s) input$expr[, s]); 
}); 
g<-Reduce('intersect', lapply(expr.all, rownames)); 
expr.all<-do.call('cbind', lapply(expr.all, function(d) d[g, ])); 
```

`r home.url` 

# Description
  
`r WriteDescription(yml$description)`

`r home.url` 

## Pairwise comparisons

This report compares the results of the  following pairwise comparisons.  

```{r link_lines, include=FALSE}
lns <- sapply(names(yml$input$comparison), function(nm) {
  c <- yml$input$comparison[[nm]];
  ind <- c$index[1];
  l <- paste(grp[[nm]], collapse=' vs. ', sep='');
  if (is.null(ind) | identical('', ind) | identical(NA, ind)) l else paste('[', l, '](', yml$input$comparison[[1]]$index, ')', sep=''); 
});
lns <- as.vector(lns); 
```
  - **`r comp[1]`**: `r lns[1]`
  - **`r comp[2]`**: `r lns[2]`
  - **`r comp[3]`**: `r lns[3]`

`r home.url` 

# Gene-level comparison

## Global delta-delta correlation

```{r gene_level, include=TRUE}
stat<-lapply(de, function(de) de$result$stat); 
deg<-lapply(de, function(de) de$DEG); 
names(stat)<-names(deg)<-comp; 

# union set of gene id
if (prms$gene$union) gid<-Reduce('union', lapply(stat, rownames)) else 
  gid<-Reduce('intersect', lapply(stat, rownames)); 

# Gene annotation
anno<-lapply(res, function(res) res$inputs$anno); 
cnm <- Reduce('intersect', lapply(anno, colnames)); 
anno[[1]] <- anno[[1]][, colnames(anno[[1]]) %in% cnm, drop = FALSE]; 
anno[[2]] <- anno[[2]][, colnames(anno[[1]]), drop=FALSE];
anno[[3]] <- anno[[3]][, colnames(anno[[1]]), drop=FALSE];
anno.all <- lapply(anno, function(a) a[rownames(a) %in% gid, , drop=FALSE]); 
anno <- anno.all[[1]]; 
anno <- rbind(anno, anno.all[[2]][setdiff(rownames(anno.all[[2]]), rownames(anno)), , drop=FALSE]); 
anno <- rbind(anno, anno.all[[3]][setdiff(rownames(anno.all[[3]]), rownames(anno)), , drop=FALSE]); 
anno <- anno[gid, , drop=FALSE];
rownames(anno) <- gid;
anno[is.na(anno[, 1]), 1] <- rownames(anno)[is.na(anno[, 1])]; 

# DE statistics
s <- lapply(stat, function(s) data.frame(s)[gid, ]); 
l2r<-sapply(stat, function(stat) stat[, 'LogFC']); 
l2r[is.na(l2r)]<-0;
pvl<-sapply(stat, function(stat) stat[, 'Pvalue']); 
pvl[is.na(pvl)]<-1;
fdr<-sapply(stat, function(stat) stat[, 'FDR']); 
fdr[is.na(fdr)]<-1;
rnk<-apply(-sign(l2r)*log10(pvl), 2, rank);
l2r<-round(l2r, 3);
rownames(l2r)<-rownames(pvl)<-rownames(fdr)<-rownames(rnk)<-gid; 

tbl <- lapply(stat, function(s) s[gid, c(1,2,4,5,6)]); 
cnm <- lapply(tbl, colnames); 
tbl <- do.call('cbind', tbl);
rownames(tbl) <- gid; 
tbl <- cbind(anno[rownames(tbl), ], FormatNumeric(tbl)); 
colnames(tbl)[ncol(anno)+c(1:15)]<-paste(rep(names(grp), each=5), unlist(cnm), sep='::'); 
tbl[, 1] <- CleanHtmlTags(tbl[, 1], FALSE); 
tbl0 <- tbl;
write.csv(tbl0, paste(path.tbl, 'gene_stat.csv', sep='/')); 
saveRDS(tbl0, paste(path.r, 'gene_stat.rds', sep='/')); 
tbl[, 1] <- AddHref(tbl[, 1], UrlEntrezGene(rownames(tbl))); 
CreateDatatable(tbl, paste(path.tbl, 'gene_table.html', sep='/'), 
                caption='Gene-level differential expression of all 3 pairwise comparisons')->fn; 

corr <- round(sapply(1:3, function(i) cor(l2r[, ind1[i]], l2r[, ind2[i]], use='pairwise')), 3);
lns <- paste('  - Corr(', comp[ind1], ':', comp[ind2], ') = ', corr, sep='');
lns <- paste(lns, collapse='\n'); 

```

Both comparisons reported the log ratio of sample and control group means for each gene. The global agreement of log ratios of all genes indicates how much the results of these 3 comparisons are similar to or different from each other. Full table of gene-level statistics side-by-side is [here](table/gene_table.html). 

`r lns`

<div align='center'>
```{r log_ratio, include=TRUE, fig.width=8, fig.height=4, out.width='800px'}
par(mfrow=c(1, 2));
n.sig<-apply(pvl, 1, function(p) length(p[p<=prms$gene$pvalue]));
col<-c('lightgrey', 'yellow', 'orange', 'red')[n.sig+1];

scatterplot3d(l2r, pch=19, angle=45, cex.symbols =.25, color=col); 
scatterplot3d(l2r, pch=19, angle=135, cex.symbols =.25, color=col); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 1.** This plot shows the global correlation (correlation coefficient = `r corr`) between the 3 pairwise comparisons: **`r comp[1]`**, **`r comp[2]`**, and **`r comp[3]`**. The same 3D plot was showed in 2 different angles.  Genes obtained p values less than `r prms$gene$pvalue` from any 1, any 2, or all 3 comparisons were highlighted in yellow, orange, or red respectively. The correlatio coefficients between log-ratios of each pair of comparisons are: 
</div>

`r home.url` 

## Differentially expression genes (DEGs)

```{r deg, include=TRUE}
deg.n<-t(sapply(deg, function(deg) sapply(deg, nrow))); 
colnames(deg.n)<-c('DEG, > control', 'DEG, < control');
n0<-sapply(stat, nrow);
n1<-apply(pvl, 2, function(x) length(x[!is.na(x) & x<= prms$gene$pvalue])); 
deg.n<-cbind(n0, n1, deg.n);
colnames(deg.n)[1:2]<-c('Total_gene', paste('P <', prms$gene$pvalue));
#pander::pander(deg.n, align=c('l', 'c', 'c', 'c'), );
```

Both comparisons identified DEGs from 2 compared groups. Check report of individual comparisons for how the DEGs were selected. Overlapped DEGs identified by all 3 comparisons are worthy of a closer look. 

<div style="color:darkblue; padding:0 2cm">
**Table 1.** Number of DEGs identified by each comparison.
</div>

<div align='center', style="padding:0 2cm">
`r kable(deg.n, align=c('c', 'c', 'c', 'c'))`
</div>


```{r deg_overlap, include=FALSE}
path.deg<-paste(path, 'DEG', sep='/');
if (!file.exists(path.deg)) dir.create(path.deg, recursive = TRUE); 

deg.up<-lapply(deg, function(x) rownames(x[[1]])); 
deg.dn<-lapply(deg, function(x) rownames(x[[2]])); 
deg.up <- lapply(deg.up, function(x) x[x %in% rownames(tbl)]);
deg.dn <- lapply(deg.dn, function(x) x[x %in% rownames(tbl)]);

deg.all<-list(up=deg.up, down=deg.dn);
fn.tbl<-lapply(names(deg.all), function(nm) {
  gs<-deg.all[[nm]]; 
  sapply(1:3, function(i) {
    c<-comp[c(ind1[i], ind2[i])]; 
    g<-intersect(gs[[ind1[i]]], gs[[ind2[i]]]); 
    t<-tbl[g, , drop=FALSE]; 
    f<-paste(paste(nm, c[1], c[2], sep='_'), '.html', sep=''); 
    CreateDatatable(t, paste(path.deg, f, sep='/'), caption=paste(nm, ', ', c[1], ' and ', c[2], sep='')); 
    names(f)<-paste(c, collapse=' and '); 
    f;
  })
}); 

deg.up3<-Reduce('intersect', deg.up);
deg.dn3<-Reduce('intersect', deg.dn);
CreateDatatable(tbl[deg.up3, , drop=FALSE], paste(path.deg, 'up_all3.html', sep='/'), 
                caption='Genes up-regulated in all 3 comparisons')->f; 
CreateDatatable(tbl[deg.dn3, , drop=FALSE], paste(path.deg, 'down_all3.html', sep='/'), 
                caption='Genes down-regulated in all 3 comparisons')->f;

lns<-sapply(fn.tbl, function(f) {
  l<-paste('  - [', names(f), '](', paste('DEG', f, sep='/'), ')', sep=''); 
  paste(l, collapse='\n'); 
})

lns.up<-paste(' - [All 3 comparisons](DEG/up_all3.html)', lns[[1]], sep='\n'); 
lns.dn<-paste(' - [All 3 comparisons](DEG/down_all3.html)', lns[[2]], sep='\n'); 
```

<div align='center'>
```{r deg_overlap_up, include=TRUE, fig.width=6, fig.height=6, out.width='480px'}
PlotVenn3Way(deg.up, names=comp)->x;
```
</div>

<div style="color:darkblue; padding:0 2.5cm">
**Figure 2A.** Overlapping of DEGs, higher expression comparing to control groups. Click links to view overlapping genes:

`r lns.up`
</div>

<div align='center'>
```{r deg_overlap_dn, include=TRUE, fig.width=6, fig.height=6, out.width='480px'}
PlotVenn3Way(deg.dn, names=comp)->x; 
```
</div>

<div style="color:darkblue; padding:0 2.5cm">
**Figure 2B.** Overlapping of DEGs, lower expression comparing to control groups. Click links to view overlapping genes:

`r lns.dn`
</div>

`r home.url` 

## ANOVA

```{r anova, include=FALSE}
d <- expr.all;

# Prepare for ANOVA
s<-smpl[colnames(d), ]
for (i in 1:ncol(s)) s[[i]]<-as.factor(s[[i]]);
s$d<-d[1, ];
d0 <- d;
if (!prms$gene$processing$logged) d0 <- log2(d0+1);
if (!prms$gene$processing$normalized) d0 <- NormLoess(d0); 
d.anova<-lapply(1:nrow(d0), function(i) d0[i, ]);
f<-formula(prms$gene$anova$formula);

# Run ANOVA
paov <- parallel::mclapply(d.anova, function(x) {
  s$d<-x;
  smm<-summary(aov(f, data=s))[[1]];
  if (length(smm)==1) smm[[1]] else smm;
}, mc.cores=prms$gene$anova$core);

# sort column names
cnm<-rownames(paov[[1]]); 
cnm<-gsub(' ', '', cnm[-length(cnm)]); 
cnm0<-c(prms$gene$anova$f1, prms$gene$anova$f2, paste(prms$gene$anova$f1, prms$gene$anova$f2, sep=':'));

# p value table
paov<-t(sapply(paov, function(x) x[-nrow(x), 5])); 
dimnames(paov)<-list(rownames(d), cnm); 
paov<-paov[, c(cnm0, setdiff(cnm, cnm0))]; 
paov<-paov[order(paov[, 3]), ];
colnames(paov)<-paste('P', colnames(paov), sep='_'); 
sig.aov<-paov[paov[, 3]<=prms$gene$anova$p, , drop=FALSE]; 

# full table
m <- sapply(grps, function(g) exp(log(2)*rowMeans(d0[, g, drop=FALSE]))); 
aov.stat <- cbind(m[rownames(paov), ], paov); 
cnm <- c(colnames(anno), colnames(aov.stat));
aov.tbl <- data.frame(tbl[rownames(paov), colnames(anno)], FormatNumeric(aov.stat), stringsAsFactors = FALSE);
colnames(aov.tbl) <- cnm; 
CreateDatatable(aov.tbl, paste(path.tbl, 'anova_table.html', sep='/'), caption='ANOVA results'); 

# Write out
saveRDS(d, paste(path.r, 'expr.rds', sep='/'));
saveRDS(anno[rownames(d), ], paste(path.r, 'anno.rds', sep='/'));
saveRDS(aov.stat, paste(path.r, 'anova_stat.rds', sep='/'));
write.csv(d, paste(path.tbl, 'expr.csv', sep='/'));
write.csv(anno[rownames(d), ], paste(path.tbl, 'anno.csv', sep='/'));
write.csv(aov.stat, paste(path.tbl, 'anova_stat.csv', sep='/'));
```

2-way ANOVA analysis was performed to identify genes responding to **_`r prms$gene$anova$f2`_** differently in different **_`r prms$gene$anova$f1`_**. The analysis reported 3 p values, corresponding to the effect of **_`r prms$gene$anova$f2`_**, **_`r prms$gene$anova$f1`_**, and their interaction. The analysis identified `r nrow(sig.aov)` significant genes with interaction p values less than `r prms$gene$anova$p`. The full ANOVA results were summarized in a table [here](table/anova_table.html). 

<div align='center'>
```{r aov_top, out.width='800px'}
mn <- apply(pvl, 1, min); 
sg <- paov[rownames(paov) %in% rownames(pvl)[mn < 0.05], , drop=FALSE]; 
id <- rownames(sg)[1:min(4, nrow(sg))];
ttl <- sapply(id, function(id) if (is.na(anno[id, 1])) id else paste(id, anno[id, 1], sep=': ')); 

par(mar=c(8, 2, 2, 2), mfrow=c(2,2), omi=c(0, 1, 0, 0)); 
c<-rep(c("#FF666633", "#FF000088", "#66FF6633", "#00FF0088", "#6666FF33", "#0000FF88"), sapply(grps, length)); 
for (i in 1:length(id)) {
  barplot(d0[id[i], ], col=c, las=3, ylab='Normalized expression level', cex.lab=1.5, main=ttl[i], cex.main=1.5);
  if (i==1) legend('topleft', legend=names(grps), pch=15, col=unique(c), bty='n', cex = 0.5);
}
title(ylab='Normalized expression level', outer=TRUE, line=1, cex.lab=2); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 3.** Examples: the top 4 genes having the most significant interactive p value, among the genes with significant differential expression in at least one of the three pairwise comparisons. 
</div>

`r home.url` 

# Gene set-level comparison

Genes are often grouped into pre-defined gene sets according to their function, interaction, location, etc. Analysis then can be performed on genes in the same gene set as a unit instead of individual genes.

## Gene set average

```{r geneset_average, include=FALSE}
# Gene sets, filtering by size
lst<-gset$list;
gns<-unlist(lst, use.names=FALSE); 
gst<-rep(names(lst), sapply(lst, length));
lst<-split(gns[gns %in% gid], gst[gns %in% gid]);
sz<-sapply(lst, length);
lst<-lst[sz>=prms$geneset$min & sz<=prms$geneset$max];
sz<-sapply(lst, length); 

# split gene log-ratio into gene sets
gns<-unlist(lst, use.names=FALSE);
gst<-rep(names(lst), sapply(lst, length));
lr<-l2r[gns, ];
splt<-apply(lr, 2, function(lr) split(lr, gst)); 
mean.gs<-sapply(splt, function(x) sapply(x, mean));

mn<-apply(mean.gs, 1, min);
mx<-apply(mean.gs, 1, max);
gs.tbl<-gset[[1]][rownames(mean.gs), ];
gs.tbl$Name<-AddHref(gs.tbl$Name, gs.tbl$URL);
gs.tbl<-cbind(gs.tbl[, 1:3], FormatNumeric(cbind(mean.gs, Diff=mx-mn)));

corr<-round(sapply(1:3, function(i) cor(mean.gs[, ind1[i]], mean.gs[, ind2[i]])), 3);
lns<-paste('  - Corr(', comp[ind1], ':', comp[ind2], ') = ', corr, sep='');
lns<-paste(lns, collapse='\n'); 

CreateDatatable(gs.tbl, paste(path.tbl, 'geneset_table.html', sep='/'), rownames = FALSE, caption='Gene set average'); 
saveRDS(gs.tbl, paste(path.r, 'geneset_stat.rds', sep='/'));
write.csv(gs.tbl, paste(path.tbl, 'geneset_stat.csv', sep='/')); 
```

Average differential expression of genes in the same gene set. The gene set-level mean of log-ratio were summarized in this table [here](table/geneset_table.html).

<div align='center'>
```{r geneset_average_plot, include=TRUE, fig.width=8, fig.height=4, out.width='800px'}
par(mfrow=c(1, 2));

col<-rep('#88888888', nrow(mean.gs));
col[mn>=1]<-'#FF0000';
col[mx<=-1]<-'#0000FF';

scatterplot3d(mean.gs, pch=19, angle=45, cex.symbols =.25, color=col); 
scatterplot3d(mean.gs, pch=19, angle=135, cex.symbols =.25, color=col); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 4.**  Each dot represents a gene set and the average log-ratio of all genes in this gene set. The same 3D plot was showed in 2 different angles.

`r lns`
</div>

`r home.url` 

## Gene set over-representation analysis (ORA)

```{r ora, include=FALSE}
path.ora<-paste(path, 'ORA', sep='/');
if (!file.exists(path.ora)) dir.create(path.ora, recursive = TRUE); 

ora.stat<-lapply(ora, function(x) lapply(x[1:2], function(x) x[[1]])); 
ora.sets<-lapply(ora.stat, function(x) lapply(x, rownames)); 
ora.univ<-lapply(ora, function(x) lapply(x[1:2], function(x) rownames(x[[3]]))); 
ora.univ<-Reduce('union', c(ora.univ[[1]], ora.univ[[2]])); 
ora.n<-lapply(ora, function(o) sapply(o[1:2], function(o) o[[3]][, 'n11'][ora.univ])); 
ora.n<-do.call('cbind', ora.n);
ora.n[is.na(ora.n)]<-0; 
rownames(ora.n)<-ora.univ;
colnames(ora.n)<-paste('N', colnames(ora.n), sep='_'); 

ora.or<-lapply(ora, function(o) sapply(o[1:2], function(o) {
  c<-o[[3]]; 
  c<-(c[, 1]/c[, 2])*(c[, 4]/c[, 3]); 
  c<-c[ora.univ];
  c;
})); 
ora.or<-do.call('cbind', ora.or);
ora.or[is.na(ora.or)]<-1; 
rownames(ora.or)<-ora.univ;
colnames(ora.or)<-paste('OR', colnames(ora.or), sep='_'); 
ora.or.hi<-ora.or[, seq(1, ncol(ora.or), 2)]; 
ora.or.lo<-ora.or[, seq(2, ncol(ora.or), 2)]; 
ora.or<-cbind(ora.or.hi, OR_High_Range=apply(ora.or.hi, 1, function(x) max(x)-min(x)),
              ora.or.lo, OR_Low_Range=apply(ora.or.lo, 1, function(x) max(x)-min(x))); 

ora.p<-lapply(ora.stat, function(o) sapply(o, function(o) o[, 'P_HyperGeo'][ora.univ]));
ora.p<-do.call('cbind', ora.p); 
rownames(ora.p)<-ora.univ;
colnames(ora.p)<-paste('P', colnames(ora.p), sep='_'); 
ora.p[is.na(ora.p)]<-0.5;
ora.p.hi<-ora.p[, seq(1, ncol(ora.p), 2)]; 
ora.p.lo<-ora.p[, seq(2, ncol(ora.p), 2)]; 
ora.p<-cbind(ora.p.hi, P_High_Range=apply(ora.p.hi, 1, function(x) min(x)/max(x)),
              ora.p.lo, P_Low_Range=apply(ora.p.lo, 1, function(x) min(x)/max(x))); 

ora.tbl<-FormatNumeric(cbind(ora.n, ora.or, ora.p)); 
ora.tbl<-cbind(gset[[1]][rownames(ora.tbl), ], ora.tbl); 
ora.tbl$Name<-AddHref(ora.tbl$Name, ora.tbl$URL); 
ora.tbl<-ora.tbl[, -5];
CreateDatatable(ora.tbl, paste(path, 'table/ora_table.html', sep='/'), caption='Combined table of ORA statistics'); 

saveRDS(ora.tbl, paste(path.r, 'ora_stat.rds', sep='/'));
write.csv(ora.tbl, paste(path.tbl, 'ora_stat.csv', sep='/')); 

# split significant gene sets into subgroups by sources
gset.src<-sort(unique(gset[[1]][, 'Source'])); 
ora.s<-do.call('c', ora.stat); 
fn.tbl<-lapply(names(ora.s), function(nm) {
  s<-ora.s[[nm]]; 
  lapply(gset.src, function(src) {
    a<-gset[[1]][rownames(s), ];
    a$Name<-AddHref(a$Name, a$URL)
    a<-a[, 1:3]; 
    t<-cbind(a[a$Source==src, , drop=FALSE], FormatNumeric(s[a$Source==src, , drop=FALSE]))[, -1]; 
    f<-paste('ORA/', src, '_', nm, '.html', sep=''); 
    CreateDatatable(t, paste(path, f, sep='/'), rownames = FALSE, caption=paste(src, nm, sep=': '));
    c<-nrow(t); 
    names(c)<-f;
    c;
  });
}); 
lnk<-sapply(fn.tbl, function(fn) {
  c<-as.vector(unlist(fn));
  f<-sapply(fn, names); 
  paste('[', c, '](', f, ')', sep='');
});
colnames(lnk) <- paste(rep(comp, each=2), paste('Higher_in_', unlist(grp), sep=''), sep='::'); 
rownames(lnk) <- gset.src;
```

Each 2-group comparison performs gene set over-representation analysis (ORA) that identifies gene sets over-represented with differentially expressed genes. The results of ORA of both 2-group comparisons are summarized and compared here.  The ORA of each gene set reports an odds ratio and p value. These statistics from both comparisons were combined and listed side-by-side, as well as the difference of their odds ratios and ratio of their p values (p set to 0.5 when not available), in this table [here](table/ora_table.html)

<div style="color:darkblue; padding:0 2cm">
**Table 2.** Gene sets were broken down into subgroups by their sources. Click on the numbers of over-represented gene sets to see a full list. 
</div>

<div align='center', style="padding:0 2cm">
`r kable(lnk, align=c('c', 'c'))`
</div>

```{r ora_overlap, include=FALSE}
ora.up<-lapply(ora.sets, function(x) x[[1]]); 
ora.dn<-lapply(ora.sets, function(x) x[[2]]); 

ora.all<-list(up=ora.up, down=ora.dn); 
fn.tbl<-lapply(names(ora.all), function(nm) {
  gs<-ora.all[[nm]]; 
  x<-c('up'='Higher', 'down'='Lower'); 
  sapply(1:3, function(i) {
    c<-comp[c(ind1[i], ind2[i])]; 
    g<-intersect(gs[[ind1[i]]], gs[[ind2[i]]]); 
    t<-ora.tbl[g, c(1:4, grep(x[nm], colnames(ora.tbl))), drop=FALSE]; 
    f<-paste(paste(nm, c[1], c[2], sep='_'), '.html', sep=''); 
    CreateDatatable(t, paste(path.ora, f, sep='/'), caption=paste(nm, ', ', c[1], ' and ', c[2], sep='')); 
    names(f)<-paste(c, collapse=' and '); 
    f;
  })
}); 

ora.up3<-Reduce('intersect', ora.up);
ora.dn3<-Reduce('intersect', ora.dn);
CreateDatatable(ora.tbl[ora.up3, c(1:4, grep('Higher', colnames(ora.tbl))), drop=FALSE], paste(path.ora, 'up_all3.html', sep='/'), caption='Gene sets up-regulated in all 3 comparisons')->f; 
CreateDatatable(ora.tbl[ora.dn3, c(1:4, grep('Lower', colnames(ora.tbl))), drop=FALSE], paste(path.ora, 'down_all3.html', sep='/'), caption='Gene sets down-regulated in all 3 comparisons')->f;

lns<-sapply(fn.tbl, function(f) {
  l<-paste('  - [', names(f), '](', paste('ORA', f, sep='/'), ')', sep=''); 
  paste(l, collapse='\n'); 
})

lns.up<-paste(' - [All 3 comparisons](ORA/up_all3.html)', lns[[1]], sep='\n'); 
lns.dn<-paste(' - [All 3 comparisons](ORA/down_all3.html)', lns[[2]], sep='\n'); 
```

<div align='center'>
```{r ora_overlap_up, include=TRUE, fig.width=6, fig.height=6, out.width='480px'}
PlotVenn3Way(ora.up, names=comp)->x;
```
</div>

<div style="color:darkblue; padding:0 2.5cm">
**Figure 5A.** The overlapping of over-represented gene sets by up-regulated genes in all 3 comparisons. Click links to view overlapping gene sets:

`r lns.up`
</div>

<div align='center'>
```{r ora_overlap_dn, include=TRUE, fig.width=6, fig.height=6, out.width='480px'}
PlotVenn3Way(ora.dn, names=comp)->x;
```
</div>

<div style="color:darkblue; padding:0 2.5cm">
**Figure 5B.** The overlapping of over-represented gene sets by down-regulated genes in all 3 comparisons. Click links to view overlapping gene sets:

`r lns.dn`
</div>

`r home.url` 

## Gene set enrichment analysis (GSEA) 

```{r gsea, include=FALSE}
path.gsea<-paste(path, 'GSEA', sep='/');
if (!file.exists(path.gsea)) dir.create(path.gsea, recursive = TRUE); 

gsea.stat<-lapply(gsea, function(x) x$stat); 
gsea.stat<-lapply(gsea.stat, function(x) {
  rownames(x)<-paste(x[[1]], x[[2]], sep=':'); 
  x;
})
gsea.univ<-Reduce('union', lapply(gsea.stat, rownames)); 
gsea.stat<-lapply(gsea.stat, function(x) x[gsea.univ, ]); 
for (i in 1:length(gsea.stat)) rownames(gsea.stat[[i]])<-gsea.univ; 

x<-sapply(1:length(gsea.stat), function(i) 1-as.numeric(is.na(gsea.stat[[i]][, 1])));
y<-max.col(x, 'first');
gsea.anno<-lapply(1:3, function(i) gsea.stat[[i]][y==i, 1:3, drop=FALSE]);
gsea.anno<-do.call('rbind', gsea.anno)[gsea.univ, ]; 

gsea.tbl<-FormatNumeric(do.call('cbind', lapply(gsea.stat, function(x) x[, 4:6]))); 
gsea.tbl<-cbind(gsea.anno, gsea.tbl); 
colnames(gsea.tbl)[4:ncol(gsea.tbl)]<-paste(unlist(lapply(gsea.stat, function(x) colnames(x)[4:6]), use.names=FALSE), rep(names(grp), each=3), sep='_'); 
CreateDatatable(gsea.tbl, paste(path.tbl, 'gsea_table.html', sep='/'), rownames = FALSE, caption='GSEA statistics'); 

saveRDS(gsea.tbl, paste(path.r, 'gsea_stat.rds', sep='/'));
write.csv(gsea.tbl, paste(path.tbl, 'gsea_stat.csv', sep='/')); 

gsea.sets<-lapply(gsea.stat, function(x) split(x[x[, 'PValue']<=0.05, -ncol(x)], x[x[, 'PValue']<=0.05, ncol(x)])[2:1]); 
gsea.s<-do.call('c', gsea.sets); 
names(gsea.s)<-names(ora.s); 
gset.cll<-sort(unique(unlist(lapply(gsea.s, function(x) x$Collection), use.names=FALSE)));
fn.tbl<-lapply(names(gsea.s), function(nm) {
  s<-gsea.s[[nm]]; 
  lapply(gset.cll, function(cll) {
    t<-s[s$Collection==cll, , drop=FALSE]; 
    t<-FormatNumeric(t); 
    f<-paste('GSEA/', cll, '_', nm, '.html', sep=''); 
    CreateDatatable(t, paste(path, f, sep='/'), rownames = FALSE, caption=paste(cll, nm, sep=': '));
    c<-nrow(t); 
    names(c)<-f;
    c;
  });
}); 
lnk<-sapply(fn.tbl, function(fn) {
  c<-as.vector(unlist(fn));
  f<-sapply(fn, names); 
  paste('[', c, '](', f, ')', sep='');
});
colnames(lnk) <- paste(rep(comp, each=2), paste('Higher_in_', unlist(grp), sep=''), sep='::'); 
rownames(lnk) <- gset.cll;

gsea.nes<-sapply(gsea.stat, function(x) x[, 'NES']);
gsea.nes[is.na(gsea.nes)]<-0; 
gsea.p<-sapply(gsea.stat, function(x) x[, 'PValue']);
gsea.p[is.na(gsea.p)]<-1; 

corr<-round(sapply(1:3, function(i) cor(gsea.nes[, ind1[i]], gsea.nes[, ind2[i]])), 3);
lns<-paste('  - Corr(', comp[ind1], ':', comp[ind2], ') = ', corr, sep='');
lns<-paste(lns, collapse='\n'); 
```

Each 2-group comparison performs gene set enrichment analysis (GSEA) on genes ranked by their differential expression. The results of GSEA of both 2-group comparisons are summarized and compared here. The GSEA of each gene set reports an enrichment score and p value. These statistics from both comparisons were combined and listed side-by-side in this table [here](table/gsea_table.html)

<div style="color:darkblue; padding:0 2cm">
**Table 3.** Gene sets were broken down into subgroups by collections. Click on the numbers of enriched gene sets to see a full list. 
</div>

<div align='center', style="padding:0 2cm">
`r kable(lnk, align='c')`
</div>

<div align='center'>
```{r nes, fig.width=8, fig.height=4, out.width='800px'}
n.sig<-apply(gsea.p, 1, function(p) length(p[p<0.01])); 
col<-c('lightgrey', 'yellow', 'orange', 'red')[n.sig+1]; 
par(mfrow=c(1, 2)); 
scatterplot3d(gsea.nes, pch=19, angle=45, cex.symbols =.25, color=col); 
scatterplot3d(gsea.nes, pch=19, angle=135, cex.symbols =.25, color=col); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 6.** This plot shows the global correlation (correlation coefficient = `r corr`) of nominal enrichment scores between the 3 pairwise comparisons: **`r comp[1]`**, **`r comp[2]`**, and **`r comp[3]`**. The same 3D plot was showed in 2 different angles. Gene sets obtained p values less than 0.01 from any 1, any 2, or all 3 comparisons were highlighted in yellow, orange, or red respectively. The correlatio coefficients between enrichment scores of each pair of comparisons are: 

`r lns`
</div>

```{r gsea_overlap, include=FALSE}
gsea.up<-lapply(gsea.sets, function(x) rownames(x[[1]])); 
gsea.dn<-lapply(gsea.sets, function(x) rownames(x[[2]])); 

gsea.all<-list(up=gsea.up, down=gsea.dn); 
fn.tbl<-lapply(names(gsea.all), function(nm) {
  gs<-gsea.all[[nm]]; 
  sapply(1:3, function(i) {
    c<-comp[c(ind1[i], ind2[i])]; 
    g<-intersect(gs[[ind1[i]]], gs[[ind2[i]]]); 
    t<-gsea.tbl[g, , drop=FALSE]; 
    f<-paste(paste(nm, c[1], c[2], sep='_'), '.html', sep=''); 
    CreateDatatable(t, paste(path.gsea, f, sep='/'), rownames = FALSE, caption=paste(nm, ', ', c[1], ' and ', c[2], sep='')); 
    names(f)<-paste(c, collapse=' and '); 
    f;
  })
}); 

gsea.up3<-Reduce('intersect', gsea.up);
gsea.dn3<-Reduce('intersect', gsea.dn);
CreateDatatable(gsea.tbl[gsea.up3, , drop=FALSE], paste(path.gsea, 'up_all3.html', sep='/'), rownames=FALSE, caption='Gene sets up-regulated in all 3 comparisons')->f; 
CreateDatatable(gsea.tbl[gsea.dn3, c(1:4, grep('Lower', colnames(gsea.tbl))), drop=FALSE], rownames=FALSE, paste(path.gsea, 'down_all3.html', sep='/'), caption='Gene sets down-regulated in all 3 comparisons')->f;

lns<-sapply(fn.tbl, function(f) {
  l<-paste('  - [', names(f), '](', paste('GSEA', f, sep='/'), ')', sep=''); 
  paste(l, collapse='\n'); 
})

lns.up<-paste(' - [All 3 comparisons](GSEA/up_all3.html)', lns[[1]], sep='\n'); 
lns.dn<-paste(' - [All 3 comparisons](GSEA/down_all3.html)', lns[[2]], sep='\n'); 
```

<div align='center'>
```{r gsea_overlap_up, include=TRUE, fig.width=6, fig.height=6, out.width='480px'}
PlotVenn3Way(gsea.up, names=comp)->x;
```
</div>

<div style="color:darkblue; padding:0 2.5cm">
**Figure 7A.** The overlapping of enriched gene sets by up-regulated genes in all 3 comparisons. Click links to view overlapping gene sets:

`r lns.up`
</div>

<div align='center'>
```{r gsea_overlap_dn, include=TRUE, fig.width=6, fig.height=6, out.width='480px'}
PlotVenn3Way(gsea.dn, names=comp)->x;
```
</div>

<div style="color:darkblue; padding:0 2.5cm">
**Figure 7B.** The overlapping of enriched gene sets by down-regulated genes in all 3 comparisons. Click links to view overlapping gene sets:

`r lns.dn`
</div>

`r home.url` 

## Gene clustering

```{r clustering, include=FALSE}
path.cl<-paste(path, 'CLUSTER', sep='/');
if (!file.exists(path.cl)) dir.create(path.cl, recursive = TRUE);

# normalize data
d1 <- lapply(res, function(x) {
  y<-x$inputs; 
  y$expr[, unlist(y$comparison)];
});
d1 <- do.call('cbind', d1); 
if (!prms$gene$processing$logged) d1 <- log2(d1+1); 
if (!prms$gene$processing$normalized) d1 <- NormLoess(d1+1); 
for (i in 1:3) {
  y <- res[[i]]$inputs$comparison;
  d1[, unlist(y)] <- d1[, unlist(y)] - rowMeans(d1[, y[[1]]]);
  d1[, unlist(y)] <- t(apply(d1[, unlist(y)], 1, function(x) x/sd(x))); 
}

# select seeds
x<-paov[order(paov[, 3]), ];
x<-x[x[,3]<=prms$geneset$cluster$panova, , drop=FALSE]; 
d0<-d1[rownames(d1) %in% rownames(x), ]; 
d0<-d0[1:min(nrow(d0), prms$geneset$cluster$top), , drop=FALSE];

# initiate clusters
hc<-hclust(as.dist(1-cor(t(d0)))); 
cl<-cutree(hc, k=prms$geneset$cluster$seed);
cl<-split(names(cl), cl); 

# merge similar clusters
flag<-TRUE;
while(flag) {
  cat('Number of clusters ', length(cl), '\n'); 
  ms<-sapply(cl, function(cl) colMeans(d0[cl, ])); 
  tr<-cutree(hclust(as.dist(1-cor(ms))), k=length(cl)-1); # find the 2 most similar clusters
  i<-tr[duplicated(tr)];
  c<-ms[, tr==i]; 
  r<-cor(c[, 1], c[, 2]); 
  p<-cor.test(c[, 1], c[, 2])$p.value[[1]]; 
  if (r>prms$geneset$cluster$merge$corr & p<prms$geneset$cluster$merge$p) {
    cl[tr==i][[1]]<-as.vector(unlist(cl[tr==i]));
    cl<-cl[names(cl)!=names(i)]; 
  } else flag<-FALSE;
}

# Sort clusters
m<-sapply(cl, function(cl) colMeans(d1[cl, ])); 
ind<-apply(m, 2, function(x) which(x==max(x)));  
cl<-cl[order(ind)]; 

# re-cluster genes
reCl<-function(d, cl, r, dif, nmax) {
  d <- d[!is.na(rowMeans(d)), , drop=FALSE];
  cl <- lapply(cl, function(cl) cl[cl %in% rownames(d)]); 
  md<-sapply(cl, function(cl) apply(d[cl[cl %in% rownames(d)], , drop=FALSE], 2, median));
  corr<-cor(t(d), md); 
  c<-lapply(1:ncol(corr), function(i) {
    corr<-corr[rev(order(corr[, i])), ]; 
    mx<-apply(corr[, -i, drop=FALSE], 1, max);
    id<-rownames(corr)[corr[, i]>=r & (corr[, i]-mx)>dif]; 
    id[1:min(length(id), nmax)]; 
  });
  c;
}
for (i in 1:prms$geneset$cluster$recluster$time) 
  cl<-reCl(d0, cl, prms$geneset$cluster$recluster$corr, prms$geneset$cluster$recluster$diff, prms$geneset$cluster$recluster$nmax); 
cls<-reCl(d1, cl, prms$geneset$cluster$recluster$corr, prms$geneset$cluster$recluster$diff, prms$geneset$cluster$recluster$nmax); 
ms<-t(sapply(cls, function(x) colMeans(d1[x, ]))); 
rownames(ms)<-names(cls)<-names(cl)<-paste('Cluster', 1:length(cls), sep='_'); 

fn.htmp<-sapply(names(cls), function(nm) {
  x<-d1[cls[[nm]], ];
  f<-paste(path.cl, '/Heatmap_', nm, '.pdf', sep=''); 
  sz<-CalculateColoredBlockSize(x);
  pdf(f, width = max(sz[2]/3, sz[1]), height = sz[2]);
  PlotColoredBlock(x, num.breaks = 31, key = 'Normalized expression', groups = grps); 
  dev.off();
  f;
});

saveRDS(list(cluster=cls, initial=cl, seed=d0, data=d1), file=paste(path.r, 'cluster.rds', sep='/')); 
```

The top `r nrow(d0)` genes with significant ANOVA p values (p <= 'r prms$geneset$cluster$panova') were used as seeds to perform a gene-gene clustering analysis and `r length(cls)` clusters were identified. ORA was performed on the clusters to identify their functional association (see table below); 

```{r clustering_ora, include=FALSE}
fn.tbl<-lapply(names(cls), function(nm) {
  cat(nm, '\n');
  cl<-cls[[nm]]; 
  s<-TestGSE(cl, gid, gset[[2]])[[1]]; 
  t<-gset[[1]][rownames(s), ];
  t$Name<-AddHref(t$Name, t$URL); 
  t<-cbind(t[, 1:3], FormatNumeric(s)); 
  f<-paste('CLUSTER/ORA_', nm, '.html', sep='');
  CreateDatatable(t, paste(path, f, sep='/'), rownames=FALSE, caption=paste('Over-represented gene set in', nm));
  c<-nrow(t); 
  names(c)<-f;
  t<-do.call('cbind', lapply(stat, function(s) s[cl, c(1, 2, 4, 5, 6)])); 
  t<-FormatNumeric(t);
  colnames(t)[-grep('^Mean', colnames(t))]<-paste(colnames(t)[-grep('^Mean', colnames(t))], rep(names(grp), each=3), sep='_'); 
  t<-data.frame(anno[rownames(t), ], t, stringsAsFactors = FALSE);
  CreateDatatable(t, paste(path.cl, '/', nm, '.html', sep=''), caption=nm);
  c;
}); 
mm<-round(sapply(grps, function(x) rowMeans(ms[, x, drop=FALSE])), 4);
n<-as.vector(unlist(fn.tbl)); 
sz<-sapply(cls, length);
lnk<-paste('[', n, '](', sapply(fn.tbl, names), ')', sep='');
cl.tbl<-data.frame(Size=sz, mm, Gene_set=lnk, stringsAsFactors = FALSE);
cl.tbl[, 1]<-paste('[', sz, '](CLUSTER/', names(cls), '.html)', sep=''); 
cl.tbl<-cbind(ID=rownames(cl.tbl), cl.tbl); 
cl.tbl[, 1]<-paste('[', cl.tbl[, 1], '](CLUSTER/Heatmap_', names(cls), '.pdf)', sep=''); 
colnames(cl.tbl)[3:8] <- paste(rep(comp, each=2), unlist(grp), sep='::');
```

<div style="color:darkblue; padding:0 0.5cm">
**Table 4.** This table lists the number of genes in each cluster (click the numbers to see gene lists), the average expression of all genes in a cluster of all sample groups (normalized so the mean of the control groups equals to 0 and the mean is the number of standard deviations), and then the gene sets over-represented in each cluster (click the numbers to see gene set lists). 
</div>

<div align='center', style="padding:0 0.5cm">
`r kable(cl.tbl, row.names=FALSE, align='c')`
</div>

<div align='center'>
```{r clustering_heatmap, include=TRUE, fig.width=CalculateColoredBlockSize(ms)[1], fig.height=CalculateColoredBlockSize(ms)[2], out.width='600px'}
mx<-max(abs(ms), na.rm=TRUE); 
PlotColoredBlock(ms, min=-1*mx, max=mx, key='Normalized expression level', groups=grps); 
```
</div>

<div style="color:darkblue; padding:0 2cm">
**Figure 8.**  This plot shows below the average expression levels of each cluster. Data was normalized before the analysis, so the mean of the control groups was zero and the standard deviation of all samples of each gene was 1.0. Values indicate number of standard deviation from mean of relative control group.
</div>

<div align='center'>
```{r clustering_mean, include=TRUE, fig.width=8, fig.height=8, out.width='750px'}
ms<-t(ms); 
m<-sapply(grps, function(g) colMeans(ms[g, , drop=FALSE]));
se<-sapply(grps, function(g) apply(ms[g, , drop=FALSE],  2, function(x) sd(x)/sqrt(length(x)))); 
PlotSeries(m, se, c('', names(grps))); 
```
</div>

<div style="color:darkblue; padding:0 1cm">
**Figure 9.** This plot summarizes the group means and standard errors of all clusters.
</div>

`r home.url` 

***

# Appendix 

Check out the **[RoCA home page](http://zhezhangsh.github.io/RoCA)** for more information.  

## Reproduce this report

To reproduce this report: 

  1. Find the data analysis template you want to use and an example of its pairing YAML file  [here](https://github.com/zhezhangsh/RoCA/wiki/Templates-and-examples) and download the YAML example to your working directory

  2. To generate a new report using your own input data and parameter, edit the following items in the YAML file:

    - _output_        : where you want to put the output files
    - _home_          : the URL if you have a home page for your project
    - _analyst_       : your name
    - _description_   : background information about your project, analysis, etc.
    - _input_         : where are your input data, read instruction for preparing them
    - _parameter_     : parameters for this analysis; read instruction about how to prepare input data

  3. Run the code below within ***R Console*** or ***RStudio***, preferablly with a new R session:

```{r reproduce_report, eval=FALSE, echo=TRUE}
if (!require(devtools)) { install.packages('devtools'); require(devtools); }
if (!require(RCurl)) { install.packages('RCurl'); require(RCurl); }
if (!require(RoCA)) { install_github('zhezhangsh/RoCAR'); require(RoCA); }

CreateReport(filename.yaml);  # filename.yaml is the YAML file you just downloaded and edited for your analysis
```

If there is no complaint, go to the _output_ folder and open the ***index.html*** file to view report. 

## Session information

```{r session_info, echo=FALSE}
sessionInfo(); 
```

`r home.url`

***
**END OF DOCUMENT**
